{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6d6f36b7",
   "metadata": {},
   "source": [
    "# Benchmarking Pyxis\n",
    "\n",
    "The following notebook contains tests for benchmarking `pyxis` functions/commands for runtime and memory usage, both against example test files (several provided in the `example-files` directory) and also against a real [ChIP peaks ISL1 transcription factor dataset](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM5838045) (provided in the `analysis` directory). \n",
    "\n",
    "There are also comparisons drawn to [HOMER](http://homer.ucsd.edu/homer/ngs/peakMotifs.html)'s analogous command, `findMotifsGenome.pl`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2f031ded",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/esquan/final_project\n"
     ]
    }
   ],
   "source": [
    "# should cd into outer pyxis directory\n",
    "%cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "85f3b56f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "import pyfaidx\n",
    "import pyxis.myutils\n",
    "import random\n",
    "import scipy.stats\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "44dd85e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "!python setup.py install --user > path.txt 2> /dev/null\n",
    "# determining where pyxis script installed to, and saving that path to export later\n",
    "with open('path.txt', 'r') as file:\n",
    "    for line in file:\n",
    "        if 'Installing pyxis script to' in line:\n",
    "            path = line.split()[-1]\n",
    "            f = open(\"scriptpath.txt\", \"w\")\n",
    "            f.write(path)\n",
    "            f.close()\n",
    "            break"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b842b720",
   "metadata": {},
   "source": [
    "## I. Runtime"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "809a4d8f",
   "metadata": {},
   "source": [
    "### Loading in the Example Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1dff0043",
   "metadata": {},
   "outputs": [],
   "source": [
    "# example-files directory test files\n",
    "cwd = os.getcwd()\n",
    "f_peaks_dir = cwd + '/example-files/peaks.bed'      # foreground peaks file\n",
    "b_peaks_dir = cwd + '/example-files/background.bed' # background peaks file\n",
    "ref_dir = cwd + '/example-files/ref.fa'             # reference genome file\n",
    "pwms_dir = cwd + '/example-files/test.pwms'         # file of known PWMs\n",
    "\n",
    "# index reference genome file\n",
    "test_ref = pyfaidx.Fasta(ref_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c770ba07",
   "metadata": {},
   "source": [
    "### Testing Runtime of Various myutil.py Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d4b9e391",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runtime for ReadBED:\n",
      "183 µs ± 9.18 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n",
      "\n",
      "Runtime for ReadPWMS:\n",
      "247 µs ± 6.29 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)\n",
      "\n",
      "Runtime for WriteFastaSeq:\n",
      "8.87 µs ± 139 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)\n",
      "\n",
      "Runtime for ComputeNucFreqs:\n",
      "23.9 µs ± 709 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n"
     ]
    }
   ],
   "source": [
    "print(\"Runtime for ReadBED:\")\n",
    "%timeit readbed = pyxis.myutils.ReadBED(f_peaks_dir, test_ref)\n",
    "print(\"\\nRuntime for ReadPWMS:\")\n",
    "%timeit readpwms = pyxis.myutils.ReadPWMS(pwms_dir)\n",
    "print(\"\\nRuntime for WriteFastaSeq:\")\n",
    "%timeit pyxis.myutils.WriteFastaSeq(test_ref, 'chr6', 1, 100)\n",
    "print(\"\\nRuntime for ComputeNucFreqs:\")\n",
    "f_seqs = ['ACTAGCTACG', 'TAGCATGCTAGAC', 'TACTATCATGGGT', 'GTAGCTAGTACAGTACAGTAA', 'ATCAGTAC', 'TAGCTAGCTA', 'GTCAGTACGATAC']\n",
    "%timeit pyxis.myutils.ComputeNucFreqs(f_seqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "dd706dc4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Runtime for RandomBkSequence:\n",
      "88.2 µs ± 1.15 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n",
      "\n",
      "Runtime for ScoreSeq:\n",
      "1.48 µs ± 14.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)\n",
      "\n",
      "Runtime for ReverseComplement:\n",
      "6.29 µs ± 46.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)\n",
      "\n",
      "Runtime for FindMaxScore:\n",
      "141 µs ± 2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nRuntime for RandomBkSequence:\")\n",
    "seq_lens = [10, 13, 13, 21, 8, 10, 13]\n",
    "%timeit bk_seqs = pyxis.myutils.RandomBkSequence(seq_lens, 7, test_ref)\n",
    "print(\"\\nRuntime for ScoreSeq:\")\n",
    "bach_pwm = np.array([\n",
    "    [-0.705,-0.938,-1.27,1.041],\n",
    "    [-2.694,-1.138,1.265,-2.694],\n",
    "    [-1.681,1.298,-2.819,-2.394],\n",
    "    [-1.138,-1.818,-3.126,1.245],\n",
    "    [-1.421,-1.389,1.147,-1.022],\n",
    "    [1.101,-1.818,-1.454,-0.516],\n",
    "    [-1.681,-0.108,0.978,-1.358],\n",
    "    [-2.961,-2.584,-3.126,1.343],\n",
    "    [-1.389,1.294,-2.584,-3.573],\n",
    "    [1.34,-2.819,-3.126,-2.584],\n",
    "    [-1.454,0.244,-0.163,0.495]\n",
    "])\n",
    "%timeit score = pyxis.myutils.ScoreSeq(bach_pwm, f_seqs)\n",
    "print(\"\\nRuntime for ReverseComplement:\")\n",
    "test_seq = 'GTAGCTAGCATTACATTACTGTACGTAC'\n",
    "%timeit pyxis.myutils.ReverseComplement(test_seq)\n",
    "print(\"\\nRuntime for FindMaxScore:\")\n",
    "%timeit pyxis.myutils.FindMaxScore(bach_pwm, test_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "34b62d36",
   "metadata": {},
   "outputs": [],
   "source": [
    "PWMList = [np.array([[-1.83361299, -2.15554108, -2.31903981,  0.81189105, -6.96289593,\n",
    "         -4.15554107, -1.67749378, -1.26245629, -0.14271704,  1.44649493,\n",
    "         -1.792971  , -2.71496849, -2.3779335 ,  1.16122531,  1.15084616,\n",
    "          1.62956103],\n",
    "        [-0.43933405,  1.17153032,  1.12456684, -4.37793349, -3.50346438,\n",
    "         -6.96289593, -0.71496849, -6.96289593, -0.00869969, -1.67749378,\n",
    "         -4.6409679 , -2.10491501,  1.34544303, -1.29047066, -1.40830715,\n",
    "         -2.71496849],\n",
    "        [ 0.22692855, -0.81314888, -3.962896  , -4.96289599, -4.6409679 ,\n",
    "         -5.96289597,  1.57626281, -3.3779335 , -0.67749379, -1.91850188,\n",
    "         -2.71496849,  1.58785078, -1.08025295, -2.87543316, -0.792971  ,\n",
    "         -1.71496849],\n",
    "        [ 0.85728296, -0.06807824,  0.63701684,  1.11391959,  1.94999333,\n",
    "          1.97079465, -3.3779335 ,  1.79865523,  0.56066595, -0.51995251,\n",
    "          1.81518112, -0.71496849, -0.33353938,  0.28503151, -0.27639548,\n",
    "         -1.15554108]]),\n",
    "np.array([[  0.55777767,  -0.41119543,   1.40925515,  -5.96578425,\n",
    "           1.97085365,  -4.96578427,   1.9441088 ,   1.87656606,\n",
    "          -1.41119543,   1.18903382],\n",
    "        [ -0.96578428,   0.78240857,  -4.96578427, -31.21928095,\n",
    "          -5.96578425,  -3.50635266,  -5.96578425,  -6.96578421,\n",
    "          -0.45798964,  -0.81603716],\n",
    "        [  0.31961793,   0.18396284,  -3.96578428,   1.99421765,\n",
    "          -4.96578427,  -2.64385619,  -5.96578425,  -1.83650127,\n",
    "           1.48129894,   0.03421572],\n",
    "        [ -0.38082178,  -1.35107444,   0.31961793, -31.21928095,\n",
    "          -4.96578427,   1.89530262,  -3.05889368,  -4.64385618,\n",
    "          -3.26534456,  -2.96578428]]),\n",
    " np.array([[ 1.73767068e-01,  8.67105730e-01, -1.79585928e+00,\n",
    "         -4.57989644e-01, -7.36965593e-01, -4.96578427e+00,\n",
    "          1.92599942e+00,  1.75872957e+00,  1.88283866e+00,\n",
    "         -4.96578427e+00, -3.96578428e+00, -2.57346686e+00,\n",
    "         -3.79585928e+00,  1.48954294e+00],\n",
    "        [ 2.29004027e-02, -2.96578428e+00, -3.79585928e+00,\n",
    "         -1.83650127e+00,  3.01002257e-01,  1.88908410e+00,\n",
    "         -3.50635266e+00, -4.96578427e+00, -3.12192809e+01,\n",
    "         -5.38082176e+00, -4.96578427e+00,  9.88412026e-01,\n",
    "          1.74157485e+00, -1.96578428e+00],\n",
    "        [ 5.01821266e-01,  6.70840336e-01,  1.76213617e+00,\n",
    "          3.82943870e-01,  1.53156789e-01, -4.64385618e+00,\n",
    "         -3.96578428e+00, -7.95859282e-01, -1.87832144e+00,\n",
    "          1.95307895e+00,  6.11644544e-01, -9.65784284e-01,\n",
    "         -3.96578428e+00, -9.43416471e-01],\n",
    "        [-1.18442457e+00, -1.13289427e+00, -2.01158797e+00,\n",
    "          7.55314904e-01,  7.86098352e-02, -2.15842936e+00,\n",
    "         -4.38082177e+00, -6.96578421e+00, -4.64385618e+00,\n",
    "         -3.79585928e+00,  1.24853484e+00,  4.17920008e-01,\n",
    "         -9.43416471e-01, -1.26534457e+00]])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "bb4ca271",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Runtime for RandomSequence:\n",
      "23.1 µs ± 2.67 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n",
      "\n",
      "GetThreshold:\n",
      "637 µs ± 3.14 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)\n",
      "\n",
      "Runtime for ComputeEnrichment:\n",
      "2.15 ms ± 280 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)\n",
      "\n",
      "Runtime for pwm_to_ppm:\n",
      "29.5 µs ± 496 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nRuntime for RandomSequence:\")\n",
    "test_freqs = [0.15, 0.34, 0.26, 0.25]\n",
    "%timeit pyxis.myutils.RandomSequence(50, test_freqs)\n",
    "print(\"\\nGetThreshold:\")\n",
    "numsim = 10000\n",
    "pval = 0.01\n",
    "for i in range(3):\n",
    "    null_scores = [pyxis.myutils.ScoreSeq(PWMList[i], pyxis.myutils.RandomSequence(PWMList[i].shape[1], test_freqs)) for j in range(numsim)]\n",
    "%timeit pyxis.myutils.GetThreshold(null_scores, pval)\n",
    "print(\"\\nRuntime for ComputeEnrichment:\")\n",
    "%timeit pyxis.myutils.ComputeEnrichment(50, 20, 50, 25)\n",
    "print(\"\\nRuntime for pwm_to_ppm:\")\n",
    "%timeit pyxis.myutils.pwm_to_ppm(bach_pwm, test_freqs, 1e-10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47ae2dbd",
   "metadata": {},
   "source": [
    "#### Summary:\n",
    "Using small toy data and example test files provided in `example-files` directory:\n",
    "\n",
    "| Function | Mean Runtime Per Loop ± Std. Dev. of 7 Runs (10000 loops each) |\n",
    "| :------- | :-----------------------------------------------------: |\n",
    "| ReadBED | 183 µs ± 9.18 µs |\n",
    "| ReadPWMS | 247 µs ± 6.29 µs |\n",
    "| WriteFastaSeq | 8.87 µs ± 139 ns |\n",
    "| ComputeNucFreqs | 23.9 µs ± 709 ns |\n",
    "| RandomBkSequence | 88.2 µs ± 1.15 µs |\n",
    "| ScoreSeq | 1.48 µs ± 14.5 ns |\n",
    "| ReverseComplement | 6.29 µs ± 46.8 ns |\n",
    "| FindMaxScore | 141 µs ± 2 µs |\n",
    "| RandomSequence | 23.1 µs ± 2.67 µs |\n",
    "| GetThreshold | 637 µs ± 3.14 µs |\n",
    "| ComputeEnrichment | 2.15 ms ± 280 µs |\n",
    "| pwm_to_ppm | 29.5 µs ± 496 ns |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0d6f83d",
   "metadata": {},
   "source": [
    "### Testing Pyxis Commands with Example Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "741cd437",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Basic Usage Command:\n",
      "10 loops, best of 5: 1.97 sec per loop\n",
      "\n",
      "Adding a Background:\n",
      "10 loops, best of 5: 2.02 sec per loop\n",
      "\n",
      "Including a specified p-value:\n",
      "10 loops, best of 5: 2.03 sec per loop\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "printf 'Basic Usage Command:\\n'\n",
    "python -m timeit -n 10 -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms > /dev/null\")'\n",
    "printf '\\nAdding a Background:\\n'\n",
    "python -m timeit -n 10 -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -b example-files/background.bed > /dev/null\")'\n",
    "printf '\\nIncluding a specified p-value:\\n'\n",
    "python -m timeit -n 10 -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -p 0.005 > /dev/null\")'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "09bbb376",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generating motif sequence logos:\n",
      "10 loops, best of 5: 2.82 sec per loop\n",
      "\n",
      "Including pseudovalue for generating sequence logos:\n",
      "10 loops, best of 5: 2.78 sec per loop\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "printf 'Including generating motif sequence logos:\\n'\n",
    "python -m timeit -n 10 -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -s > /dev/null\")'\n",
    "printf '\\nIncluding pseudovalue for generating sequence logos:\\n'\n",
    "python -m timeit -n 10 -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -s --pseudo 1e-6 > /dev/null\")'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfdf0e4d",
   "metadata": {},
   "source": [
    "#### Summary:\n",
    "\n",
    "| Command | Mean Runtime Per Loop (10 loops, best of 5) |\n",
    "| :------- | :-----------------------------------------------------: |\n",
    "| Basic Usage | 1.97 sec |\n",
    "| Basic Usage + Background Provided | 2.02 sec |\n",
    "| Basic Usage + P-value Provided | 2.03 sec |\n",
    "| Basic Usage + Seqlogo | 2.82 sec |\n",
    "| Basic Usage + Seqlogo + Pseudovalue | 2.78 sec |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ce00a67",
   "metadata": {},
   "source": [
    "### Testing Pyxis Commands with a Larger Dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47f92ca4",
   "metadata": {},
   "source": [
    "This larger, real-life dataset we will test on `pyxis` is provided at GEO accession [GSM5838045](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM5838045) (ISL1_WT_d6CP_Rep1) contains **peaks for the ISL1 transcription factor**. The ISL1 TFs were pulled down during ChIP through an anti-human-ISLET-1 antibody. This dataset is part of the [GSE195476 series](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE195476).\n",
    "\n",
    "The genome build used for this dataset was **hg19**. Due to its high memory usage, the build itself will not be included in the `pyxis` repo. If you are interested in running the below commands with the ISL1 dataset (as well as the commands in `PyxisAnalysis.ipynb`), please install `hg19.fa.gz` [here](https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/) and move it into the `benchmark` directory of `pyxis`. Be sure to also decompress the file before running `pyxis` on it– you can do this in Linux with `gzip -d`!\n",
    "\n",
    "The PWMs file serving as our motif database for this demonstration has position weight matrices selected based off Figure 3D from the source paper: [The multi-lineage transcription factor ISL1 controls cardiomyocyte cell fate through interaction with NKX2.5](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10679653/). Several other motifs picked at random from [HOCOMOCO](https://hocomoco11.autosome.org/) are also included for testing purposes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "7bf7d611",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/esquan/final_project/pyxis'"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd # Check that you are still in the outer pyxis directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7960e491",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "motif_name\tpval\tlog_pval\tnum_peak_motif\tpct_peak_motif\tnum_bg_motif\tpct_bg_motif\tenriched_status\n",
      "ISL1_MOUSE.H11MO.0.A\t3.8681244406275597e-295\t-294.4125\t4835\t0.781856403622251\t2879\t0.4655562742561449\tSig\n",
      "NKX25_MOUSE.H11MO.0.A\t6.859704709383046e-281\t-280.16369\t4801\t0.7763583441138422\t2890\t0.46733505821474774\tSig\n",
      "LHX3_MOUSE.H11MO.0.C\t1.0008831183450168e-267\t-266.99962\t4212\t0.6811125485122898\t2286\t0.36966364812419145\tSig\n",
      "NKX22_MOUSE.H11MO.0.A\t4.799427573329636e-218\t-217.31881\t5292\t0.8557567917205692\t3761\t0.6081824062095731\tSig\n",
      "BACH2_HUMAN.H11MO.0.A\t4.856501062484705e-217\t-216.31368\t4781\t0.773124191461837\t3115\t0.503719275549806\tSig\n",
      "KAISO_HUMAN.H11MO.1.A\t1.840030567469271e-86\t-85.73517\t2888\t0.4670116429495472\t1826\t0.29527813712807244\tSig\n",
      "MLX_HUMAN.H11MO.0.D\t8.11901614846211e-82\t-81.0905\t4744\t0.7671410090556274\t5535\t0.8950517464424321\tSig\n",
      "ALX3_HUMAN.H11MO.0.D\t3.2897559701277345e-16\t-15.48284\t6100\t0.9864165588615783\t5960\t0.963777490297542\tSig\n",
      "GATA4_MOUSE.H11MO.0.A\t0.0009722257333249527\t-3.01223\t6184\t1.0\t6173\t0.9982212160413971\tNon-sig\n",
      "ELK1_HUMAN.H11MO.0.B\t0.03549233860694681\t-1.44987\t6176\t0.9987063389391979\t6164\t0.9967658473479948\tNon-sig\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "# Checking results from pyxis for ISL1 peaks dataset\n",
    "pyxis analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed benchmark/hg19.fa benchmark/isl1.pwms > /dev/null\n",
    "cat pyxis_enrichments.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ac1c06fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Runtime for pyxis with ISL1 peaks dataset:\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/10] .... \n",
      "[2/10] .... \n",
      "[3/10] .... \n",
      "[4/10] .... \n",
      "[5/10] .... \n",
      "[6/10] .... \n",
      "[7/10] .... \n",
      "[8/10] .... \n",
      "[9/10] .... \n",
      "[10/10] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n",
      "1 loop, best of 5: 341 sec per loop\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "# Due to the ISL1 dataset's large size, the following command may take a while since it runs 5 loops of the command \n",
    "# with a best of 5 to calculate average runtime. Feel free to revisit this notebook in a few minutes.\n",
    "printf 'Runtime for pyxis with ISL1 peaks dataset:\\n'\n",
    "python -m timeit -s 'import os' 'os.system(\"pyxis analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed benchmark/hg19.fa benchmark/isl1.pwms\")'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9825523e",
   "metadata": {},
   "source": [
    "### Comparing Runtime with HOMER's findMotifsGenome.pl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b060c27d",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 loop, best of 5: 186 sec per loop\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing actual sizes of regions (-size given)\n",
      "\tFragment size set to given\n",
      "\tMotif length set at 11,\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground fragment size set to 21 (avg size of targets)\n",
      "\tBackground files for 21 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 20 bp\n",
      "\t\t10 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t2\n",
      "\t0.4\t4\t2\n",
      "\t0.45\t5\t2\n",
      "\t0.6\t7\t3\n",
      "\t0.7\t8\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t2\t26\t3.738\n",
      "\t4\t2\t91\t1.068\n",
      "\t5\t2\t163\t0.596\n",
      "\t7\t3\t142\t1.027\n",
      "\t8\t1\t64\t0.759\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t27.67%\t0.968\n",
      "\t\tC\t23.23%\t22.33%\t1.040\n",
      "\t\tG\t23.23%\t22.33%\t1.040\n",
      "\t\tT\t26.77%\t27.67%\t0.968\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.87%\t0.936\n",
      "\t\tCA\t7.63%\t6.31%\t1.210\n",
      "\t\tGA\t5.61%\t6.11%\t0.917\n",
      "\t\tTA\t5.74%\t7.27%\t0.790\n",
      "\t\tAC\t7.38%\t6.40%\t1.153\n",
      "\t\tCC\t4.15%\t4.89%\t0.850\n",
      "\t\tGC\t6.08%\t5.11%\t1.190\n",
      "\t\tTC\t5.61%\t6.11%\t0.917\n",
      "\t\tAG\t6.59%\t5.74%\t1.148\n",
      "\t\tCG\t5.31%\t5.50%\t0.965\n",
      "\t\tGG\t4.15%\t4.89%\t0.850\n",
      "\t\tTG\t7.63%\t6.31%\t1.210\n",
      "\t\tAT\t5.43%\t7.50%\t0.724\n",
      "\t\tCT\t6.59%\t5.74%\t1.148\n",
      "\t\tGT\t7.38%\t6.40%\t1.153\n",
      "\t\tTT\t7.37%\t7.87%\t0.936\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t26.77%\t26.87%\t0.997\n",
      "\t\tC\t23.23%\t23.13%\t1.004\n",
      "\t\tG\t23.23%\t23.13%\t1.004\n",
      "\t\tT\t26.77%\t26.87%\t0.997\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t7.37%\t7.36%\t1.000\n",
      "\t\tCA\t7.63%\t7.44%\t1.027\n",
      "\t\tGA\t5.61%\t5.71%\t0.982\n",
      "\t\tTA\t5.74%\t5.95%\t0.964\n",
      "\t\tAC\t7.38%\t7.15%\t1.031\n",
      "\t\tCC\t4.15%\t4.20%\t0.988\n",
      "\t\tGC\t6.08%\t6.24%\t0.974\n",
      "\t\tTC\t5.61%\t5.71%\t0.982\n",
      "\t\tAG\t6.59%\t6.51%\t1.012\n",
      "\t\tCG\t5.31%\t5.27%\t1.006\n",
      "\t\tGG\t4.15%\t4.20%\t0.988\n",
      "\t\tTG\t7.63%\t7.44%\t1.027\n",
      "\t\tAT\t5.43%\t5.79%\t0.938\n",
      "\t\tCT\t6.59%\t6.51%\t1.012\n",
      "\t\tGT\t7.38%\t7.15%\t1.031\n",
      "\t\tTT\t7.37%\t7.36%\t1.000\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t==================================================================================\n",
      "\tTotal number of Oligos: 638\n",
      "\tAutoadjustment for sequence coverage in background: 0.95x\n",
      "\n",
      "\tOligos: 638 of 5455 max\n",
      "\tTree  : 1728 of 27275 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 638 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t83.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t83.39% skipped, 16.61% checked (106 of 638), of those checked:\n",
      "\t\t0.00% not in target, 16.61% increased p-value, 11.13% high p-value\n",
      "\n",
      "\tScreening oligos 638 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 638), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AACGTTGATGC... (-3.061)\n",
      "\t\tRound 1: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 AACGTTGATGC T:1.0(10.00%),B:3.7(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 AACGTTGATGC T:1.0(10.00%),B:3.7(0.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GCCATTCTCTC... (-3.061)\n",
      "\t\tRound 1: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\tRound 2: -3.06 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.71%),P:1e-1\n",
      "\t\t=Final=: -2.71 GCCATTCTCTC T:1.0(10.00%),B:3.9(0.89%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTTTATCTAAA... (-2.389)\n",
      "\t\tRound 1: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\tRound 2: -2.39 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.19%),P:1e-1\n",
      "\t\t=Final=: -2.22 TTTTATCTAAA T:1.0(10.00%),B:5.0(1.15%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGATCAATACT... (-2.177)\n",
      "\t\tRound 1: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\tRound 2: -2.18 CGATCAATACT T:1.0(10.00%),B:6.5(1.42%),P:1e0\n",
      "\t\t=Final=: -2.05 CGATCAATACT T:1.0(10.00%),B:6.5(1.48%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTATGCAGGA... (-1.452)\n",
      "\t\tRound 1: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\tRound 2: -1.45 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.83%),P:1e0\n",
      "\t\t=Final=: -1.42 ATTATGCAGGA T:1.0(10.00%),B:12.6(2.87%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.452 > -7.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t496 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs11\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# HOMER output for the test files in pyxis\n",
    "python -m timeit -s 'import os' 'os.system(\"findMotifsGenome.pl example-files/peaks.bed example-files/ref.fa homer-results/example-data -size given -len 11 > /dev/null\")'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "f79be3aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 loop, best of 1: 3.12e+03 sec per loop\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed\n",
      "\tGenome = benchmark/hg19.fa\n",
      "\tOutput Directory = homer-results/isl1\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 6184\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 6184\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground files for 200 bp fragments found.\n",
      "\tCustom genome sequence file: benchmark/hg19.fa\n",
      "\n",
      "\tExtracting sequences from file: benchmark/hg19.fa\n",
      "\tLooking for peak sequences in a single file (benchmark/hg19.fa)\n",
      "\tExtracting 566 sequences from chr1\n",
      "\tExtracting 546 sequences from chr2\n",
      "\tExtracting 422 sequences from chr3\n",
      "\tExtracting 424 sequences from chr4\n",
      "\tExtracting 423 sequences from chr5\n",
      "\tExtracting 377 sequences from chr6\n",
      "\tExtracting 335 sequences from chr7\n",
      "\tExtracting 95 sequences from chrX\n",
      "\tExtracting 318 sequences from chr8\n",
      "\tExtracting 267 sequences from chr9\n",
      "\tExtracting 330 sequences from chr10\n",
      "\tExtracting 273 sequences from chr11\n",
      "\tExtracting 328 sequences from chr12\n",
      "\tExtracting 220 sequences from chr13\n",
      "\tExtracting 193 sequences from chr14\n",
      "\tExtracting 201 sequences from chr15\n",
      "\tExtracting 167 sequences from chr16\n",
      "\tExtracting 146 sequences from chr17\n",
      "\tExtracting 203 sequences from chr18\n",
      "\tExtracting 136 sequences from chr20\n",
      "\tExtracting 12 sequences from chrY\n",
      "\tExtracting 63 sequences from chr19\n",
      "\tExtracting 60 sequences from chr22\n",
      "\tExtracting 54 sequences from chr21\n",
      "\tExtracting 1 sequences from chrUn_gl000212\n",
      "\tExtracting 2 sequences from chr7_gl000195_random\n",
      "\tExtracting 1 sequences from chrUn_gl000224\n",
      "\tExtracting 1 sequences from chr17_gl000205_random\n",
      "\tExtracting 2 sequences from chr9_gl000199_random\n",
      "\tExtracting 10 sequences from chrUn_gl000220\n",
      "\tExtracting 1 sequences from chr19_gl000208_random\n",
      "\tExtracting 6 sequences from chrM\n",
      "\tExtracting 1 sequences from chrUn_gl000226\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 201 bp\n",
      "\t\t6184 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.2\t0\t2\n",
      "\t0.25\t1\t5\n",
      "\t0.3\t2\t14\n",
      "\t0.35\t3\t203\n",
      "\t0.4\t4\t1108\n",
      "\t0.45\t5\t2102\n",
      "\t0.5\t6\t1713\n",
      "\t0.6\t7\t951\n",
      "\t0.7\t8\t75\n",
      "\t0.8\t9\t9\n",
      "\t10\t10\t2\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t0\t2\t14\t1.012\n",
      "\t1\t5\t35\t1.012\n",
      "\t2\t14\t99\t1.002\n",
      "\t3\t203\t1438\t1.000\n",
      "\t4\t1108\t7851\t1.000\n",
      "\t5\t2102\t14893\t1.000\n",
      "\t6\t1713\t12137\t1.000\n",
      "\t7\t951\t6738\t1.000\n",
      "\t8\t75\t531\t1.001\n",
      "\t9\t9\t64\t0.996\n",
      "\t10\t2\t14\t1.012\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.70%\t0.999\n",
      "\t\tC\t22.34%\t22.30%\t1.002\n",
      "\t\tG\t22.34%\t22.30%\t1.002\n",
      "\t\tT\t27.66%\t27.70%\t0.999\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.53%\t0.966\n",
      "\t\tCA\t7.50%\t7.64%\t0.982\n",
      "\t\tGA\t6.42%\t6.23%\t1.029\n",
      "\t\tTA\t5.50%\t5.29%\t1.039\n",
      "\t\tAC\t5.10%\t5.16%\t0.988\n",
      "\t\tCC\t5.59%\t6.05%\t0.924\n",
      "\t\tGC\t5.24%\t4.86%\t1.078\n",
      "\t\tTC\t6.42%\t6.23%\t1.029\n",
      "\t\tAG\t8.28%\t7.50%\t1.103\n",
      "\t\tCG\t0.97%\t1.11%\t0.876\n",
      "\t\tGG\t5.59%\t6.05%\t0.924\n",
      "\t\tTG\t7.50%\t7.64%\t0.982\n",
      "\t\tAT\t6.04%\t6.50%\t0.929\n",
      "\t\tCT\t8.28%\t7.50%\t1.103\n",
      "\t\tGT\t5.10%\t5.16%\t0.988\n",
      "\t\tTT\t8.24%\t8.53%\t0.966\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/isl1/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.66%\t1.000\n",
      "\t\tC\t22.34%\t22.34%\t1.000\n",
      "\t\tG\t22.34%\t22.34%\t1.000\n",
      "\t\tT\t27.66%\t27.66%\t1.000\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.32%\t0.990\n",
      "\t\tCA\t7.50%\t7.57%\t0.992\n",
      "\t\tGA\t6.42%\t6.39%\t1.004\n",
      "\t\tTA\t5.50%\t5.38%\t1.023\n",
      "\t\tAC\t5.10%\t5.11%\t0.997\n",
      "\t\tCC\t5.59%\t5.77%\t0.967\n",
      "\t\tGC\t5.24%\t5.07%\t1.034\n",
      "\t\tTC\t6.42%\t6.39%\t1.004\n",
      "\t\tAG\t8.28%\t8.01%\t1.034\n",
      "\t\tCG\t0.97%\t0.99%\t0.982\n",
      "\t\tGG\t5.59%\t5.77%\t0.967\n",
      "\t\tTG\t7.50%\t7.57%\t0.992\n",
      "\t\tAT\t6.04%\t6.22%\t0.972\n",
      "\t\tCT\t8.28%\t8.01%\t1.034\n",
      "\t\tGT\t5.10%\t5.11%\t0.997\n",
      "\t\tTT\t8.24%\t8.32%\t0.990\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t\t1 of 1006 (1e-1166) Gata1(Zf)/K562-GATA1-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t2 of 1006 (1e-1165) Gata2(Zf)/K562-GATA2-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t3 of 1006 (1e-1159) Gata6(Zf)/HUG1N-GATA6-ChIP-Seq(GSE51936)/Homer\n",
      "\t\t4 of 1006 (1e-1140) Gata4(Zf)/Heart-Gata4-ChIP-Seq(GSE35151)/Homer\n",
      "\t\t5 of 1006 (1e-1039) GATA3(Zf)/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t6 of 1006 (1e-905) TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer\n",
      "\t\t7 of 1006 (1e-808) Isl1(Homeobox)/Neuron-Isl1-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t8 of 1006 (1e-766) At5g05790(MYBrelated)/col-At5g05790-DAP-Seq(GSE60143)/Homer\n",
      "\t\t9 of 1006 (1e-747) At3g11280(MYBrelated)/col-At3g11280-DAP-Seq(GSE60143)/Homer\n",
      "\t\t10 of 1006 (1e-617) At1g49010(MYBrelated)/col-At1g49010-DAP-Seq(GSE60143)/Homer\n",
      "\t\t11 of 1006 (1e-549) At5g58900(MYBrelated)/colamp-At5g58900-DAP-Seq(GSE60143)/Homer\n",
      "\t\t12 of 1006 (1e-548) At5g08520(MYBrelated)/colamp-At5g08520-DAP-Seq(GSE60143)/Homer\n",
      "\t\t13 of 1006 (1e-487) ELT-3(Gata)/cElegans-L1-ELT3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t14 of 1006 (1e-484) PQM-1(?)/cElegans-L3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t15 of 1006 (1e-407) Nkx2.5(Homeobox)/HL1-Nkx2.5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t16 of 1006 (1e-385) TEAD1(TEAD)/HepG2-TEAD1-ChIP-Seq(Encode)/Homer\n",
      "\t\t17 of 1006 (1e-379) TEAD3(TEA)/HepG2-TEAD3-ChIP-Seq(Encode)/Homer\n",
      "\t\t18 of 1006 (1e-372) TEAD4(TEA)/Tropoblast-Tead4-ChIP-Seq(GSE37350)/Homer\n",
      "\t\t19 of 1006 (1e-337) TEAD(TEA)/Fibroblast-PU.1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t20 of 1006 (1e-336) Nkx3.1(Homeobox)/LNCaP-Nkx3.1-ChIP-Seq(GSE28264)/Homer\n",
      "\t\t21 of 1006 (1e-327) Bapx1(Homeobox)/VertebralCol-Bapx1-ChIP-Seq(GSE36672)/Homer\n",
      "\t\t22 of 1006 (1e-319) Hand2(bHLH)/Mesoderm-Hand2-ChIP-Seq(GSE61475)/Homer\n",
      "\t\t23 of 1006 (1e-301) AT5G56840(MYBrelated)/colamp-AT5G56840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t24 of 1006 (1e-280) Unknown5/Drosophila-Promoters/Homer\n",
      "\t\t25 of 1006 (1e-268) Nkx2.1(Homeobox)/LungAC-Nkx2.1-ChIP-Seq(GSE43252)/Homer\n",
      "\t\t26 of 1006 (1e-256) At5g47390(MYBrelated)/col-At5g47390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t27 of 1006 (1e-231) TEAD2(TEA)/Py2T-Tead2-ChIP-Seq(GSE55709)/Homer\n",
      "\t\t28 of 1006 (1e-229) AT5G61620(MYBrelated)/colamp-AT5G61620-DAP-Seq(GSE60143)/Homer\n",
      "\t\t29 of 1006 (1e-207) Nkx6.1(Homeobox)/Islet-Nkx6.1-ChIP-Seq(GSE40975)/Homer\n",
      "\t\t30 of 1006 (1e-204) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
      "\t\t31 of 1006 (1e-199) Tbx20(T-box)/Heart-Tbx20-ChIP-Seq(GSE29636)/Homer\n",
      "\t\t32 of 1006 (1e-180) Tgif1(Homeobox)/mES-Tgif1-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t33 of 1006 (1e-152) En1(Homeobox)/SUM149-EN1-ChIP-Seq(GSE120957)/Homer\n",
      "\t\t34 of 1006 (1e-151) Lhx3(Homeobox)/Neuron-Lhx3-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t35 of 1006 (1e-150) Meis1(Homeobox)/MastCells-Meis1-ChIP-Seq(GSE48085)/Homer\n",
      "\t\t36 of 1006 (1e-138) Tgif2(Homeobox)/mES-Tgif2-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t37 of 1006 (1e-134) Nanog(Homeobox)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t38 of 1006 (1e-129) At1g19000(MYBrelated)/colamp-At1g19000-DAP-Seq(GSE60143)/Homer\n",
      "\t\t39 of 1006 (1e-116) DLX2(Homeobox)/BasalGanglia-Dlx2-ChIP-seq(GSE124936)/Homer\n",
      "\t\t40 of 1006 (1e-106) AT4G00250(GeBP)/col-AT4G00250-DAP-Seq(GSE60143)/Homer\n",
      "\t\t41 of 1006 (1e-104) ATHB33(ZFHD)/col-ATHB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t42 of 1006 (1e-103) Lhx2(Homeobox)/HFSC-Lhx2-ChIP-Seq(GSE48068)/Homer\n",
      "\t\t43 of 1006 (1e-97) LHX9(Homeobox)/Hct116-LHX9.V5-ChIP-Seq(GSE116822)/Homer\n",
      "\t\t44 of 1006 (1e-96) Eomes(T-box)/H9-Eomes-ChIP-Seq(GSE26097)/Homer\n",
      "\t\t45 of 1006 (1e-87) DLX5(Homeobox)/BasalGanglia-Dlx5-ChIP-seq(GSE124936)/Homer\n",
      "\t\t46 of 1006 (1e-84) Lhx1(Homeobox)/EmbryoCarcinoma-Lhx1-ChIP-Seq(GSE70957)/Homer\n",
      "\t\t47 of 1006 (1e-75) Barx1(Homeobox)/Stomach-Barx1.3xFlag-ChIP-Seq(GSE69483)/Homer\n",
      "\t\t48 of 1006 (1e-75) At1g74840(MYBrelated)/col100-At1g74840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t49 of 1006 (1e-71) DLX1(Homeobox)/BasalGanglia-Dlx1-ChIP-seq(GSE124936)/Homer\n",
      "\t\t50 of 1006 (1e-64) Dlx3(Homeobox)/Kerainocytes-Dlx3-ChIP-Seq(GSE89884)/Homer\n",
      "\t\t51 of 1006 (1e-64) AtGRF6(GRF)/col-AtGRF6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t52 of 1006 (1e-60) Otx2(Homeobox)/EpiLC-Otx2-ChIP-Seq(GSE56098)/Homer\n",
      "\t\t53 of 1006 (1e-60) GRF9(GRF)/colamp-GRF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t54 of 1006 (1e-57) bcd(Homeobox)/Embryo-Bcd-ChIP-Seq(GSE86966)/Homer\n",
      "\t\t55 of 1006 (1e-51) Pdx1(Homeobox)/Islet-Pdx1-ChIP-Seq(SRA008281)/Homer\n",
      "\t\t56 of 1006 (1e-50) CRX(Homeobox)/Retina-Crx-ChIP-Seq(GSE20012)/Homer\n",
      "\t\t57 of 1006 (1e-50) ATHB23(ZFHD)/col-ATHB23-DAP-Seq(GSE60143)/Homer\n",
      "\t\t58 of 1006 (1e-50) Foxo3(Forkhead)/U2OS-Foxo3-ChIP-Seq(E-MTAB-2701)/Homer\n",
      "\t\t59 of 1006 (1e-50) Foxa2(Forkhead)/Liver-Foxa2-ChIP-Seq(GSE25694)/Homer\n",
      "\t\t60 of 1006 (1e-49) ATHB34(ZFHD)/colamp-ATHB34-DAP-Seq(GSE60143)/Homer\n",
      "\t\t61 of 1006 (1e-46) Foxa3(Forkhead)/Liver-Foxa3-ChIP-Seq(GSE77670)/Homer\n",
      "\t\t62 of 1006 (1e-45) Hoxb4(Homeobox)/ES-Hoxb4-ChIP-Seq(GSE34014)/Homer\n",
      "\t\t63 of 1006 (1e-45) GATA:SCL(Zf,bHLH)/Ter119-SCL-ChIP-Seq(GSE18720)/Homer\n",
      "\t\t64 of 1006 (1e-44) Foxo1(Forkhead)/RAW-Foxo1-ChIP-Seq(Fan_et_al.)/Homer\n",
      "\t\t65 of 1006 (1e-44) MYB101(MYB)/colamp-MYB101-DAP-Seq(GSE60143)/Homer\n",
      "\t\t66 of 1006 (1e-41) ATHB24(ZFHD)/colamp-ATHB24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t67 of 1006 (1e-40) ATHB25(ZFHD)/colamp-ATHB25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t68 of 1006 (1e-39) FOXA1(Forkhead)/LNCAP-FOXA1-ChIP-Seq(GSE27824)/Homer\n",
      "\t\t69 of 1006 (1e-38) AT2G38300(G2like)/col-AT2G38300-DAP-Seq(GSE60143)/Homer\n",
      "\t\t70 of 1006 (1e-36) FOXA1(Forkhead)/MCF7-FOXA1-ChIP-Seq(GSE26831)/Homer\n",
      "\t\t71 of 1006 (1e-34) MafA(bZIP)/Islet-MafA-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t72 of 1006 (1e-34) Tbr1(T-box)/Cortex-Tbr1-ChIP-Seq(GSE71384)/Homer\n",
      "\t\t73 of 1006 (1e-34) FOXK2(Forkhead)/U2OS-FOXK2-ChIP-Seq(E-MTAB-2204)/Homer\n",
      "\t\t74 of 1006 (1e-34) AT2G40260(G2like)/colamp-AT2G40260-DAP-Seq(GSE60143)/Homer\n",
      "\t\t75 of 1006 (1e-33) FOXP1(Forkhead)/H9-FOXP1-ChIP-Seq(GSE31006)/Homer\n",
      "\t\t76 of 1006 (1e-33) BMYB(HTH)/Hela-BMYB-ChIP-Seq(GSE27030)/Homer\n",
      "\t\t77 of 1006 (1e-32) AT3G10030(Trihelix)/colamp-AT3G10030-DAP-Seq(GSE60143)/Homer\n",
      "\t\t78 of 1006 (1e-32) bZIP52(bZIP)/colamp-bZIP52-DAP-Seq(GSE60143)/Homer\n",
      "\t\t79 of 1006 (1e-32) MYB65(MYB)/colamp-MYB65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t80 of 1006 (1e-31) FoxL2(Forkhead)/Ovary-FoxL2-ChIP-Seq(GSE60858)/Homer\n",
      "\t\t81 of 1006 (1e-31) At5g04390(C2H2)/col200-At5g04390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t82 of 1006 (1e-30) FOXM1(Forkhead)/MCF7-FOXM1-ChIP-Seq(GSE72977)/Homer\n",
      "\t\t83 of 1006 (1e-29) Replumless(BLH)/Arabidopsis-RPL.GFP-ChIP-Seq(GSE78727)/Homer\n",
      "\t\t84 of 1006 (1e-29) FOXK1(Forkhead)/HEK293-FOXK1-ChIP-Seq(GSE51673)/Homer\n",
      "\t\t85 of 1006 (1e-28) Fox:Ebox(Forkhead,bHLH)/Panc1-Foxa2-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t86 of 1006 (1e-27) HOXA2(Homeobox)/mES-Hoxa2-ChIP-Seq(Donaldson_et_al.)/Homer\n",
      "\t\t87 of 1006 (1e-27) Mef2d(MADS)/Retina-Mef2d-ChIP-Seq(GSE61391)/Homer\n",
      "\t\t88 of 1006 (1e-27) STZ(C2H2)/colamp-STZ-DAP-Seq(GSE60143)/Homer\n",
      "\t\t89 of 1006 (1e-25) Ap4(bHLH)/AML-Tfap4-ChIP-Seq(GSE45738)/Homer\n",
      "\t\t90 of 1006 (1e-25) Pbx3(Homeobox)/GM12878-PBX3-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t91 of 1006 (1e-24) bZIP18(bZIP)/colamp-bZIP18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t92 of 1006 (1e-24) HOXA1(Homeobox)/mES-Hoxa1-ChIP-Seq(SRP084292)/Homer\n",
      "\t\t93 of 1006 (1e-24) MYB70(MYB)/col-MYB70-DAP-Seq(GSE60143)/Homer\n",
      "\t\t94 of 1006 (1e-24) Myf5(bHLH)/GM-Myf5-ChIP-Seq(GSE24852)/Homer\n",
      "\t\t95 of 1006 (1e-24) AMYB(HTH)/Testes-AMYB-ChIP-Seq(GSE44588)/Homer\n",
      "\t\t96 of 1006 (1e-23) PBX2(Homeobox)/K562-PBX2-ChIP-Seq(Encode)/Homer\n",
      "\t\t97 of 1006 (1e-23) HLH-1(bHLH)/cElegans-Embryo-HLH1-ChIP-Seq(modEncode)/Homer\n",
      "\t\t98 of 1006 (1e-22) MYB(HTH)/ERMYB-Myb-ChIPSeq(GSE22095)/Homer\n",
      "\t\t99 of 1006 (1e-22) Tbx6(T-box)/ESC-Tbx6-ChIP-Seq(GSE93524)/Homer\n",
      "\t\t100 of 1006 (1e-20) ZNF264(Zf)/HEK293-ZNF264.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t101 of 1006 (1e-19) E-box/Drosophila-Promoters/Homer\n",
      "\t\t102 of 1006 (1e-19) Pknox1(Homeobox)/ES-Prep1-ChIP-Seq(GSE63282)/Homer\n",
      "\t\t103 of 1006 (1e-19) Nur77(NR)/K562-NR4A1-ChIP-Seq(GSE31363)/Homer\n",
      "\t\t104 of 1006 (1e-19) Atoh1(bHLH)/Cerebellum-Atoh1-ChIP-Seq(GSE22111)/Homer\n",
      "\t\t105 of 1006 (1e-19) MyoG(bHLH)/C2C12-MyoG-ChIP-Seq(GSE36024)/Homer\n",
      "\t\t106 of 1006 (1e-18) AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t107 of 1006 (1e-18) MyoD(bHLH)/Myotube-MyoD-ChIP-Seq(GSE21614)/Homer\n",
      "\t\t108 of 1006 (1e-17) CTCF(Zf)/CD4+-CTCF-ChIP-Seq(Barski_et_al.)/Homer\n",
      "\t\t109 of 1006 (1e-17) Tcf21(bHLH)/ArterySmoothMuscle-Tcf21-ChIP-Seq(GSE61369)/Homer\n",
      "\t\t110 of 1006 (1e-17) CELF2(RRM)/JSL1-CELF2-CLIP-Seq(GSE71264)/Homer\n",
      "\t\t111 of 1006 (1e-17) TATA-Box(TBP)/Promoter/Homer\n",
      "\t\t112 of 1006 (1e-17) NLP7(RWPRK)/col-NLP7-DAP-Seq(GSE60143)/Homer\n",
      "\t\t113 of 1006 (1e-16) Foxh1(Forkhead)/hESC-FOXH1-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t114 of 1006 (1e-16) SCL(bHLH)/HPC7-Scl-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t115 of 1006 (1e-16) Hoxa9(Homeobox)/ChickenMSG-Hoxa9.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t116 of 1006 (1e-16) LIN-39(Homeobox)/cElegans.L3-LIN39-ChIP-Seq(modEncode)/Homer\n",
      "\t\t117 of 1006 (1e-16) Foxf1(Forkhead)/Lung-Foxf1-ChIP-Seq(GSE77951)/Homer\n",
      "\t\t118 of 1006 (1e-15) VND6(NAC)/col-VND6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t119 of 1006 (1e-15) AT1G69570(C2C2dof)/col-AT1G69570-DAP-Seq(GSE60143)/Homer\n",
      "\t\t120 of 1006 (1e-15) MYB77(MYB)/col-MYB77-DAP-Seq(GSE60143)/Homer\n",
      "\t\t121 of 1006 (1e-14) ZFX(Zf)/mES-Zfx-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t122 of 1006 (1e-13) FoxD3(forkhead)/ZebrafishEmbryo-Foxd3.biotin-ChIP-seq(GSE106676)/Homer\n",
      "\t\t123 of 1006 (1e-13) Smad2(MAD)/ES-SMAD2-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t124 of 1006 (1e-13) CArG(MADS)/PUER-Srf-ChIP-Seq(Sullivan_et_al.)/Homer\n",
      "\t\t125 of 1006 (1e-13) Smad4(MAD)/ESC-SMAD4-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t126 of 1006 (1e-13) WRKY28(WRKY)/col-WRKY28-DAP-Seq(GSE60143)/Homer\n",
      "\t\t127 of 1006 (1e-12) RAP211(AP2EREBP)/colamp-RAP211-DAP-Seq(GSE60143)/Homer\n",
      "\t\t128 of 1006 (1e-12) LEF1(HMG)/H1-LEF1-ChIP-Seq(GSE64758)/Homer\n",
      "\t\t129 of 1006 (1e-12) Tcf3(HMG)/mES-Tcf3-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t130 of 1006 (1e-12) NeuroD1(bHLH)/Islet-NeuroD1-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t131 of 1006 (1e-12) Tbx5(T-box)/HL1-Tbx5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t132 of 1006 (1e-12) Tcf4(HMG)/Hct116-Tcf4-ChIP-Seq(SRA012054)/Homer\n",
      "\t\t133 of 1006 (1e-12) Tcf12(bHLH)/GM12878-Tcf12-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t134 of 1006 (1e-11) CDF3(C2C2dof)/colamp-CDF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t135 of 1006 (1e-11) VIP1(bZIP)/col-VIP1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t136 of 1006 (1e-11) SND3(NAC)/col-SND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t137 of 1006 (1e-11) Brachyury(T-box)/Mesoendoderm-Brachyury-ChIP-exo(GSE54963)/Homer\n",
      "\t\t138 of 1006 (1e-11) NeuroG2(bHLH)/Fibroblast-NeuroG2-ChIP-Seq(GSE75910)/Homer\n",
      "\t\t139 of 1006 (1e-11) ZNF652/HepG2-ZNF652.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t140 of 1006 (1e-11) Bcl6(Zf)/Liver-Bcl6-ChIP-Seq(GSE31578)/Homer\n",
      "\t\t141 of 1006 (1e-11) MYB56(MYB)/colamp-MYB56-DAP-Seq(GSE60143)/Homer\n",
      "\t\t142 of 1006 (1e-10) PBX1(Homeobox)/MCF7-PBX1-ChIP-Seq(GSE28007)/Homer\n",
      "\t\t143 of 1006 (1e-10) VND2(NAC)/col-VND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t144 of 1006 (1e-10) MYB88(MYB)/col-MYB88-DAP-Seq(GSE60143)/Homer\n",
      "\t\t145 of 1006 (1e-10) ANAC071(NAC)/col-ANAC071-DAP-Seq(GSE60143)/Homer\n",
      "\t\t146 of 1006 (1e-10) KAN2(G2like)/colamp-KAN2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t147 of 1006 (1e-10) TR4(NR),DR1/Hela-TR4-ChIP-Seq(GSE24685)/Homer\n",
      "\t\t148 of 1006 (1e-10) ANAC045(NAC)/col-ANAC045-DAP-Seq(GSE60143)/Homer\n",
      "\t\t149 of 1006 (1e-10) MYB73(MYB)/col-MYB73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t150 of 1006 (1e-10) ERF13(AP2EREBP)/colamp-ERF13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t151 of 1006 (1e-10) ERF115(AP2EREBP)/colamp-ERF115-DAP-Seq(GSE60143)/Homer\n",
      "\t\t152 of 1006 (1e-9) NTM1(NAC)/col-NTM1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t153 of 1006 (1e-9) SMB(NAC)/colamp-SMB-DAP-Seq(GSE60143)/Homer\n",
      "\t\t154 of 1006 (1e-9) Rap210(AP2EREBP)/col-Rap210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t155 of 1006 (1e-9) Zic(Zf)/Cerebellum-ZIC1.2-ChIP-Seq(GSE60731)/Homer\n",
      "\t\t156 of 1006 (1e-9) WRKY42(WRKY)/colamp-WRKY42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t157 of 1006 (1e-9) GSC(Homeobox)/FrogEmbryos-GSC-ChIP-Seq(DRA000576)/Homer\n",
      "\t\t158 of 1006 (1e-9) HNF4a(NR),DR1/HepG2-HNF4a-ChIP-Seq(GSE25021)/Homer\n",
      "\t\t159 of 1006 (1e-9) Rbpj1(?)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t160 of 1006 (1e-8) PPARa(NR),DR1/Liver-Ppara-ChIP-Seq(GSE47954)/Homer\n",
      "\t\t161 of 1006 (1e-8) Unknown-ESC-element(?)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t162 of 1006 (1e-8) ERF105(AP2EREBP)/colamp-ERF105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t163 of 1006 (1e-8) VND4(NAC)/colamp-VND4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t164 of 1006 (1e-8) Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer\n",
      "\t\t165 of 1006 (1e-8) MYB81(MYB)/col-MYB81-DAP-Seq(GSE60143)/Homer\n",
      "\t\t166 of 1006 (1e-8) Fos(bZIP)/TSC-Fos-ChIP-Seq(GSE110950)/Homer\n",
      "\t\t167 of 1006 (1e-8) TCF4(bHLH)/SHSY5Y-TCF4-ChIP-Seq(GSE96915)/Homer\n",
      "\t\t168 of 1006 (1e-8) BHLHA15(bHLH)/NIH3T3-BHLHB8.HA-ChIP-Seq(GSE119782)/Homer\n",
      "\t\t169 of 1006 (1e-8) ERF104(AP2EREBP)/col-ERF104-DAP-Seq(GSE60143)/Homer\n",
      "\t\t170 of 1006 (1e-7) GATA3(Zf),DR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t171 of 1006 (1e-7) bZIP50(bZIP)/colamp-bZIP50-DAP-Seq(GSE60143)/Homer\n",
      "\t\t172 of 1006 (1e-7) Fra1(bZIP)/BT549-Fra1-ChIP-Seq(GSE46166)/Homer\n",
      "\t\t173 of 1006 (1e-7) NTM2(NAC)/col-NTM2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t174 of 1006 (1e-7) At3g60580(C2H2)/col-At3g60580-DAP-Seq(GSE60143)/Homer\n",
      "\t\t175 of 1006 (1e-7) WRKY29(WRKY)/colamp-WRKY29-DAP-Seq(GSE60143)/Homer\n",
      "\t\t176 of 1006 (1e-7) ANAC057(NAC)/colamp-ANAC057-DAP-Seq(GSE60143)/Homer\n",
      "\t\t177 of 1006 (1e-7) Knotted(Homeobox)/Corn-KN1-ChIP-Seq(GSE39161)/Homer\n",
      "\t\t178 of 1006 (1e-7) NF1-halfsite(CTF)/LNCaP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t179 of 1006 (1e-7) WRKY27(WRKY)/colamp-WRKY27-DAP-Seq(GSE60143)/Homer\n",
      "\t\t180 of 1006 (1e-7) GATA(Zf),IR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t181 of 1006 (1e-7) VND1(NAC)/col-VND1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t182 of 1006 (1e-7) AT2G33550(Trihelix)/colamp-AT2G33550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t183 of 1006 (1e-7) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
      "\t\t184 of 1006 (1e-7) VND3(NAC)/colamp-VND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t185 of 1006 (1e-7) WUS1(Homeobox)/colamp-WUS1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t186 of 1006 (1e-7) ANAC058(NAC)/col-ANAC058-DAP-Seq(GSE60143)/Homer\n",
      "\t\t187 of 1006 (1e-7) ANAC083(NAC)/col-ANAC083-DAP-Seq(GSE60143)/Homer\n",
      "\t\t188 of 1006 (1e-7) bHLH80(bHLH)/col-bHLH80-DAP-Seq(GSE60143)/Homer\n",
      "\t\t189 of 1006 (1e-7) bZIP69(bZIP)/col-bZIP69-DAP-Seq(GSE60143)/Homer\n",
      "\t\t190 of 1006 (1e-7) Pitx1(Homeobox)/Chicken-Pitx1-ChIP-Seq(GSE38910)/Homer\n",
      "\t\t191 of 1006 (1e-7) Zic3(Zf)/mES-Zic3-ChIP-Seq(GSE37889)/Homer\n",
      "\t\t192 of 1006 (1e-7) TCFL2(HMG)/K562-TCF7L2-ChIP-Seq(GSE29196)/Homer\n",
      "\t\t193 of 1006 (1e-7) TAGL1(MADS)/Tomato-TAGL1-ChIP-Seq(GSE116581)/Homer\n",
      "\t\t194 of 1006 (1e-6) ANAC005(NAC)/col-ANAC005-DAP-Seq(GSE60143)/Homer\n",
      "\t\t195 of 1006 (1e-6) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t196 of 1006 (1e-6) AT3G25990(Trihelix)/colamp-AT3G25990-DAP-Seq(GSE60143)/Homer\n",
      "\t\t197 of 1006 (1e-6) Tbet(T-box)/CD8-Tbet-ChIP-Seq(GSE33802)/Homer\n",
      "\t\t198 of 1006 (1e-6) Tbox:Smad(T-box,MAD)/ESCd5-Smad2_3-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t199 of 1006 (1e-6) bHLH122(bHLH)/col100-bHLH122-DAP-Seq(GSE60143)/Homer\n",
      "\t\t200 of 1006 (1e-6) Twist2(bHLH)/Myoblast-Twist2.Ty1-ChIP-Seq(GSE127998)/Homer\n",
      "\t\t201 of 1006 (1e-6) ANAC028(NAC)/col-ANAC028-DAP-Seq(GSE60143)/Homer\n",
      "\t\t202 of 1006 (1e-6) WRKY18(WRKY)/col-WRKY18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t203 of 1006 (1e-6) RXR(NR),DR1/3T3L1-RXR-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t204 of 1006 (1e-6) bZIP42(bZIP)/colamp-bZIP42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t205 of 1006 (1e-6) Six1(Homeobox)/Myoblast-Six1-ChIP-Chip(GSE20150)/Homer\n",
      "\t\t206 of 1006 (1e-6) ANAC070(NAC)/colamp-ANAC070-DAP-Seq(GSE60143)/Homer\n",
      "\t\t207 of 1006 (1e-6) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t208 of 1006 (1e-5) ERF73(AP2EREBP)/col-ERF73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t209 of 1006 (1e-5) BORIS(Zf)/K562-CTCFL-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t210 of 1006 (1e-5) ZNF7(Zf)/HepG2-ZNF7.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t211 of 1006 (1e-5) Tbx21(T-box)/GM12878-TBX21-ChIP-Seq(Encode)/Homer\n",
      "\t\t212 of 1006 (1e-5) CBF1(AP2EREBP)/colamp-CBF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t213 of 1006 (1e-5) Smad3(MAD)/NPC-Smad3-ChIP-Seq(GSE36673)/Homer\n",
      "\t\t214 of 1006 (1e-5) STAT4(Stat)/CD4-Stat4-ChIP-Seq(GSE22104)/Homer\n",
      "\t\t215 of 1006 (1e-5) Initiator/Drosophila-Promoters/Homer\n",
      "\t\t216 of 1006 (1e-5) CBF2(AP2EREBP)/colamp-CBF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t217 of 1006 (1e-5) ERF11(AP2EREBP)/col-ERF11-DAP-Seq(GSE60143)/Homer\n",
      "\t\t218 of 1006 (1e-5) NAC2(NAC)/colamp-NAC2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t219 of 1006 (1e-5) PHA-4(Forkhead)/cElegans-Embryos-PHA4-ChIP-Seq(modEncode)/Homer\n",
      "\t\t220 of 1006 (1e-5) CBF3(AP2EREBP)/colamp-CBF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t221 of 1006 (1e-5) PCF/Arabidopsis-Promoters/Homer\n",
      "\t\t222 of 1006 (1e-5) ANAC004(NAC)/colamp-ANAC004-DAP-Seq(GSE60143)/Homer\n",
      "\t\t223 of 1006 (1e-5) Zic2(Zf)/ESC-Zic2-ChIP-Seq(SRP197560)/Homer\n",
      "\t\t224 of 1006 (1e-5) Ascl1(bHLH)/NeuralTubes-Ascl1-ChIP-Seq(GSE55840)/Homer\n",
      "\t\t225 of 1006 (1e-5) Hoxc9(Homeobox)/Ainv15-Hoxc9-ChIP-Seq(GSE21812)/Homer\n",
      "\t\t226 of 1006 (1e-5) CBF4(AP2EREBP)/colamp-CBF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t227 of 1006 (1e-5) ANAC020(NAC)/col-ANAC020-DAP-Seq(GSE60143)/Homer\n",
      "\t\t228 of 1006 (1e-5) dof42(C2C2dof)/col-dof42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t229 of 1006 (1e-5) Rfx2(HTH)/LoVo-RFX2-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t230 of 1006 (1e-5) ANAC096(NAC)/colamp-ANAC096-DAP-Seq(GSE60143)/Homer\n",
      "\t\t231 of 1006 (1e-5) Fra2(bZIP)/Striatum-Fra2-ChIP-Seq(GSE43429)/Homer\n",
      "\t\t232 of 1006 (1e-5) EBF1(EBF)/Near-E2A-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t233 of 1006 (1e-5) Olig2(bHLH)/Neuron-Olig2-ChIP-Seq(GSE30882)/Homer\n",
      "\t\t234 of 1006 (1e-5) ERF15(AP2EREBP)/colamp-ERF15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t235 of 1006 (1e-5) CRF10(AP2EREBP)/col100-CRF10-DAP-Seq(GSE60143)/Homer\n",
      "\t\t236 of 1006 (1e-5) At4g16750(AP2EREBP)/col-At4g16750-DAP-Seq(GSE60143)/Homer\n",
      "\t\t237 of 1006 (1e-5) CUC1(NAC)/col-CUC1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t238 of 1006 (1e-5) WRKY40(WRKY)/colamp-WRKY40-DAP-Seq(GSE60143)/Homer\n",
      "\t\t239 of 1006 (1e-5) Stat3(Stat)/mES-Stat3-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t240 of 1006 (1e-4) RFX(HTH)/K562-RFX3-ChIP-Seq(SRA012198)/Homer\n",
      "\t\t241 of 1006 (1e-4) ERF2(AP2EREBP)/colamp-ERF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t242 of 1006 (1e-4) At1g19210(AP2EREBP)/colamp-At1g19210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t243 of 1006 (1e-4) Tlx?(NR)/NPC-H3K4me1-ChIP-Seq(GSE16256)/Homer\n",
      "\t\t244 of 1006 (1e-4) RAP26(AP2EREBP)/colamp-RAP26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t245 of 1006 (1e-4) WRKY65(WRKY)/colamp-WRKY65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t246 of 1006 (1e-4) ANAC062(NAC)/colamp-ANAC062-DAP-Seq(GSE60143)/Homer\n",
      "\t\t247 of 1006 (1e-4) Six2(Homeobox)/NephronProgenitor-Six2-ChIP-Seq(GSE39837)/Homer\n",
      "\t\t248 of 1006 (1e-4) AT5G23930(mTERF)/col-AT5G23930-DAP-Seq(GSE60143)/Homer\n",
      "\t\t249 of 1006 (1e-4) Mef2b(MADS)/HEK293-Mef2b.V5-ChIP-Seq(GSE67450)/Homer\n",
      "\t\t250 of 1006 (1e-4) WRKY33(WRKY)/col-WRKY33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t251 of 1006 (1e-4) SOC1(MADS)/Seedling-SOC1-ChIP-Seq(GSE45846)/Homer\n",
      "\t\t252 of 1006 (1e-4) AGL6(MADS)/col-AGL6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t253 of 1006 (1e-4) Prop1(Homeobox)/GHFT1-PROP1.biotin-ChIP-Seq(GSE77302)/Homer\n",
      "\t\t254 of 1006 (1e-4) JunB(bZIP)/DendriticCells-Junb-ChIP-Seq(GSE36099)/Homer\n",
      "\t\t255 of 1006 (1e-4) ANAC016(NAC)/col-ANAC016-DAP-Seq(GSE60143)/Homer\n",
      "\t\t256 of 1006 (1e-4) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
      "\t\t257 of 1006 (1e-4) AT1G12630(AP2EREBP)/colamp-AT1G12630-DAP-Seq(GSE60143)/Homer\n",
      "\t\t258 of 1006 (1e-4) REF6(Zf)/Arabidopsis-REF6-ChIP-Seq(GSE106942)/Homer\n",
      "\t\t259 of 1006 (1e-4) WRKY15(WRKY)/col-WRKY15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t260 of 1006 (1e-4) Sox9(HMG)/Limb-SOX9-ChIP-Seq(GSE73225)/Homer\n",
      "\t\t261 of 1006 (1e-4) ANAC092(NAC)/colamp-ANAC092-DAP-Seq(GSE60143)/Homer\n",
      "\t\t262 of 1006 (1e-4) bZIP44(bZIP)/colamp-bZIP44-DAP-Seq(GSE60143)/Homer\n",
      "\t\t263 of 1006 (1e-4) HOXA9(Homeobox)/HSC-Hoxa9-ChIP-Seq(GSE33509)/Homer\n",
      "\t\t264 of 1006 (1e-4) SVP(MADS)/col-SVP-DAP-Seq(GSE60143)/Homer\n",
      "\t\t265 of 1006 (1e-4) WRKY6(WRKY)/colamp-WRKY6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t266 of 1006 (1e-4) ABR1(AP2EREBP)/colamp-ABR1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t267 of 1006 (1e-4) WRKY25(WRKY)/colamp-WRKY25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t268 of 1006 (1e-4) NST1(NAC)/colamp-NST1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t269 of 1006 (1e-4) Stat3+il21(Stat)/CD4-Stat3-ChIP-Seq(GSE19198)/Homer\n",
      "\t\t270 of 1006 (1e-4) MYB105(MYB)/colamp-MYB105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t271 of 1006 (1e-4) ANAC050(NAC)/colamp-ANAC050-DAP-Seq(GSE60143)/Homer\n",
      "\t\t272 of 1006 (1e-4) PRDM14(Zf)/H1-PRDM14-ChIP-Seq(GSE22767)/Homer\n",
      "\t\t273 of 1006 (1e-4) TGA6(bZIP)/colamp-TGA6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t274 of 1006 (1e-4) AT5G47660(Trihelix)/colamp-AT5G47660-DAP-Seq(GSE60143)/Homer\n",
      "\t\t275 of 1006 (1e-4) Rfx6(HTH)/Min6b1-Rfx6.HA-ChIP-Seq(GSE62844)/Homer\n",
      "\t\t276 of 1006 (1e-4) PAX3:FKHR-fusion(Paired,Homeobox)/Rh4-PAX3:FKHR-ChIP-Seq(GSE19063)/Homer\n",
      "\t\t277 of 1006 (1e-4) PPARE(NR),DR1/3T3L1-Pparg-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t278 of 1006 (1e-4) AT1G76870(Trihelix)/col-AT1G76870-DAP-Seq(GSE60143)/Homer\n",
      "\t\t279 of 1006 (1e-4) AT1G77200(AP2EREBP)/colamp-AT1G77200-DAP-Seq(GSE60143)/Homer\n",
      "\t\t280 of 1006 (1e-4) STAT6(Stat)/Macrophage-Stat6-ChIP-Seq(GSE38377)/Homer\n",
      "\t\t281 of 1006 (1e-4) AGL16(MADS)/col-AGL16-DAP-Seq(GSE60143)/Homer\n",
      "\t\t282 of 1006 (1e-4) Atf3(bZIP)/GBM-ATF3-ChIP-Seq(GSE33912)/Homer\n",
      "\t\t283 of 1006 (1e-4) ZNF711(Zf)/SHSY5Y-ZNF711-ChIP-Seq(GSE20673)/Homer\n",
      "\t\t284 of 1006 (1e-4) SHN3(AP2EREBP)/col-SHN3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t285 of 1006 (1e-4) ERF1(AP2EREBP)/colamp-ERF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t286 of 1006 (1e-4) ERF3(AP2EREBP)/colamp-ERF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t287 of 1006 (1e-4) ERF8(AP2EREBP)/colamp-ERF8-DAP-Seq(GSE60143)/Homer\n",
      "\t\t288 of 1006 (1e-3) CEJ1(AP2EREBP)/col-CEJ1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t289 of 1006 (1e-3) ERF5(AP2EREBP)/colamp-ERF5-DAP-Seq(GSE60143)/Homer\n",
      "\t\t290 of 1006 (1e-3) DDF1(AP2EREBP)/col-DDF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t291 of 1006 (1e-3) TGA1(bZIP)/colamp-TGA1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t292 of 1006 (1e-3) SPL15(SBP)/colamp-SPL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t293 of 1006 (1e-3) WRKY14(WRKY)/colamp-WRKY14-DAP-Seq(GSE60143)/Homer\n",
      "\t\t294 of 1006 (1e-3) NFAT(RHD)/Jurkat-NFATC1-ChIP-Seq(Jolma_et_al.)/Homer\n",
      "\t\t295 of 1006 (1e-3) WRKY24(WRKY)/colamp-WRKY24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t296 of 1006 (1e-3) Bach1(bZIP)/K562-Bach1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t297 of 1006 (1e-3) At4g31060(AP2EREBP)/colamp-At4g31060-DAP-Seq(GSE60143)/Homer\n",
      "\t\t298 of 1006 (1e-3) RRTF1(AP2EREBP)/colamp-RRTF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t299 of 1006 (1e-3) ANAC053(NAC)/colamp-ANAC053-DAP-Seq(GSE60143)/Homer\n",
      "\t\t300 of 1006 (1e-3) STAT5(Stat)/mCD4+-Stat5-ChIP-Seq(GSE12346)/Homer\n",
      "\t\t301 of 1006 (1e-3) AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t302 of 1006 (1e-3) AT4G18450(AP2EREBP)/col-AT4G18450-DAP-Seq(GSE60143)/Homer\n",
      "\t\t303 of 1006 (1e-3) WRKY71(WRKY)/col-WRKY71-DAP-Seq(GSE60143)/Homer\n",
      "\t\t304 of 1006 (1e-3) DDF2(AP2EREBP)/col-DDF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t305 of 1006 (1e-3) Zac1(Zf)/Neuro2A-Plagl1-ChIP-Seq(GSE75942)/Homer\n",
      "\t\t306 of 1006 (1e-3) WRKY31(WRKY)/colamp-WRKY31-DAP-Seq(GSE60143)/Homer\n",
      "\t\t307 of 1006 (1e-3) DEAR2(AP2EREBP)/colamp-DEAR2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t308 of 1006 (1e-3) RAP212(AP2EREBP)/col-RAP212-DAP-Seq(GSE60143)/Homer\n",
      "\t\t309 of 1006 (1e-3) WRKY55(WRKY)/col-WRKY55-DAP-Seq(GSE60143)/Homer\n",
      "\t\t310 of 1006 (1e-3) WRKY75(WRKY)/col-WRKY75-DAP-Seq(GSE60143)/Homer\n",
      "\t\t311 of 1006 (1e-3) MYB57(MYB)/col-MYB57-DAP-Seq(GSE60143)/Homer\n",
      "\t\t312 of 1006 (1e-3) PUCHI(AP2EREBP)/colamp-PUCHI-DAP-Seq(GSE60143)/Homer\n",
      "\t\t313 of 1006 (1e-3) SND2(NAC)/colamp-SND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t314 of 1006 (1e-3) HAP3(CCAATHAP3)/col-HAP3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t315 of 1006 (1e-3) WRKY26(WRKY)/colamp-WRKY26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t316 of 1006 (1e-3) ERRg(NR)/Kidney-ESRRG-ChIP-Seq(GSE104905)/Homer\n",
      "\t\t317 of 1006 (1e-3) TGA2(bZIP)/colamp-TGA2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t318 of 1006 (1e-3) Six4(Homeobox)/MCF7-SIX4-ChIP-Seq(Encode)/Homer\n",
      "\t\t319 of 1006 (1e-3) AGL15(MADS)/col-AGL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t320 of 1006 (1e-3) WRKY3(WRKY)/col-WRKY3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t321 of 1006 (1e-3) Zfp281(Zf)/ES-Zfp281-ChIP-Seq(GSE81042)/Homer\n",
      "\t\t322 of 1006 (1e-3) ERF38(AP2EREBP)/col-ERF38-DAP-Seq(GSE60143)/Homer\n",
      "\t\t323 of 1006 (1e-3) WT1(Zf)/Kidney-WT1-ChIP-Seq(GSE90016)/Homer\n",
      "\t\t324 of 1006 (1e-2) Esrrb(NR)/mES-Esrrb-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t325 of 1006 (1e-2) GATA(Zf),IR3/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t326 of 1006 (1e-2) ERF4(AP2EREBP)/colamp-ERF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t327 of 1006 (1e-2) MYB33(MYB)/col-MYB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t328 of 1006 (1e-2) Oct6(POU,Homeobox)/NPC-Pou3f1-ChIP-Seq(GSE35496)/Homer\n",
      "\t\t329 of 1006 (1e-2) AT3G60490(AP2EREBP)/colamp-AT3G60490-DAP-Seq(GSE60143)/Homer\n",
      "\t\t330 of 1006 (1e-2) Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer\n",
      "\t\t331 of 1006 (1e-2) MYB116(MYB)/colamp-MYB116-DAP-Seq(GSE60143)/Homer\n",
      "\t\t332 of 1006 (1e-2) EBF2(EBF)/BrownAdipose-EBF2-ChIP-Seq(GSE97114)/Homer\n",
      "\t\t333 of 1006 (1e-2) WRKY47(WRKY)/colamp-WRKY47-DAP-Seq(GSE60143)/Homer\n",
      "\t\t334 of 1006 (1e-2) OBP3(C2C2dof)/col-OBP3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t335 of 1006 (1e-2) TINY(AP2EREBP)/col-TINY-DAP-Seq(GSE60143)/Homer\n",
      "\t\t336 of 1006 (1e-2) CRE(bZIP)/Promoter/Homer\n",
      "\t\t337 of 1006 (1e-2) WRKY46(WRKY)/colamp-WRKY46-DAP-Seq(GSE60143)/Homer\n",
      "\t\t338 of 1006 (1e-2) ERF9(AP2EREBP)/colamp-ERF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t339 of 1006 (1e-2) Sox10(HMG)/SciaticNerve-Sox3-ChIP-Seq(GSE35132)/Homer\n",
      "\t\t340 of 1006 (1e-2) CUC3(NAC)/col-CUC3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t341 of 1006 (1e-2) WRKY20(WRKY)/col-WRKY20-DAP-Seq(GSE60143)/Homer\n",
      "\t\t342 of 1006 (1e-2) AGL13(MADS)/col-AGL13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t343 of 1006 (1e-2) TGA4(bZIP)/colamp-TGA4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t344 of 1006 (1e-2) EGL-5(Homeobox)/cElegans-L3-EGL5-ChIP-Seq(modEncode)/Homer\n",
      "\t\t345 of 1006 (1e-2) At2g33710(AP2EREBP)/colamp-At2g33710-DAP-Seq(GSE60143)/Homer\n",
      "\t\t346 of 1006 (1e-2) NF-E2(bZIP)/K562-NFE2-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t347 of 1006 (1e-2) At5g18450(AP2EREBP)/col-At5g18450-DAP-Seq(GSE60143)/Homer\n",
      "\t\t348 of 1006 (1e-2) Nrf2(bZIP)/Lymphoblast-Nrf2-ChIP-Seq(GSE37589)/Homer\n",
      "\t\t349 of 1006 (1e-2) EBNA1(EBV-virus)/Raji-EBNA1-ChIP-Seq(GSE30709)/Homer\n",
      "\t\t350 of 1006 (1e-2) Phox2a(Homeobox)/Neuron-Phox2a-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t351 of 1006 (1e-2) ANAC075(NAC)/col-ANAC075-DAP-Seq(GSE60143)/Homer\n",
      "\t\t352 of 1006 (1e-2) NF1(CTF)/LNCAP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t353 of 1006 (1e-2) GEI-11(Myb?)/cElegans-L4-GEI11-ChIP-Seq(modEncode)/Homer\n",
      "\t\t354 of 1006 (1e-2) AT5G05550(Trihelix)/col-AT5G05550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t355 of 1006 (1e-2) Pax7(Paired,Homeobox)/Myoblast-Pax7-ChIP-Seq(GSE25064)/Homer\n",
      "\t\t356 of 1006 (1e-2) MYB93(MYB)/colamp-MYB93-DAP-Seq(GSE60143)/Homer\n",
      "\t\t357 of 1006 (1e-2) ANAC087(NAC)/col-ANAC087-DAP-Seq(GSE60143)/Homer\n",
      "\t\t358 of 1006 (1e-2) PRDM1(Zf)/Hela-PRDM1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t359 of 1006 (1e-2) RAP21(AP2EREBP)/colamp-RAP21-DAP-Seq(GSE60143)/Homer\n",
      "\t\t360 of 1006 (1e-2) ANAC013(NAC)/col-ANAC013-DAP-Seq(GSE60143)/Homer\n",
      "\t\t361 of 1006 (1e-2) WRKY43(WRKY)/colamp-WRKY43-DAP-Seq(GSE60143)/Homer\n",
      "\t\t362 of 1006 (1e-2) At2g44940(AP2EREBP)/colamp-At2g44940-DAP-Seq(GSE60143)/Homer\n",
      "\t\t363 of 1006 (1e-2) At1g22810(AP2EREBP)/colamp-At1g22810-DAP-Seq(GSE60143)/Homer\n",
      "\t\t364 of 1006 (1e-2) JGL(C2H2)/col-JGL-DAP-Seq(GSE60143)/Homer\n",
      "\t\t365 of 1006 (1e-2) WRKY22(WRKY)/colamp-WRKY22-DAP-Seq(GSE60143)/Homer\n",
      "\t\t366 of 1006 (1e-2) CEBP:AP1(bZIP)/ThioMac-CEBPb-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t367 of 1006 (1e-2) At5g22890(C2H2)/col-At5g22890-DAP-Seq(GSE60143)/Homer\n",
      "\t\t368 of 1006 (1e-2) REST-NRSF(Zf)/Jurkat-NRSF-ChIP-Seq/Homer\n",
      "\t\t369 of 1006 (1e-2) TGA10(bZIP)/colamp-TGA10-DAP-Seq(GSE60143)/Homer\n",
      "\t\t370 of 1006 (1e-2) WRKY30(WRKY)/colamp-WRKY30-DAP-Seq(GSE60143)/Homer\n",
      "\t\t371 of 1006 (1e-2) EBF(EBF)/proBcell-EBF-ChIP-Seq(GSE21978)/Homer\n",
      "\t\t372 of 1006 (1e-2) At1g36060(AP2EREBP)/colamp-At1g36060-DAP-Seq(GSE60143)/Homer\n",
      "\t\t373 of 1006 (1e-2) HNF1b(Homeobox)/PDAC-HNF1B-ChIP-Seq(GSE64557)/Homer\n",
      "\t\t374 of 1006 (1e-2) RKD2(RWPRK)/colamp-RKD2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t375 of 1006 (1e-2) ZNF317(Zf)/HEK293-ZNF317.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t376 of 1006 (1e-2) Pitx1:Ebox(Homeobox,bHLH)/Hindlimb-Pitx1-ChIP-Seq(GSE41591)/Homer\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 32886\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 32886 of 34497 max\n",
      "\tTree  : 67180 of 172485 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 32886 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.88% skipped, 98.12% checked (32267 of 32886), of those checked:\n",
      "\t\t1.88% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 32886 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.88% skipped, 98.12% checked (32267 of 32886), of those checked:\n",
      "\t\t0.00% not in target, 48.21% increased p-value, 48.60% high p-value\n",
      "\n",
      "\tScreening oligos 32886 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t60.91% skipped, 39.09% checked (12855 of 32886), of those checked:\n",
      "\t\t0.00% not in target, 22.23% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAA... (-1268.315)\n",
      "\t\tRound 1: -1962.29 GCACTTAA T:3149.0(39.91%),B:4353.8(9.78%),P:1e-852\n",
      "\t\tRound 2: -1962.29 GCACTTAA T:3149.0(39.91%),B:4353.8(9.78%),P:1e-852\n",
      "\t\t=Final=: -2232.79 GCACTTAA T:2596.0(41.98%),B:4078.9(9.65%),P:1e-969\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CTTATCTG... (-897.402)\n",
      "\t\tRound 1: -2358.49 STTATCWB T:3719.0(45.20%),B:4823.6(10.78%),P:1e-1024\n",
      "\t\tRound 2: -2447.15 CTTATCTB T:3286.0(41.22%),B:3709.7(8.40%),P:1e-1062\n",
      "\t\tRound 3: -2447.15 CTTATCTB T:3286.0(41.22%),B:3709.7(8.40%),P:1e-1062\n",
      "\t\t=Final=: -2677.30 CTTATCTB T:2646.0(42.79%),B:3505.8(8.29%),P:1e-1162\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: ACATTCCA... (-391.158)\n",
      "\t\tRound 1: -895.34 ACATTCCA T:1833.0(25.65%),B:3456.0(7.85%),P:1e-388\n",
      "\t\tRound 2: -895.34 ACATTCCA T:1833.0(25.65%),B:3456.0(7.85%),P:1e-388\n",
      "\t\t=Final=: -835.12 ACATTCCA T:1220.0(19.73%),B:2130.1(5.04%),P:1e-362\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: GCTAATGG... (-364.592)\n",
      "\t\tRound 1: -747.25 GCTAATGG T:1437.0(20.74%),B:2647.9(6.07%),P:1e-324\n",
      "\t\tRound 2: -747.25 GCTAATGG T:1437.0(20.74%),B:2647.9(6.07%),P:1e-324\n",
      "\t\t=Final=: -778.49 GCTAATGG T:1293.0(20.91%),B:2524.4(5.97%),P:1e-338\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GCTGTCAG... (-335.250)\n",
      "\t\tRound 1: -601.18 GCTGTCAG T:2871.0(37.14%),B:8603.3(18.41%),P:1e-261\n",
      "\t\tRound 2: -640.46 GCTGTCAG T:3935.0(47.08%),B:12684.0(25.92%),P:1e-278\n",
      "\t\tRound 3: -669.72 SCTGTCAG T:3782.0(45.75%),B:11820.6(24.39%),P:1e-290\n",
      "\t\tRound 4: -669.72 SCTGTCAG T:3782.0(45.75%),B:11820.6(24.39%),P:1e-290\n",
      "\t\t=Final=: -558.49 SCTGTCAG T:2901.0(46.91%),B:11431.2(27.04%),P:1e-242\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: CCCCCCCC... (-274.111)\n",
      "\t\tRound 1: -274.11 CCCCCCCC T:166.0(2.65%),B:86.6(0.20%),P:1e-119\n",
      "\t\tRound 2: -274.11 CCCCCCCC T:166.0(2.65%),B:86.6(0.20%),P:1e-119\n",
      "\t\t=Final=: -2.21 CCCCCCCC T:530.0(8.57%),B:3439.3(8.14%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: CTTTATCT... (-156.431)\n",
      "\t\tRound 1: -454.15 CMTTATCT T:2755.0(35.95%),B:9205.1(19.57%),P:1e-197\n",
      "\t\tRound 2: -479.19 KCTTATCT T:3846.0(46.31%),B:13792.2(27.84%),P:1e-208\n",
      "\t\tRound 3: -479.19 KCTTATCT T:3846.0(46.31%),B:13792.2(27.84%),P:1e-208\n",
      "\t\t=Final=: -508.53 KCTTATCT T:2815.0(45.52%),B:11270.7(26.66%),P:1e-220\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: CTGAGTGC... (-139.366)\n",
      "\t\tRound 1: -175.24 CTGAGTGC T:334.0(5.26%),B:662.9(1.55%),P:1e-76\n",
      "\t\tRound 2: -175.24 CTGAGTGC T:334.0(5.26%),B:662.9(1.55%),P:1e-76\n",
      "\t\t=Final=: -171.27 CTGAGTGC T:321.0(5.19%),B:653.6(1.55%),P:1e-74\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: TGATAATC... (-132.536)\n",
      "\t\tRound 1: -23.16 TGATAATC T:1222.0(17.93%),B:6842.1(14.94%),P:1e-10\n",
      "\t\tRound 2: -47.04 TGHTAATC T:831.0(12.58%),B:3973.3(8.97%),P:1e-20\n",
      "\t\tRound 3: -54.15 WSATTAAK T:986.0(14.74%),B:4725.6(10.58%),P:1e-23\n",
      "\t\tRound 4: -90.84 TGATTAAB T:1394.0(20.18%),B:6388.3(14.02%),P:1e-39\n",
      "\t\tRound 5: -90.84 TGATTAAB T:1394.0(20.18%),B:6388.3(14.02%),P:1e-39\n",
      "\t\t=Final=: -93.05 TGATTAAB T:1118.0(18.08%),B:5144.6(12.17%),P:1e-40\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: AATGGAAT... (-124.427)\n",
      "\t\tRound 1: -172.11 AMTCGAAT T:286.0(4.52%),B:510.4(1.20%),P:1e-74\n",
      "\t\tRound 2: -172.11 AMTCGAAT T:286.0(4.52%),B:510.4(1.20%),P:1e-74\n",
      "\t\t=Final=: -14.98 AMTCGAAT T:27.0(0.44%),B:58.8(0.14%),P:1e-6\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: TGTTGACA... (-114.684)\n",
      "\t\tRound 1: -204.78 TGTTGACA T:1184.0(17.43%),B:4094.2(9.23%),P:1e-88\n",
      "\t\tRound 2: -204.78 TGTTGACA T:1184.0(17.43%),B:4094.2(9.23%),P:1e-88\n",
      "\t\t=Final=: -190.69 TGTTGACA T:1051.0(17.00%),B:3869.3(9.15%),P:1e-82\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: AAATTCCT... (-81.630)\n",
      "\t\tRound 1: -114.94 AAATTCCT T:993.0(14.84%),B:3964.8(8.95%),P:1e-49\n",
      "\t\tRound 2: -144.44 VCATTCCA T:1306.0(19.04%),B:5236.6(11.65%),P:1e-62\n",
      "\t\tRound 3: -144.44 VCATTCCA T:1306.0(19.04%),B:5236.6(11.65%),P:1e-62\n",
      "\t\t=Final=: -99.07 VCATTCCA T:1067.0(17.25%),B:4781.3(11.31%),P:1e-43\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: AAAGCACT... (-75.533)\n",
      "\t\tRound 1: -149.51 AAAGCACT T:1615.0(22.99%),B:6753.9(14.76%),P:1e-64\n",
      "\t\tRound 2: -149.51 AAAGCACT T:1615.0(22.99%),B:6753.9(14.76%),P:1e-64\n",
      "\t\t=Final=: -139.09 AAAGCACT T:1248.0(20.18%),B:5371.2(12.71%),P:1e-60\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: ACCACTTA... (-71.225)\n",
      "\t\tRound 1: -152.93 ACCACTTA T:1166.0(17.19%),B:4450.6(9.99%),P:1e-66\n",
      "\t\tRound 2: -157.19 ACCACTTA T:852.0(12.87%),B:2907.1(6.65%),P:1e-68\n",
      "\t\tRound 3: -157.19 ACCACTTA T:852.0(12.87%),B:2907.1(6.65%),P:1e-68\n",
      "\t\t=Final=: -165.33 ACCACTTA T:804.0(13.00%),B:2799.7(6.62%),P:1e-71\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: GAGTGTTT... (-43.975)\n",
      "\t\tRound 1: -70.87 GAGTGCTT T:1083.0(16.07%),B:4992.1(11.14%),P:1e-30\n",
      "\t\tRound 2: -70.87 GAGTGCTT T:1083.0(16.07%),B:4992.1(11.14%),P:1e-30\n",
      "\t\t=Final=: -75.48 GAGTGCTT T:823.0(13.31%),B:3677.5(8.70%),P:1e-32\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: AGGGCCTG... (-36.430)\n",
      "\t\tRound 1: -58.33 AGGGCCTG T:1271.0(18.58%),B:6257.5(13.76%),P:1e-25\n",
      "\t\tRound 2: -58.33 AGGGCCTG T:1271.0(18.58%),B:6257.5(13.76%),P:1e-25\n",
      "\t\t=Final=: -53.55 AGGGCCTG T:1036.0(16.75%),B:5223.4(12.36%),P:1e-23\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: ATCGAATG... (-36.145)\n",
      "\t\tRound 1: -56.16 ATGGAATG T:81.0(1.30%),B:133.2(0.31%),P:1e-24\n",
      "\t\tRound 2: -56.16 ATGGAATG T:81.0(1.30%),B:133.2(0.31%),P:1e-24\n",
      "\t\t=Final=: -30.07 ATGGAATG T:13.0(0.21%),B:4.7(0.01%),P:1e-13\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: TGACAGCT... (-34.175)\n",
      "\t\tRound 1: -56.11 TGACAGCT T:1078.0(16.00%),B:5210.7(11.60%),P:1e-24\n",
      "\t\tRound 2: -56.11 TGACAGCT T:1078.0(16.00%),B:5210.7(11.60%),P:1e-24\n",
      "\t\t=Final=: -55.26 TGACAGCT T:965.0(15.60%),B:4772.1(11.29%),P:1e-23\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: ACATTCTT... (-33.730)\n",
      "\t\tRound 1: -75.50 ACATTCYC T:471.0(7.33%),B:1724.3(4.00%),P:1e-32\n",
      "\t\tRound 2: -75.50 ACATTCYC T:471.0(7.33%),B:1724.3(4.00%),P:1e-32\n",
      "\t\t=Final=: -73.42 ACATTCYC T:447.0(7.23%),B:1675.7(3.96%),P:1e-31\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: CGCCGCCG... (-31.322)\n",
      "\t\tRound 1: -62.96 CGCCGCCG T:334.0(5.26%),B:1162.2(2.71%),P:1e-27\n",
      "\t\tRound 2: -63.33 CGCCGCSG T:759.0(11.55%),B:3338.7(7.59%),P:1e-27\n",
      "\t\tRound 3: -66.69 SGCCGCSG T:279.0(4.41%),B:881.0(2.06%),P:1e-28\n",
      "\t\tRound 4: -69.03 SKCSGCSG T:357.0(5.61%),B:1224.5(2.85%),P:1e-29\n",
      "\t\tRound 5: -69.03 SKCSGCSG T:357.0(5.61%),B:1224.5(2.85%),P:1e-29\n",
      "\t\t=Final=: -30.52 SKCSGCSG T:230.0(3.72%),B:928.1(2.20%),P:1e-13\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: TGATAATG... (-29.219)\n",
      "\t\tRound 1: -15.60 TGATAATG T:183.0(2.92%),B:831.1(1.95%),P:1e-6\n",
      "\t\tRound 2: -25.87 AGATAGTG T:304.0(4.80%),B:1358.6(3.16%),P:1e-11\n",
      "\t\tRound 3: -38.21 AGATAGTG T:235.0(3.73%),B:870.8(2.04%),P:1e-16\n",
      "\t\tRound 4: -38.21 AGATAGTG T:235.0(3.73%),B:870.8(2.04%),P:1e-16\n",
      "\t\t=Final=: -37.21 AGATAGTG T:228.0(3.69%),B:859.6(2.03%),P:1e-16\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: CTAATCCT... (-25.528)\n",
      "\t\tRound 1: -56.89 CTAATCCT T:252.0(3.99%),B:816.9(1.91%),P:1e-24\n",
      "\t\tRound 2: -56.89 CTAATCCT T:252.0(3.99%),B:816.9(1.91%),P:1e-24\n",
      "\t\t=Final=: -56.99 CTAATCCT T:244.0(3.95%),B:798.4(1.89%),P:1e-24\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: GCTAAGTA... (-25.343)\n",
      "\t\tRound 1: -61.25 GCTAAGTA T:1023.0(15.25%),B:4810.0(10.75%),P:1e-26\n",
      "\t\tRound 2: -124.30 GSTAAGTG T:1548.0(22.15%),B:6728.7(14.71%),P:1e-53\n",
      "\t\tRound 3: -129.28 GSTAAGTG T:1292.0(18.86%),B:5335.0(11.86%),P:1e-56\n",
      "\t\tRound 4: -129.28 GSTAAGTG T:1292.0(18.86%),B:5335.0(11.86%),P:1e-56\n",
      "\t\t=Final=: -123.30 GSTAAGTG T:1176.0(19.02%),B:5130.2(12.14%),P:1e-53\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: TCCATTCG... (-24.586)\n",
      "\t\tRound 1: -29.95 TCCATTCG T:31.0(0.50%),B:37.7(0.09%),P:1e-13\n",
      "\t\tRound 2: -30.06 TCCGTTCG T:38.0(0.61%),B:56.8(0.13%),P:1e-13\n",
      "\t\tRound 3: -30.06 TCCGTTCG T:38.0(0.61%),B:56.8(0.13%),P:1e-13\n",
      "\t\t=Final=: -22.21 TCCGTTCG T:32.0(0.52%),B:56.8(0.13%),P:1e-9\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: AAAGCTTT... (-23.610)\n",
      "\t\tRound 1: -40.12 AAASCTTT T:333.0(5.24%),B:1352.8(3.15%),P:1e-17\n",
      "\t\tRound 2: -40.12 AAASCTTT T:333.0(5.24%),B:1352.8(3.15%),P:1e-17\n",
      "\t\t=Final=: -35.47 AAASCTTT T:314.0(5.08%),B:1324.1(3.13%),P:1e-15\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs8\n",
      "\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 480646\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 480646 of 551887 max\n",
      "\tTree  : 997144 of 2759435 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 480646 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.39% skipped, 67.61% checked (324956 of 480646), of those checked:\n",
      "\t\t32.39% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 480646 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.39% skipped, 67.61% checked (324956 of 480646), of those checked:\n",
      "\t\t0.00% not in target, 30.60% increased p-value, 32.01% high p-value\n",
      "\n",
      "\tScreening oligos 480646 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.01% skipped, 25.99% checked (124923 of 480646), of those checked:\n",
      "\t\t0.00% not in target, 7.24% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAACG... (-1110.859)\n",
      "\t\tRound 1: -2046.24 GCACTTAACR T:3527.0(43.47%),B:5081.7(11.33%),P:1e-888\n",
      "\t\tRound 2: -2123.16 GCACTTAGCN T:2857.0(37.00%),B:3361.0(7.64%),P:1e-922\n",
      "\t\tRound 3: -2123.16 GCACTTAGCN T:2857.0(37.00%),B:3361.0(7.64%),P:1e-922\n",
      "\t\t=Final=: -2434.08 GCACTTAGCN T:2833.0(45.81%),B:4554.2(10.77%),P:1e-1057\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CTTATCTCCG... (-594.210)\n",
      "\t\tRound 1: -1938.47 CTTATCTCCG T:2590.0(34.22%),B:3095.9(7.06%),P:1e-841\n",
      "\t\tRound 2: -2207.94 CTTATCTNNN T:2907.0(37.51%),B:3319.8(7.55%),P:1e-958\n",
      "\t\tRound 3: -2264.19 CTTATCAGNN T:3242.0(40.80%),B:3948.9(8.92%),P:1e-983\n",
      "\t\tRound 4: -2264.19 CTTATCAGNN T:3242.0(40.80%),B:3948.9(8.92%),P:1e-983\n",
      "\t\t=Final=: -2453.00 CTTATCAGNN T:2602.0(42.08%),B:3715.5(8.79%),P:1e-1065\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GGGGGGGGGG... (-444.185)\n",
      "\t\tRound 1: -444.18 GGGGGGGGGG T:149.0(2.38%),B:21.0(0.05%),P:1e-192\n",
      "\t\tRound 2: -444.18 GGGGGGGGGG T:149.0(2.38%),B:21.0(0.05%),P:1e-192\n",
      "\t\t=Final=: -3.78 GGGGGGGGGG T:95.0(1.54%),B:524.8(1.24%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CGAATGGAAT... (-314.891)\n",
      "\t\tRound 1: -475.16 CGAATGGAAT T:955.0(14.31%),B:1876.6(4.34%),P:1e-206\n",
      "\t\tRound 2: -525.32 SGAATGGAHT T:1607.0(22.89%),B:4032.0(9.09%),P:1e-228\n",
      "\t\tRound 3: -562.27 GGAATGBNWT T:2261.0(30.63%),B:6386.4(14.02%),P:1e-244\n",
      "\t\tRound 4: -636.74 GGAATGKAWT T:1908.0(26.55%),B:4654.0(10.42%),P:1e-276\n",
      "\t\tRound 5: -684.98 SKAATGBNHH T:2452.0(32.74%),B:6460.3(14.17%),P:1e-297\n",
      "\t\tRound 6: -710.81 GGAATGGAMT T:2086.0(28.63%),B:4978.1(11.11%),P:1e-308\n",
      "\t\tRound 7: -710.81 GGAATGGAMT T:2086.0(28.63%),B:4978.1(11.11%),P:1e-308\n",
      "\t\t=Final=: -517.12 GGAATGGAMT T:1503.0(24.30%),B:4277.8(10.12%),P:1e-224\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GTGTTGACAG... (-280.307)\n",
      "\t\tRound 1: -486.81 GTGTTGACAG T:1451.0(20.92%),B:3611.3(8.19%),P:1e-211\n",
      "\t\tRound 2: -526.93 GTGTTGACAG T:2040.0(28.10%),B:5706.7(12.63%),P:1e-228\n",
      "\t\tRound 3: -552.61 GTGTTGACAG T:1521.0(21.81%),B:3593.3(8.15%),P:1e-239\n",
      "\t\tRound 4: -586.34 BDGYTGACAG T:5248.0(57.20%),B:18757.3(35.83%),P:1e-254\n",
      "\t\tRound 5: -650.52 BDGCTGACAG T:3791.0(45.83%),B:12009.9(24.73%),P:1e-282\n",
      "\t\tRound 6: -700.65 BTGCTGACAG T:2939.0(37.83%),B:8245.2(17.72%),P:1e-304\n",
      "\t\tRound 7: -713.35 NDSCTGACAR T:3338.0(41.71%),B:9725.2(20.55%),P:1e-309\n",
      "\t\tRound 8: -713.35 NDSCTGACAR T:3338.0(41.71%),B:9725.2(20.55%),P:1e-309\n",
      "\t\t=Final=: -545.20 NDSCTGACAR T:2151.0(34.78%),B:7332.8(17.35%),P:1e-236\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: TAAGTGCTTT... (-138.531)\n",
      "\t\tRound 1: -221.15 TAAGTGCTTT T:4046.0(48.02%),B:18265.1(35.08%),P:1e-96\n",
      "\t\tRound 2: -244.26 TAAGTGCTTT T:2981.0(38.25%),B:12466.6(25.54%),P:1e-106\n",
      "\t\tRound 3: -244.26 TAAGTGCTTT T:2981.0(38.25%),B:12466.6(25.54%),P:1e-106\n",
      "\t\t=Final=: -278.28 TAAGTGCTTT T:2380.0(38.49%),B:10552.5(24.96%),P:1e-120\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: GAGTGATTAG... (-95.920)\n",
      "\t\tRound 1: -149.48 GAGTGATTAG T:3220.0(40.59%),B:15282.6(30.34%),P:1e-64\n",
      "\t\tRound 2: -179.97 GAGTGATTAG T:2087.0(28.65%),B:8793.3(18.78%),P:1e-78\n",
      "\t\tRound 3: -207.39 GAGTGATTAG T:2240.0(30.39%),B:9223.3(19.60%),P:1e-90\n",
      "\t\tRound 4: -252.99 GRGTGATTAV T:2454.0(32.76%),B:9739.1(20.58%),P:1e-109\n",
      "\t\tRound 5: -258.53 RRSTGWTTAV T:3619.0(44.30%),B:15459.1(30.63%),P:1e-112\n",
      "\t\tRound 6: -258.53 RRSTGWTTAV T:3619.0(44.30%),B:15459.1(30.63%),P:1e-112\n",
      "\t\t=Final=: -257.23 RRSTGWTTAV T:2694.0(43.56%),B:12680.9(30.00%),P:1e-111\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: CGCTAATGGG... (-84.822)\n",
      "\t\tRound 1: -230.18 CGCTAATGGG T:957.0(14.34%),B:2903.1(6.64%),P:1e-99\n",
      "\t\tRound 2: -282.24 NGCTAATGGR T:857.0(12.94%),B:2223.1(5.12%),P:1e-122\n",
      "\t\tRound 3: -282.24 NGCTAATGGR T:857.0(12.94%),B:2223.1(5.12%),P:1e-122\n",
      "\t\t=Final=: -280.07 NGCTAATGGR T:796.0(12.87%),B:2157.6(5.10%),P:1e-121\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: GATAAGAGGA... (-76.253)\n",
      "\t\tRound 1: -8.67 GATAAGAGGA T:20.0(0.32%),B:50.6(0.12%),P:1e-3\n",
      "\t\tRound 2: -46.91 GATAAGAGRV T:255.0(4.04%),B:899.0(2.10%),P:1e-20\n",
      "\t\tRound 3: -84.60 GATAAGAKRV T:593.0(9.14%),B:2247.8(5.18%),P:1e-36\n",
      "\t\tRound 4: -127.90 GATAAGRGDV T:791.0(12.01%),B:2846.5(6.51%),P:1e-55\n",
      "\t\tRound 5: -127.90 GATAAGRGDV T:791.0(12.01%),B:2846.5(6.51%),P:1e-55\n",
      "\t\t=Final=: -114.52 GATAAGRGDV T:579.0(9.36%),B:2032.0(4.81%),P:1e-49\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: GGAGATAAGA... (-71.158)\n",
      "\t\tRound 1: -23.51 GGAGATAAGA T:32.0(0.52%),B:51.4(0.12%),P:1e-10\n",
      "\t\tRound 2: -94.61 GGAGATAADA T:789.0(11.98%),B:3135.1(7.15%),P:1e-41\n",
      "\t\tRound 3: -170.85 DGAGATARNV T:1020.0(15.21%),B:3593.5(8.15%),P:1e-74\n",
      "\t\tRound 4: -188.82 TDAGATAAAV T:1066.0(15.84%),B:3670.8(8.32%),P:1e-82\n",
      "\t\tRound 5: -189.32 DNAGATARBV T:1126.0(16.65%),B:3945.0(8.91%),P:1e-82\n",
      "\t\tRound 6: -189.32 DNAGATARBV T:1126.0(16.65%),B:3945.0(8.91%),P:1e-82\n",
      "\t\t=Final=: -191.09 DNAGATARBV T:1043.0(16.87%),B:3825.1(9.05%),P:1e-82\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: CCGACATTCC... (-70.095)\n",
      "\t\tRound 1: -158.67 CCRACATTCC T:466.0(7.26%),B:1211.1(2.82%),P:1e-68\n",
      "\t\tRound 2: -182.39 CCKACATTCC T:369.0(5.79%),B:761.2(1.78%),P:1e-79\n",
      "\t\tRound 3: -222.18 THGAMATTCC T:795.0(12.06%),B:2261.2(5.21%),P:1e-96\n",
      "\t\tRound 4: -263.70 NHBRMATTCC T:731.0(11.15%),B:1810.0(4.19%),P:1e-114\n",
      "\t\tRound 5: -263.70 NHBRMATTCC T:731.0(11.15%),B:1810.0(4.19%),P:1e-114\n",
      "\t\t=Final=: -246.20 NHBRMATTCC T:661.0(10.69%),B:1725.0(4.08%),P:1e-106\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: CCATTATCTT... (-52.253)\n",
      "\t\tRound 1: -5.85 CGCTCAACCC T:9.0(0.15%),B:17.1(0.04%),P:1e-2\n",
      "\t\tRound 2: -13.95 CGCTCAACCC T:36.0(0.58%),B:94.5(0.22%),P:1e-6\n",
      "\t\tRound 3: -21.85 CGCTCAACYC T:42.0(0.68%),B:89.8(0.21%),P:1e-9\n",
      "\t\tRound 4: -24.00 CACTCAACMC T:74.0(1.19%),B:209.3(0.49%),P:1e-10\n",
      "\t\tRound 5: -29.59 CAYTCAACHC T:135.0(2.16%),B:452.7(1.06%),P:1e-12\n",
      "\t\tRound 6: -34.37 CAYTCAACAC T:315.0(4.97%),B:1319.7(3.07%),P:1e-14\n",
      "\t\tRound 7: -41.72 CAYTHAWCAC T:415.0(6.49%),B:1764.6(4.09%),P:1e-18\n",
      "\t\tRound 8: -46.61 CAYTCAWCAC T:526.0(8.15%),B:2299.5(5.29%),P:1e-20\n",
      "\t\tRound 9: -48.48 CWYTCAWCAC T:552.0(8.54%),B:2416.8(5.55%),P:1e-21\n",
      "\t\tRound 10: -48.48 CWYTCAWCAC T:552.0(8.54%),B:2416.8(5.55%),P:1e-21\n",
      "\t\t=Final=: -44.64 CWYTCAWCAC T:513.0(8.30%),B:2311.4(5.47%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: CTAAGAGCTG... (-47.689)\n",
      "\t\tRound 1: -22.67 CTAAGAGCTG T:308.0(4.86%),B:1425.7(3.31%),P:1e-9\n",
      "\t\tRound 2: -30.91 CTVAGAGCTG T:603.0(9.29%),B:2961.9(6.76%),P:1e-13\n",
      "\t\tRound 3: -42.18 CTRAGAGCWG T:894.0(13.46%),B:4420.2(9.93%),P:1e-18\n",
      "\t\tRound 4: -42.42 CHVADAGCWG T:996.0(14.88%),B:4997.9(11.15%),P:1e-18\n",
      "\t\tRound 5: -42.42 CHVADAGCWG T:996.0(14.88%),B:4997.9(11.15%),P:1e-18\n",
      "\t\t=Final=: -45.25 CHVADAGCWG T:916.0(14.81%),B:4645.7(10.99%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: GTTAAGTGTT... (-43.898)\n",
      "\t\tRound 1: -37.06 GTTAAGTGTT T:273.0(4.32%),B:1074.5(2.51%),P:1e-16\n",
      "\t\tRound 2: -41.37 GTTAAGTGTT T:112.0(1.79%),B:291.4(0.69%),P:1e-17\n",
      "\t\tRound 3: -50.66 GTTAAGTGNY T:480.0(7.47%),B:2003.5(4.63%),P:1e-21\n",
      "\t\tRound 4: -78.82 GTTAAGTGNT T:759.0(11.55%),B:3151.7(7.18%),P:1e-34\n",
      "\t\tRound 5: -110.36 GYTAAGTGRH T:1414.0(20.44%),B:6211.4(13.66%),P:1e-47\n",
      "\t\tRound 6: -110.36 GYTAAGTGRH T:1414.0(20.44%),B:6211.4(13.66%),P:1e-47\n",
      "\t\t=Final=: -129.95 GYTAAGTGRH T:1328.0(21.47%),B:5920.1(14.00%),P:1e-56\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: GGAGATAAGG... (-43.199)\n",
      "\t\tRound 1: -11.35 MTCSKTAMCG T:8.0(0.13%),B:6.9(0.01%),P:1e-4\n",
      "\t\tRound 2: -18.60 CTCSKTAACG T:27.0(0.44%),B:46.8(0.11%),P:1e-8\n",
      "\t\tRound 3: -19.48 MTCSKDAMCG T:24.0(0.39%),B:35.5(0.08%),P:1e-8\n",
      "\t\tRound 4: -19.48 MTCSKDAMCG T:24.0(0.39%),B:35.5(0.08%),P:1e-8\n",
      "\t\t=Final=: -18.95 MTCSKDAMCG T:23.0(0.37%),B:35.5(0.08%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: CTGATAACTG... (-43.199)\n",
      "\t\tRound 1: -15.28 CTGATAACTG T:120.0(1.92%),B:491.2(1.15%),P:1e-6\n",
      "\t\tRound 2: -30.00 CTGATAACTG T:323.0(5.09%),B:1409.8(3.28%),P:1e-13\n",
      "\t\tRound 3: -42.80 CTGRTRAMTG T:531.0(8.23%),B:2372.7(5.46%),P:1e-18\n",
      "\t\tRound 4: -51.66 CTGNTRAHTG T:514.0(7.98%),B:2174.1(5.01%),P:1e-22\n",
      "\t\tRound 5: -51.66 CTGNTRAHTG T:514.0(7.98%),B:2174.1(5.01%),P:1e-22\n",
      "\t\t=Final=: -52.45 CTGNTRAHTG T:441.0(7.13%),B:1830.3(4.33%),P:1e-22\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: CTAAGAGCTC... (-43.199)\n",
      "\t\tRound 1: -7.75 CTAAGAGCTC T:353.0(5.55%),B:2002.2(4.63%),P:1e-3\n",
      "\t\tRound 2: -18.48 CWAAGAGSTC T:373.0(5.85%),B:1867.6(4.32%),P:1e-8\n",
      "\t\tRound 3: -29.09 CWAAGAGSTC T:36.0(0.58%),B:52.9(0.12%),P:1e-12\n",
      "\t\tRound 4: -38.26 CTAAGAGGTC T:92.0(1.48%),B:223.2(0.53%),P:1e-16\n",
      "\t\tRound 5: -44.71 YWAACAGGYC T:139.0(2.22%),B:386.7(0.91%),P:1e-19\n",
      "\t\tRound 6: -44.71 YWAACAGGYC T:139.0(2.22%),B:386.7(0.91%),P:1e-19\n",
      "\t\t=Final=: -44.22 YWAACAGGYC T:136.0(2.20%),B:382.5(0.90%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: TTGGAAACGG... (-42.685)\n",
      "\t\tRound 1: -46.63 TTGGAAACGG T:16.0(0.26%),B:3.6(0.01%),P:1e-20\n",
      "\t\tRound 2: -46.63 TTGGAAACGG T:16.0(0.26%),B:3.6(0.01%),P:1e-20\n",
      "\t\t=Final=: -44.28 TTGGAAACGG T:16.0(0.26%),B:3.6(0.01%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: GCCGCCGCCG... (-40.122)\n",
      "\t\tRound 1: -65.51 GCCGCCGCCG T:448.0(6.99%),B:1697.2(3.93%),P:1e-28\n",
      "\t\tRound 2: -65.51 GCCGCCGCCG T:448.0(6.99%),B:1697.2(3.93%),P:1e-28\n",
      "\t\t=Final=: -24.49 GCCGCCGCCG T:130.0(2.10%),B:471.0(1.11%),P:1e-10\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: GATAAGAGGC... (-39.678)\n",
      "\t\tRound 1: -6.92 GATAAGAGGC T:148.0(2.37%),B:771.0(1.81%),P:1e-3\n",
      "\t\tRound 2: -23.13 GATAARAGGC T:158.0(2.52%),B:617.7(1.45%),P:1e-10\n",
      "\t\tRound 3: -33.93 GAYARAAGCC T:39.0(0.63%),B:52.4(0.12%),P:1e-14\n",
      "\t\tRound 4: -33.93 GAYARAAGCC T:39.0(0.63%),B:52.4(0.12%),P:1e-14\n",
      "\t\t=Final=: -35.59 GAYARAAGCC T:39.0(0.63%),B:51.9(0.12%),P:1e-15\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: GACTAACCTT... (-38.813)\n",
      "\t\tRound 1: -44.60 RSCTAACCTT T:57.0(0.92%),B:84.9(0.20%),P:1e-19\n",
      "\t\tRound 2: -54.09 GACTAACCTT T:52.0(0.84%),B:56.9(0.13%),P:1e-23\n",
      "\t\tRound 3: -69.13 GSCTAATCTT T:99.0(1.59%),B:161.3(0.38%),P:1e-30\n",
      "\t\tRound 4: -69.13 GSCTAATCTT T:99.0(1.59%),B:161.3(0.38%),P:1e-30\n",
      "\t\t=Final=: -70.59 GSCTAATCTT T:99.0(1.60%),B:160.9(0.38%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: TGATAAGCTC... (-38.789)\n",
      "\t\tRound 1: -6.20 TGATAAGCTC T:272.0(4.30%),B:1549.6(3.60%),P:1e-2\n",
      "\t\tRound 2: -18.19 TGATAARCTC T:758.0(11.54%),B:4169.1(9.39%),P:1e-7\n",
      "\t\tRound 3: -24.94 TGAHARRCTC T:144.0(2.30%),B:530.1(1.25%),P:1e-10\n",
      "\t\tRound 4: -26.56 TGAHANDCTC T:374.0(5.87%),B:1734.0(4.02%),P:1e-11\n",
      "\t\tRound 5: -34.07 TGACAVRCTC T:157.0(2.51%),B:528.2(1.24%),P:1e-14\n",
      "\t\tRound 6: -34.07 TGACAVRCTC T:157.0(2.51%),B:528.2(1.24%),P:1e-14\n",
      "\t\t=Final=: -34.53 TGACAVRCTC T:159.0(2.57%),B:540.1(1.28%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: AGATAACCTT... (-38.789)\n",
      "\t\tRound 1: -12.79 AGATAACCTT T:8.0(0.13%),B:5.9(0.01%),P:1e-5\n",
      "\t\tRound 2: -14.98 AGATAACCTT T:28.0(0.45%),B:59.2(0.14%),P:1e-6\n",
      "\t\tRound 3: -27.12 AGATAACCTT T:80.0(1.29%),B:220.2(0.52%),P:1e-11\n",
      "\t\tRound 4: -27.12 AGATAACCTT T:80.0(1.29%),B:220.2(0.52%),P:1e-11\n",
      "\t\t=Final=: -27.78 AGATAACCTT T:79.0(1.28%),B:216.9(0.51%),P:1e-12\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: TGTTGACTCA... (-36.203)\n",
      "\t\tRound 1: -67.62 TGTTGACWCA T:59.0(0.95%),B:56.7(0.13%),P:1e-29\n",
      "\t\tRound 2: -78.20 YGTTGACDYW T:343.0(5.40%),B:1100.1(2.57%),P:1e-33\n",
      "\t\tRound 3: -94.30 TGTTGACAYA T:330.0(5.20%),B:953.5(2.23%),P:1e-40\n",
      "\t\tRound 4: -97.16 TGTTTACWYW T:333.0(5.24%),B:952.3(2.23%),P:1e-42\n",
      "\t\tRound 5: -97.16 TGTTTACWYW T:333.0(5.24%),B:952.3(2.23%),P:1e-42\n",
      "\t\t=Final=: -83.17 TGTTTACWYW T:316.0(5.11%),B:978.7(2.32%),P:1e-36\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: CTAAGAGCTA... (-34.466)\n",
      "\t\tRound 1: -13.66 CTAAGAGCTA T:168.0(2.68%),B:772.0(1.81%),P:1e-5\n",
      "\t\tRound 2: -16.93 CTAAGAGCTA T:120.0(1.92%),B:476.0(1.12%),P:1e-7\n",
      "\t\tRound 3: -16.93 CTAAGAGCTA T:120.0(1.92%),B:476.0(1.12%),P:1e-7\n",
      "\t\t=Final=: -17.15 CTAAGAGCTA T:118.0(1.91%),B:470.5(1.11%),P:1e-7\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs10\n",
      "\n",
      "\n",
      "\t-blen automatically set to 2\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 3660198\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 3660198 of 8830118 max\n",
      "\tTree  : 8315948 of 44150590 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 3660198 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.14% skipped, 25.86% checked (946623 of 3660198), of those checked:\n",
      "\t\t74.14% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 3660198 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.14% skipped, 25.86% checked (946623 of 3660198), of those checked:\n",
      "\t\t0.00% not in target, 17.12% increased p-value, 4.12% high p-value\n",
      "\n",
      "\tScreening oligos 3660198 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t91.79% skipped, 8.21% checked (300477 of 3660198), of those checked:\n",
      "\t\t0.00% not in target, 2.59% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AAAGCACTTAAC... (-706.871)\n",
      "\t\tRound 1: -1526.12 AAAGCACTTAAC T:2617.0(34.51%),B:4017.5(9.06%),P:1e-662\n",
      "\t\tRound 2: -1905.42 AAAGCACTTAAC T:3020.0(38.64%),B:4167.3(9.39%),P:1e-827\n",
      "\t\tRound 3: -2133.79 AAAGCACTTARM T:3002.0(38.46%),B:3662.5(8.30%),P:1e-926\n",
      "\t\tRound 4: -2215.91 WARGCACTTARS T:2861.0(37.04%),B:3207.2(7.31%),P:1e-962\n",
      "\t\tRound 5: -2215.91 WARGCACTTARS T:2861.0(37.04%),B:3207.2(7.31%),P:1e-962\n",
      "\t\t=Final=: -2502.95 WARGCACTTARS T:2529.0(40.90%),B:3398.6(8.04%),P:1e-1087\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GGGGGGGGGGGG... (-585.494)\n",
      "\t\tRound 1: -585.49 GGGGGGGGGGGG T:139.0(2.22%),B:6.6(0.01%),P:1e-254\n",
      "\t\tRound 2: -585.49 GGGGGGGGGGGG T:139.0(2.22%),B:6.6(0.01%),P:1e-254\n",
      "\t\t=Final=: -5.50 GGGGGGGGGGGG T:144.0(2.33%),B:783.8(1.85%),P:1e-2\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: CCTTATCTTATC... (-339.666)\n",
      "\t\tRound 1: -678.51 CCTTATCTTATC T:2173.0(29.63%),B:5445.7(12.09%),P:1e-294\n",
      "\t\tRound 2: -905.10 CCTTATCTTATC T:1969.0(27.27%),B:3856.2(8.72%),P:1e-393\n",
      "\t\tRound 3: -1364.14 YCTTATCTYWWY T:1879.0(26.20%),B:2467.5(5.67%),P:1e-592\n",
      "\t\tRound 4: -1710.81 YCTTATCTBWNN T:2918.0(37.62%),B:4361.0(9.80%),P:1e-742\n",
      "\t\tRound 5: -1820.49 YCTTATCTBHNN T:2982.0(38.26%),B:4266.2(9.60%),P:1e-790\n",
      "\t\tRound 6: -1883.37 BCTTATCTBHNN T:1929.0(26.80%),B:1782.8(4.13%),P:1e-817\n",
      "\t\tRound 7: -2221.52 BCTTATCWNNNN T:2810.0(36.52%),B:3091.4(7.05%),P:1e-964\n",
      "\t\tRound 8: -2320.99 BCTTATCWNHNN T:2934.0(37.78%),B:3188.8(7.26%),P:1e-1007\n",
      "\t\tRound 9: -2320.99 BCTTATCWNHNN T:2934.0(37.78%),B:3188.8(7.26%),P:1e-1007\n",
      "\t\t=Final=: -2479.08 BCTTATCWNHNN T:2403.0(38.86%),B:3042.4(7.20%),P:1e-1076\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: TCGAATGGAATG... (-323.269)\n",
      "\t\tRound 1: -371.91 TSGAATGGAATG T:616.0(9.48%),B:1064.6(2.49%),P:1e-161\n",
      "\t\tRound 2: -383.04 TSGAATGGAATG T:619.0(9.53%),B:1047.8(2.45%),P:1e-166\n",
      "\t\tRound 3: -424.03 TBSAATGGAATG T:768.0(11.68%),B:1412.6(3.28%),P:1e-184\n",
      "\t\tRound 4: -451.65 TNGAATGGAATG T:768.0(11.68%),B:1341.6(3.12%),P:1e-196\n",
      "\t\tRound 5: -451.65 TNGAATGGAATG T:768.0(11.68%),B:1341.6(3.12%),P:1e-196\n",
      "\t\t=Final=: -219.09 TNGAATGGAATG T:490.0(7.92%),B:1135.0(2.68%),P:1e-95\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: AGGTGTTGACAG... (-189.106)\n",
      "\t\tRound 1: -331.36 AGGTGTTGACAG T:1670.0(23.67%),B:5367.1(11.92%),P:1e-143\n",
      "\t\tRound 2: -367.24 AGGTGTTGACAG T:760.0(11.57%),B:1549.9(3.60%),P:1e-159\n",
      "\t\tRound 3: -401.17 AGGTGTTGACAG T:918.0(13.80%),B:1989.3(4.60%),P:1e-174\n",
      "\t\tRound 4: -409.53 ARGTGTTGACAG T:1270.0(18.57%),B:3266.0(7.43%),P:1e-177\n",
      "\t\tRound 5: -409.53 ARGTGTTGACAG T:1270.0(18.57%),B:3266.0(7.43%),P:1e-177\n",
      "\t\t=Final=: -429.13 ARGTGTTGACAG T:1151.0(18.61%),B:3074.7(7.27%),P:1e-186\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: CTGGAATGTGTC... (-133.341)\n",
      "\t\tRound 1: -265.01 CTGGAATGTGTC T:705.0(10.78%),B:1705.3(3.95%),P:1e-115\n",
      "\t\tRound 2: -292.67 CTGGAATGTGTC T:812.0(12.31%),B:2004.6(4.63%),P:1e-127\n",
      "\t\tRound 3: -292.67 CTGGAATGTGTC T:812.0(12.31%),B:2004.6(4.63%),P:1e-127\n",
      "\t\t=Final=: -281.47 CTGGAATGTGTC T:748.0(12.10%),B:1947.0(4.61%),P:1e-122\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: GCTAATGGCATT... (-115.004)\n",
      "\t\tRound 1: -219.66 GCTAATGGCATT T:1554.0(22.22%),B:5717.3(12.65%),P:1e-95\n",
      "\t\tRound 2: -262.45 GCTAATGGCATT T:1393.0(20.17%),B:4623.9(10.36%),P:1e-113\n",
      "\t\tRound 3: -284.50 GCTAATGGCWTT T:1163.0(17.15%),B:3471.6(7.88%),P:1e-123\n",
      "\t\tRound 4: -301.64 GCTAATGGCWTT T:1199.0(17.63%),B:3527.6(8.00%),P:1e-130\n",
      "\t\tRound 5: -357.04 GCTAATGGCWTT T:1437.0(20.74%),B:4207.0(9.47%),P:1e-155\n",
      "\t\tRound 6: -357.04 GCTAATGGCWTT T:1437.0(20.74%),B:4207.0(9.47%),P:1e-155\n",
      "\t\t=Final=: -354.94 GCTAATGGCWTT T:1253.0(20.26%),B:3877.7(9.17%),P:1e-154\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: GAGATAAACCAG... (-93.879)\n",
      "\t\tRound 1: -129.88 GAGATAAACCAG T:133.0(2.13%),B:150.0(0.35%),P:1e-56\n",
      "\t\tRound 2: -191.57 GAGATAAACCAG T:479.0(7.45%),B:1128.3(2.63%),P:1e-83\n",
      "\t\tRound 3: -262.41 GAGADAAASCAG T:1088.0(16.13%),B:3277.6(7.46%),P:1e-113\n",
      "\t\tRound 4: -304.43 GABADAAAVCAG T:637.0(9.79%),B:1319.4(3.07%),P:1e-132\n",
      "\t\tRound 5: -322.89 GANADAAAVCAG T:631.0(9.70%),B:1240.8(2.89%),P:1e-140\n",
      "\t\tRound 6: -322.89 GANADAAAVCAG T:631.0(9.70%),B:1240.8(2.89%),P:1e-140\n",
      "\t\t=Final=: -321.25 GANADAAAVCAG T:598.0(9.67%),B:1223.7(2.89%),P:1e-139\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: GACAGCTGTCAG... (-71.399)\n",
      "\t\tRound 1: -118.93 GACAGCTGTCAG T:1392.0(20.16%),B:5979.0(13.19%),P:1e-51\n",
      "\t\tRound 2: -118.93 GACAGCTGTCAG T:1392.0(20.16%),B:5979.0(13.19%),P:1e-51\n",
      "\t\t=Final=: -85.58 GACAGCTGTCAG T:929.0(15.02%),B:4156.0(9.83%),P:1e-37\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: GCTTGTTTACTC... (-71.158)\n",
      "\t\tRound 1: -101.10 GCTTGTTTACTC T:25.0(0.40%),B:3.0(0.00%),P:1e-43\n",
      "\t\tRound 2: -101.10 GCTTGTTTACTC T:25.0(0.40%),B:3.0(0.00%),P:1e-43\n",
      "\t\t=Final=: -89.06 GCTTGTTTACTC T:25.0(0.40%),B:3.0(0.01%),P:1e-38\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: GGAATGCACTTT... (-69.576)\n",
      "\t\tRound 1: -95.35 GGAATGCACTTT T:105.0(1.68%),B:131.8(0.31%),P:1e-41\n",
      "\t\tRound 2: -118.34 GGAATGCACTTT T:269.0(4.26%),B:608.2(1.43%),P:1e-51\n",
      "\t\tRound 3: -121.28 GGAATGCACTTN T:118.0(1.89%),B:125.1(0.30%),P:1e-52\n",
      "\t\tRound 4: -132.72 GGAATGCACTTY T:130.0(2.08%),B:139.1(0.33%),P:1e-57\n",
      "\t\tRound 5: -132.72 GGAATGCACTTY T:130.0(2.08%),B:139.1(0.33%),P:1e-57\n",
      "\t\t=Final=: -132.13 GGAATGCACTTY T:126.0(2.04%),B:134.8(0.32%),P:1e-57\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: ACTTAACATTCC... (-63.555)\n",
      "\t\tRound 1: -98.77 ACTTAACATTCC T:172.0(2.74%),B:323.1(0.76%),P:1e-42\n",
      "\t\tRound 2: -109.00 ACTTAACATTCC T:180.0(2.87%),B:324.3(0.76%),P:1e-47\n",
      "\t\tRound 3: -129.98 ACTTAACATTCC T:382.0(5.99%),B:1000.7(2.34%),P:1e-56\n",
      "\t\tRound 4: -149.57 ACTTWACATTCC T:398.0(6.23%),B:982.2(2.30%),P:1e-64\n",
      "\t\tRound 5: -149.57 ACTTWACATTCC T:398.0(6.23%),B:982.2(2.30%),P:1e-64\n",
      "\t\t=Final=: -151.40 ACTTWACATTCC T:387.0(6.26%),B:971.6(2.30%),P:1e-65\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: AGTATCTTATCA... (-61.584)\n",
      "\t\tRound 1: -76.03 AGTATCTTATCA T:20.0(0.32%),B:3.0(0.00%),P:1e-33\n",
      "\t\tRound 2: -76.03 AGTATCTTATCA T:20.0(0.32%),B:3.0(0.00%),P:1e-33\n",
      "\t\t=Final=: -67.23 AGTATCTTATCA T:20.0(0.32%),B:3.0(0.01%),P:1e-29\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TGTCAATTAGCC... (-60.925)\n",
      "\t\tRound 1: -118.53 TGTCAATTAGCC T:100.0(1.60%),B:90.8(0.21%),P:1e-51\n",
      "\t\tRound 2: -139.55 TGTCAATTAGCC T:146.0(2.33%),B:169.9(0.40%),P:1e-60\n",
      "\t\tRound 3: -139.55 TGTCAATTAGCC T:146.0(2.33%),B:169.9(0.40%),P:1e-60\n",
      "\t\t=Final=: -130.39 TGTCAATTAGCC T:140.0(2.26%),B:171.8(0.41%),P:1e-56\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: GCACTTAACATT... (-56.886)\n",
      "\t\tRound 1: -62.70 NCACTTAACRNN T:1104.0(16.35%),B:5243.5(11.66%),P:1e-27\n",
      "\t\tRound 2: -88.32 KCACTTAACANN T:1363.0(19.78%),B:6256.9(13.76%),P:1e-38\n",
      "\t\tRound 3: -94.67 NCACTTAACANN T:1141.0(16.85%),B:4972.8(11.10%),P:1e-41\n",
      "\t\tRound 4: -111.11 NCACTTAACANN T:1646.0(23.37%),B:7441.5(16.14%),P:1e-48\n",
      "\t\tRound 5: -113.74 DCACTTAACAND T:1327.0(19.31%),B:5705.4(12.62%),P:1e-49\n",
      "\t\tRound 6: -113.74 DCACTTAACAND T:1327.0(19.31%),B:5705.4(12.62%),P:1e-49\n",
      "\t\t=Final=: -101.21 DCACTTAACAND T:1170.0(18.92%),B:5344.3(12.64%),P:1e-43\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: ACTAACCTTTGA... (-56.577)\n",
      "\t\tRound 1: -105.29 ACTAACCTTTGA T:41.0(0.66%),B:9.7(0.02%),P:1e-45\n",
      "\t\tRound 2: -105.29 ACTAACCTTTGA T:41.0(0.66%),B:9.7(0.02%),P:1e-45\n",
      "\t\t=Final=: -104.17 ACTAACCTTTGA T:41.0(0.66%),B:9.7(0.02%),P:1e-45\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: TTGAGTAAACAC... (-52.467)\n",
      "\t\tRound 1: -86.40 TTGAGTAAACAC T:52.0(0.84%),B:28.2(0.07%),P:1e-37\n",
      "\t\tRound 2: -86.40 TTGAGTAAACAC T:52.0(0.84%),B:28.2(0.07%),P:1e-37\n",
      "\t\t=Final=: -84.69 TTGAGTAAACAC T:51.0(0.82%),B:28.2(0.07%),P:1e-36\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: AGTGGCTGTCAG... (-48.134)\n",
      "\t\tRound 1: -60.92 AGTGGCTGTCAG T:187.0(2.98%),B:513.9(1.21%),P:1e-26\n",
      "\t\tRound 2: -60.92 AGTGGCTGTCAG T:187.0(2.98%),B:513.9(1.21%),P:1e-26\n",
      "\t\t=Final=: -62.93 AGTGGCTGTCAG T:209.0(3.38%),B:607.2(1.44%),P:1e-27\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: AATCGAATGGAA... (-47.689)\n",
      "\t\tRound 1: -10.76 NNTCGAATGGNN T:459.0(7.15%),B:2562.9(5.88%),P:1e-4\n",
      "\t\tRound 2: -16.26 NHTCGAATGGHH T:226.0(3.59%),B:1058.8(2.47%),P:1e-7\n",
      "\t\tRound 3: -22.13 HHTCGAATGGAA T:351.0(5.52%),B:1676.1(3.89%),P:1e-9\n",
      "\t\tRound 4: -22.13 HHTCGAATGGAA T:351.0(5.52%),B:1676.1(3.89%),P:1e-9\n",
      "\t\t=Final=: -21.49 HHTCGAATGGAA T:341.0(5.51%),B:1653.6(3.91%),P:1e-9\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: TACAGTATATAA... (-47.689)\n",
      "\t\tRound 1: -72.01 TACAGTATATAA T:26.0(0.42%),B:5.6(0.01%),P:1e-31\n",
      "\t\tRound 2: -73.76 TACWGTAYATAA T:28.0(0.45%),B:6.5(0.01%),P:1e-32\n",
      "\t\tRound 3: -101.10 TACWGTAYATAA T:25.0(0.40%),B:2.6(0.00%),P:1e-43\n",
      "\t\tRound 4: -101.10 TACWGTAYATAA T:25.0(0.40%),B:2.6(0.00%),P:1e-43\n",
      "\t\t=Final=: -10.98 TACWGTAYATAA T:4.0(0.06%),B:1.4(0.00%),P:1e-4\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: CTCTAATCACTC... (-44.781)\n",
      "\t\tRound 1: -80.57 MNCTAATCACNH T:409.0(6.40%),B:1388.9(3.23%),P:1e-34\n",
      "\t\tRound 2: -91.15 HNCTAATCACHH T:399.0(6.25%),B:1275.6(2.97%),P:1e-39\n",
      "\t\tRound 3: -114.13 YTCTAATCACWC T:485.0(7.54%),B:1520.7(3.53%),P:1e-49\n",
      "\t\tRound 4: -117.25 CTCTAATCACTC T:449.0(7.00%),B:1346.9(3.13%),P:1e-50\n",
      "\t\tRound 5: -117.25 CTCTAATCACTC T:449.0(7.00%),B:1346.9(3.13%),P:1e-50\n",
      "\t\t=Final=: -121.52 CTCTAATCACTC T:436.0(7.05%),B:1315.6(3.11%),P:1e-52\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: GGATAAGGATAA... (-43.199)\n",
      "\t\tRound 1: -56.89 GGATAAGGATAA T:16.0(0.26%),B:2.7(0.00%),P:1e-24\n",
      "\t\tRound 2: -56.89 GGATAAGGATAA T:16.0(0.26%),B:2.7(0.00%),P:1e-24\n",
      "\t\t=Final=: -11.18 GGATAAGGATAA T:5.0(0.08%),B:2.7(0.01%),P:1e-4\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: CTAAAAGCATTC... (-43.199)\n",
      "\t\tRound 1: -22.25 CTAAAAGCATTC T:61.0(0.98%),B:162.1(0.38%),P:1e-9\n",
      "\t\tRound 2: -26.33 CTAAAAGCATTC T:51.0(0.82%),B:108.9(0.26%),P:1e-11\n",
      "\t\tRound 3: -32.84 CTAAAAGCATTC T:68.0(1.09%),B:150.3(0.35%),P:1e-14\n",
      "\t\tRound 4: -40.88 CTAAAAGCATTC T:77.0(1.24%),B:158.9(0.37%),P:1e-17\n",
      "\t\tRound 5: -40.88 CTAAAAGCATTC T:77.0(1.24%),B:158.9(0.37%),P:1e-17\n",
      "\t\t=Final=: -42.45 CTAAAAGCATTC T:77.0(1.25%),B:156.7(0.37%),P:1e-18\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: TAAGGATAAGGA... (-43.199)\n",
      "\t\tRound 1: -9.66 NNAGGATAAGKN T:86.0(1.38%),B:373.8(0.88%),P:1e-4\n",
      "\t\tRound 2: -16.65 TNAGGATAAGGV T:65.0(1.05%),B:209.0(0.49%),P:1e-7\n",
      "\t\tRound 3: -28.77 THAGGATAAGRR T:62.0(1.00%),B:141.3(0.33%),P:1e-12\n",
      "\t\tRound 4: -28.77 THAGGATAAGRR T:62.0(1.00%),B:141.3(0.33%),P:1e-12\n",
      "\t\t=Final=: -32.93 THAGGATAAGRR T:66.0(1.07%),B:145.5(0.34%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: GAATGATTAAAG... (-43.199)\n",
      "\t\tRound 1: -76.03 GAATGATTAAAG T:20.0(0.32%),B:2.7(0.00%),P:1e-33\n",
      "\t\tRound 2: -76.03 GAATGATTAAAG T:20.0(0.32%),B:2.7(0.00%),P:1e-33\n",
      "\t\t=Final=: -67.23 GAATGATTAAAG T:20.0(0.32%),B:2.7(0.01%),P:1e-29\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs12\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 75 reduced to 18 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 18 (1e-1219) similar to GAT1/GAT1_RAPA/1-GZF3,2-GLN3(Harbison)/Yeast(0.988)\n",
      "\t\t2 of 18 (1e-1089) similar to ISL2/MA0914.1/Jaspar(0.939)\n",
      "\t\t3 of 18 (1e-401) similar to TEAD4/MA0809.2/Jaspar(0.988)\n",
      "\t\t4 of 18 (1e-361) similar to Isl1/MA1608.1/Jaspar(0.975)\n",
      "\t\t5 of 18 (1e-356) similar to Tgif1(Homeobox)/mES-Tgif1-ChIP-Seq(GSE55404)/Homer(0.936)\n",
      "\t\t6 of 18 (1e-181) similar to Hand2(bHLH)/Mesoderm-Hand2-ChIP-Seq(GSE61475)/Homer(0.919)\n",
      "\t\t7 of 18 (1e-138) similar to pqm-1/MA1703.1/Jaspar(0.711)\n",
      "\t\t8 of 18 (1e-77) similar to Foxa2(Forkhead)/Liver-Foxa2-ChIP-Seq(GSE25694)/Homer(0.960)\n",
      "\t\t9 of 18 (1e-63) similar to Dof2/MA0020.1/Jaspar(0.703)\n",
      "\t\t10 of 18 (1e-51) similar to CRZ1(MacIsaac)/Yeast(0.679)\n",
      "\t\t11 of 18 (1e-43) similar to PB0049.1_Nr2f2_1/Jaspar(0.679)\n",
      "\t\t12 of 18 (1e-38) similar to Dof2/MA0020.1/Jaspar(0.688)\n",
      "\t\t13 of 18 (1e-38) similar to Kr/dmmpmm(Down)/fly(0.783)\n",
      "\t\t14 of 18 (1e-17) similar to ERF069/MA0997.1/Jaspar(0.800)\n",
      "\t\t15 of 18 (1e-14) similar to PRDM4/MA1647.1/Jaspar(0.702)\n",
      "\t\t16 of 18 (1e-7) similar to BARHL1/MA0877.2/Jaspar(0.592)\n",
      "\t\t17 of 18 (1e-4) similar to ZBTB32/MA1580.1/Jaspar(0.698)\n",
      "\t\t18 of 18 (1e-1) similar to SeqBias: polyC-repeat(1.000)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "\n",
      "\tPosition file = analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed\n",
      "\tGenome = benchmark/hg19.fa\n",
      "\tOutput Directory = homer-results/isl1\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 6184\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 6184\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground files for 200 bp fragments found.\n",
      "\tCustom genome sequence file: benchmark/hg19.fa\n",
      "\n",
      "\tExtracting sequences from file: benchmark/hg19.fa\n",
      "\tLooking for peak sequences in a single file (benchmark/hg19.fa)\n",
      "\tExtracting 566 sequences from chr1\n",
      "\tExtracting 546 sequences from chr2\n",
      "\tExtracting 422 sequences from chr3\n",
      "\tExtracting 424 sequences from chr4\n",
      "\tExtracting 423 sequences from chr5\n",
      "\tExtracting 377 sequences from chr6\n",
      "\tExtracting 335 sequences from chr7\n",
      "\tExtracting 95 sequences from chrX\n",
      "\tExtracting 318 sequences from chr8\n",
      "\tExtracting 267 sequences from chr9\n",
      "\tExtracting 330 sequences from chr10\n",
      "\tExtracting 273 sequences from chr11\n",
      "\tExtracting 328 sequences from chr12\n",
      "\tExtracting 220 sequences from chr13\n",
      "\tExtracting 193 sequences from chr14\n",
      "\tExtracting 201 sequences from chr15\n",
      "\tExtracting 167 sequences from chr16\n",
      "\tExtracting 146 sequences from chr17\n",
      "\tExtracting 203 sequences from chr18\n",
      "\tExtracting 136 sequences from chr20\n",
      "\tExtracting 12 sequences from chrY\n",
      "\tExtracting 63 sequences from chr19\n",
      "\tExtracting 60 sequences from chr22\n",
      "\tExtracting 54 sequences from chr21\n",
      "\tExtracting 1 sequences from chrUn_gl000212\n",
      "\tExtracting 2 sequences from chr7_gl000195_random\n",
      "\tExtracting 1 sequences from chrUn_gl000224\n",
      "\tExtracting 1 sequences from chr17_gl000205_random\n",
      "\tExtracting 2 sequences from chr9_gl000199_random\n",
      "\tExtracting 10 sequences from chrUn_gl000220\n",
      "\tExtracting 1 sequences from chr19_gl000208_random\n",
      "\tExtracting 6 sequences from chrM\n",
      "\tExtracting 1 sequences from chrUn_gl000226\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 201 bp\n",
      "\t\t6184 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.2\t0\t2\n",
      "\t0.25\t1\t5\n",
      "\t0.3\t2\t14\n",
      "\t0.35\t3\t203\n",
      "\t0.4\t4\t1108\n",
      "\t0.45\t5\t2102\n",
      "\t0.5\t6\t1713\n",
      "\t0.6\t7\t951\n",
      "\t0.7\t8\t75\n",
      "\t0.8\t9\t9\n",
      "\t10\t10\t2\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t0\t2\t14\t1.012\n",
      "\t1\t5\t35\t1.012\n",
      "\t2\t14\t99\t1.002\n",
      "\t3\t203\t1438\t1.000\n",
      "\t4\t1108\t7851\t1.000\n",
      "\t5\t2102\t14893\t1.000\n",
      "\t6\t1713\t12137\t1.000\n",
      "\t7\t951\t6738\t1.000\n",
      "\t8\t75\t531\t1.001\n",
      "\t9\t9\t64\t0.996\n",
      "\t10\t2\t14\t1.012\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.69%\t0.999\n",
      "\t\tC\t22.34%\t22.31%\t1.002\n",
      "\t\tG\t22.34%\t22.31%\t1.002\n",
      "\t\tT\t27.66%\t27.69%\t0.999\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.52%\t0.967\n",
      "\t\tCA\t7.50%\t7.66%\t0.980\n",
      "\t\tGA\t6.42%\t6.22%\t1.032\n",
      "\t\tTA\t5.50%\t5.30%\t1.038\n",
      "\t\tAC\t5.10%\t5.17%\t0.987\n",
      "\t\tCC\t5.59%\t6.04%\t0.924\n",
      "\t\tGC\t5.24%\t4.88%\t1.074\n",
      "\t\tTC\t6.42%\t6.22%\t1.032\n",
      "\t\tAG\t8.28%\t7.49%\t1.105\n",
      "\t\tCG\t0.97%\t1.12%\t0.873\n",
      "\t\tGG\t5.59%\t6.04%\t0.924\n",
      "\t\tTG\t7.50%\t7.66%\t0.980\n",
      "\t\tAT\t6.04%\t6.52%\t0.928\n",
      "\t\tCT\t8.28%\t7.49%\t1.105\n",
      "\t\tGT\t5.10%\t5.17%\t0.987\n",
      "\t\tTT\t8.24%\t8.52%\t0.967\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/isl1/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.66%\t1.000\n",
      "\t\tC\t22.34%\t22.34%\t1.000\n",
      "\t\tG\t22.34%\t22.34%\t1.000\n",
      "\t\tT\t27.66%\t27.66%\t1.000\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.32%\t0.990\n",
      "\t\tCA\t7.50%\t7.57%\t0.991\n",
      "\t\tGA\t6.42%\t6.38%\t1.005\n",
      "\t\tTA\t5.50%\t5.39%\t1.022\n",
      "\t\tAC\t5.10%\t5.11%\t0.997\n",
      "\t\tCC\t5.59%\t5.77%\t0.968\n",
      "\t\tGC\t5.24%\t5.07%\t1.033\n",
      "\t\tTC\t6.42%\t6.38%\t1.005\n",
      "\t\tAG\t8.28%\t8.01%\t1.034\n",
      "\t\tCG\t0.97%\t0.99%\t0.983\n",
      "\t\tGG\t5.59%\t5.77%\t0.968\n",
      "\t\tTG\t7.50%\t7.57%\t0.991\n",
      "\t\tAT\t6.04%\t6.22%\t0.972\n",
      "\t\tCT\t8.28%\t8.01%\t1.034\n",
      "\t\tGT\t5.10%\t5.11%\t0.997\n",
      "\t\tTT\t8.24%\t8.32%\t0.990\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t\t1 of 1006 (1e-1143) Gata2(Zf)/K562-GATA2-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t2 of 1006 (1e-1137) Gata1(Zf)/K562-GATA1-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t3 of 1006 (1e-1131) Gata6(Zf)/HUG1N-GATA6-ChIP-Seq(GSE51936)/Homer\n",
      "\t\t4 of 1006 (1e-1106) Gata4(Zf)/Heart-Gata4-ChIP-Seq(GSE35151)/Homer\n",
      "\t\t5 of 1006 (1e-1033) GATA3(Zf)/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t6 of 1006 (1e-906) TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer\n",
      "\t\t7 of 1006 (1e-866) Isl1(Homeobox)/Neuron-Isl1-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t8 of 1006 (1e-747) At5g05790(MYBrelated)/col-At5g05790-DAP-Seq(GSE60143)/Homer\n",
      "\t\t9 of 1006 (1e-729) At3g11280(MYBrelated)/col-At3g11280-DAP-Seq(GSE60143)/Homer\n",
      "\t\t10 of 1006 (1e-622) At1g49010(MYBrelated)/col-At1g49010-DAP-Seq(GSE60143)/Homer\n",
      "\t\t11 of 1006 (1e-554) At5g58900(MYBrelated)/colamp-At5g58900-DAP-Seq(GSE60143)/Homer\n",
      "\t\t12 of 1006 (1e-540) At5g08520(MYBrelated)/colamp-At5g08520-DAP-Seq(GSE60143)/Homer\n",
      "\t\t13 of 1006 (1e-457) PQM-1(?)/cElegans-L3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t14 of 1006 (1e-456) ELT-3(Gata)/cElegans-L1-ELT3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t15 of 1006 (1e-387) Nkx2.5(Homeobox)/HL1-Nkx2.5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t16 of 1006 (1e-385) TEAD1(TEAD)/HepG2-TEAD1-ChIP-Seq(Encode)/Homer\n",
      "\t\t17 of 1006 (1e-375) TEAD3(TEA)/HepG2-TEAD3-ChIP-Seq(Encode)/Homer\n",
      "\t\t18 of 1006 (1e-366) TEAD4(TEA)/Tropoblast-Tead4-ChIP-Seq(GSE37350)/Homer\n",
      "\t\t19 of 1006 (1e-344) TEAD(TEA)/Fibroblast-PU.1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t20 of 1006 (1e-329) Nkx3.1(Homeobox)/LNCaP-Nkx3.1-ChIP-Seq(GSE28264)/Homer\n",
      "\t\t21 of 1006 (1e-328) Bapx1(Homeobox)/VertebralCol-Bapx1-ChIP-Seq(GSE36672)/Homer\n",
      "\t\t22 of 1006 (1e-306) Hand2(bHLH)/Mesoderm-Hand2-ChIP-Seq(GSE61475)/Homer\n",
      "\t\t23 of 1006 (1e-304) AT5G56840(MYBrelated)/colamp-AT5G56840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t24 of 1006 (1e-276) Unknown5/Drosophila-Promoters/Homer\n",
      "\t\t25 of 1006 (1e-256) At5g47390(MYBrelated)/col-At5g47390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t26 of 1006 (1e-255) Nkx2.1(Homeobox)/LungAC-Nkx2.1-ChIP-Seq(GSE43252)/Homer\n",
      "\t\t27 of 1006 (1e-228) AT5G61620(MYBrelated)/colamp-AT5G61620-DAP-Seq(GSE60143)/Homer\n",
      "\t\t28 of 1006 (1e-228) TEAD2(TEA)/Py2T-Tead2-ChIP-Seq(GSE55709)/Homer\n",
      "\t\t29 of 1006 (1e-211) Nkx6.1(Homeobox)/Islet-Nkx6.1-ChIP-Seq(GSE40975)/Homer\n",
      "\t\t30 of 1006 (1e-196) Tbx20(T-box)/Heart-Tbx20-ChIP-Seq(GSE29636)/Homer\n",
      "\t\t31 of 1006 (1e-190) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
      "\t\t32 of 1006 (1e-171) Tgif1(Homeobox)/mES-Tgif1-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t33 of 1006 (1e-157) En1(Homeobox)/SUM149-EN1-ChIP-Seq(GSE120957)/Homer\n",
      "\t\t34 of 1006 (1e-152) Lhx3(Homeobox)/Neuron-Lhx3-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t35 of 1006 (1e-144) Meis1(Homeobox)/MastCells-Meis1-ChIP-Seq(GSE48085)/Homer\n",
      "\t\t36 of 1006 (1e-139) Nanog(Homeobox)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t37 of 1006 (1e-137) Tgif2(Homeobox)/mES-Tgif2-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t38 of 1006 (1e-124) At1g19000(MYBrelated)/colamp-At1g19000-DAP-Seq(GSE60143)/Homer\n",
      "\t\t39 of 1006 (1e-117) DLX2(Homeobox)/BasalGanglia-Dlx2-ChIP-seq(GSE124936)/Homer\n",
      "\t\t40 of 1006 (1e-112) ATHB33(ZFHD)/col-ATHB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t41 of 1006 (1e-102) AT4G00250(GeBP)/col-AT4G00250-DAP-Seq(GSE60143)/Homer\n",
      "\t\t42 of 1006 (1e-101) LHX9(Homeobox)/Hct116-LHX9.V5-ChIP-Seq(GSE116822)/Homer\n",
      "\t\t43 of 1006 (1e-101) Lhx2(Homeobox)/HFSC-Lhx2-ChIP-Seq(GSE48068)/Homer\n",
      "\t\t44 of 1006 (1e-97) Eomes(T-box)/H9-Eomes-ChIP-Seq(GSE26097)/Homer\n",
      "\t\t45 of 1006 (1e-86) Lhx1(Homeobox)/EmbryoCarcinoma-Lhx1-ChIP-Seq(GSE70957)/Homer\n",
      "\t\t46 of 1006 (1e-84) DLX5(Homeobox)/BasalGanglia-Dlx5-ChIP-seq(GSE124936)/Homer\n",
      "\t\t47 of 1006 (1e-77) At1g74840(MYBrelated)/col100-At1g74840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t48 of 1006 (1e-73) Barx1(Homeobox)/Stomach-Barx1.3xFlag-ChIP-Seq(GSE69483)/Homer\n",
      "\t\t49 of 1006 (1e-72) DLX1(Homeobox)/BasalGanglia-Dlx1-ChIP-seq(GSE124936)/Homer\n",
      "\t\t50 of 1006 (1e-66) AtGRF6(GRF)/col-AtGRF6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t51 of 1006 (1e-65) GRF9(GRF)/colamp-GRF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t52 of 1006 (1e-63) Dlx3(Homeobox)/Kerainocytes-Dlx3-ChIP-Seq(GSE89884)/Homer\n",
      "\t\t53 of 1006 (1e-60) Foxo3(Forkhead)/U2OS-Foxo3-ChIP-Seq(E-MTAB-2701)/Homer\n",
      "\t\t54 of 1006 (1e-59) Otx2(Homeobox)/EpiLC-Otx2-ChIP-Seq(GSE56098)/Homer\n",
      "\t\t55 of 1006 (1e-56) Pdx1(Homeobox)/Islet-Pdx1-ChIP-Seq(SRA008281)/Homer\n",
      "\t\t56 of 1006 (1e-54) ATHB34(ZFHD)/colamp-ATHB34-DAP-Seq(GSE60143)/Homer\n",
      "\t\t57 of 1006 (1e-54) ATHB23(ZFHD)/col-ATHB23-DAP-Seq(GSE60143)/Homer\n",
      "\t\t58 of 1006 (1e-53) Foxa2(Forkhead)/Liver-Foxa2-ChIP-Seq(GSE25694)/Homer\n",
      "\t\t59 of 1006 (1e-52) GATA:SCL(Zf,bHLH)/Ter119-SCL-ChIP-Seq(GSE18720)/Homer\n",
      "\t\t60 of 1006 (1e-50) bcd(Homeobox)/Embryo-Bcd-ChIP-Seq(GSE86966)/Homer\n",
      "\t\t61 of 1006 (1e-48) ATHB25(ZFHD)/colamp-ATHB25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t62 of 1006 (1e-48) Foxa3(Forkhead)/Liver-Foxa3-ChIP-Seq(GSE77670)/Homer\n",
      "\t\t63 of 1006 (1e-48) ATHB24(ZFHD)/colamp-ATHB24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t64 of 1006 (1e-45) Hoxb4(Homeobox)/ES-Hoxb4-ChIP-Seq(GSE34014)/Homer\n",
      "\t\t65 of 1006 (1e-43) CRX(Homeobox)/Retina-Crx-ChIP-Seq(GSE20012)/Homer\n",
      "\t\t66 of 1006 (1e-42) MYB101(MYB)/colamp-MYB101-DAP-Seq(GSE60143)/Homer\n",
      "\t\t67 of 1006 (1e-41) Tbr1(T-box)/Cortex-Tbr1-ChIP-Seq(GSE71384)/Homer\n",
      "\t\t68 of 1006 (1e-41) Foxo1(Forkhead)/RAW-Foxo1-ChIP-Seq(Fan_et_al.)/Homer\n",
      "\t\t69 of 1006 (1e-40) FOXP1(Forkhead)/H9-FOXP1-ChIP-Seq(GSE31006)/Homer\n",
      "\t\t70 of 1006 (1e-38) FOXA1(Forkhead)/LNCAP-FOXA1-ChIP-Seq(GSE27824)/Homer\n",
      "\t\t71 of 1006 (1e-38) AT2G38300(G2like)/col-AT2G38300-DAP-Seq(GSE60143)/Homer\n",
      "\t\t72 of 1006 (1e-37) AT3G10030(Trihelix)/colamp-AT3G10030-DAP-Seq(GSE60143)/Homer\n",
      "\t\t73 of 1006 (1e-36) FOXK1(Forkhead)/HEK293-FOXK1-ChIP-Seq(GSE51673)/Homer\n",
      "\t\t74 of 1006 (1e-35) MYB65(MYB)/colamp-MYB65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t75 of 1006 (1e-35) FoxL2(Forkhead)/Ovary-FoxL2-ChIP-Seq(GSE60858)/Homer\n",
      "\t\t76 of 1006 (1e-33) FOXK2(Forkhead)/U2OS-FOXK2-ChIP-Seq(E-MTAB-2204)/Homer\n",
      "\t\t77 of 1006 (1e-33) AT2G40260(G2like)/colamp-AT2G40260-DAP-Seq(GSE60143)/Homer\n",
      "\t\t78 of 1006 (1e-33) BMYB(HTH)/Hela-BMYB-ChIP-Seq(GSE27030)/Homer\n",
      "\t\t79 of 1006 (1e-32) MafA(bZIP)/Islet-MafA-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t80 of 1006 (1e-32) bZIP52(bZIP)/colamp-bZIP52-DAP-Seq(GSE60143)/Homer\n",
      "\t\t81 of 1006 (1e-32) FOXA1(Forkhead)/MCF7-FOXA1-ChIP-Seq(GSE26831)/Homer\n",
      "\t\t82 of 1006 (1e-31) Fox:Ebox(Forkhead,bHLH)/Panc1-Foxa2-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t83 of 1006 (1e-28) MYB70(MYB)/col-MYB70-DAP-Seq(GSE60143)/Homer\n",
      "\t\t84 of 1006 (1e-28) Replumless(BLH)/Arabidopsis-RPL.GFP-ChIP-Seq(GSE78727)/Homer\n",
      "\t\t85 of 1006 (1e-28) bZIP18(bZIP)/colamp-bZIP18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t86 of 1006 (1e-27) FOXM1(Forkhead)/MCF7-FOXM1-ChIP-Seq(GSE72977)/Homer\n",
      "\t\t87 of 1006 (1e-27) Mef2d(MADS)/Retina-Mef2d-ChIP-Seq(GSE61391)/Homer\n",
      "\t\t88 of 1006 (1e-27) Ap4(bHLH)/AML-Tfap4-ChIP-Seq(GSE45738)/Homer\n",
      "\t\t89 of 1006 (1e-26) HOXA2(Homeobox)/mES-Hoxa2-ChIP-Seq(Donaldson_et_al.)/Homer\n",
      "\t\t90 of 1006 (1e-25) AMYB(HTH)/Testes-AMYB-ChIP-Seq(GSE44588)/Homer\n",
      "\t\t91 of 1006 (1e-25) HOXA1(Homeobox)/mES-Hoxa1-ChIP-Seq(SRP084292)/Homer\n",
      "\t\t92 of 1006 (1e-25) At5g04390(C2H2)/col200-At5g04390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t93 of 1006 (1e-25) PBX2(Homeobox)/K562-PBX2-ChIP-Seq(Encode)/Homer\n",
      "\t\t94 of 1006 (1e-23) HLH-1(bHLH)/cElegans-Embryo-HLH1-ChIP-Seq(modEncode)/Homer\n",
      "\t\t95 of 1006 (1e-22) Myf5(bHLH)/GM-Myf5-ChIP-Seq(GSE24852)/Homer\n",
      "\t\t96 of 1006 (1e-22) ZNF264(Zf)/HEK293-ZNF264.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t97 of 1006 (1e-22) Tbx6(T-box)/ESC-Tbx6-ChIP-Seq(GSE93524)/Homer\n",
      "\t\t98 of 1006 (1e-22) MYB(HTH)/ERMYB-Myb-ChIPSeq(GSE22095)/Homer\n",
      "\t\t99 of 1006 (1e-21) Pbx3(Homeobox)/GM12878-PBX3-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t100 of 1006 (1e-20) STZ(C2H2)/colamp-STZ-DAP-Seq(GSE60143)/Homer\n",
      "\t\t101 of 1006 (1e-20) Nur77(NR)/K562-NR4A1-ChIP-Seq(GSE31363)/Homer\n",
      "\t\t102 of 1006 (1e-19) NLP7(RWPRK)/col-NLP7-DAP-Seq(GSE60143)/Homer\n",
      "\t\t103 of 1006 (1e-19) MyoG(bHLH)/C2C12-MyoG-ChIP-Seq(GSE36024)/Homer\n",
      "\t\t104 of 1006 (1e-19) LIN-39(Homeobox)/cElegans.L3-LIN39-ChIP-Seq(modEncode)/Homer\n",
      "\t\t105 of 1006 (1e-18) Hoxa9(Homeobox)/ChickenMSG-Hoxa9.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t106 of 1006 (1e-18) E-box/Drosophila-Promoters/Homer\n",
      "\t\t107 of 1006 (1e-18) Foxf1(Forkhead)/Lung-Foxf1-ChIP-Seq(GSE77951)/Homer\n",
      "\t\t108 of 1006 (1e-18) Pknox1(Homeobox)/ES-Prep1-ChIP-Seq(GSE63282)/Homer\n",
      "\t\t109 of 1006 (1e-17) Atoh1(bHLH)/Cerebellum-Atoh1-ChIP-Seq(GSE22111)/Homer\n",
      "\t\t110 of 1006 (1e-17) Tcf21(bHLH)/ArterySmoothMuscle-Tcf21-ChIP-Seq(GSE61369)/Homer\n",
      "\t\t111 of 1006 (1e-17) CELF2(RRM)/JSL1-CELF2-CLIP-Seq(GSE71264)/Homer\n",
      "\t\t112 of 1006 (1e-17) AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t113 of 1006 (1e-16) MyoD(bHLH)/Myotube-MyoD-ChIP-Seq(GSE21614)/Homer\n",
      "\t\t114 of 1006 (1e-16) TATA-Box(TBP)/Promoter/Homer\n",
      "\t\t115 of 1006 (1e-16) Foxh1(Forkhead)/hESC-FOXH1-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t116 of 1006 (1e-16) Smad2(MAD)/ES-SMAD2-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t117 of 1006 (1e-15) MYB77(MYB)/col-MYB77-DAP-Seq(GSE60143)/Homer\n",
      "\t\t118 of 1006 (1e-15) SCL(bHLH)/HPC7-Scl-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t119 of 1006 (1e-15) ZFX(Zf)/mES-Zfx-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t120 of 1006 (1e-14) VND6(NAC)/col-VND6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t121 of 1006 (1e-14) LEF1(HMG)/H1-LEF1-ChIP-Seq(GSE64758)/Homer\n",
      "\t\t122 of 1006 (1e-14) Tbx5(T-box)/HL1-Tbx5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t123 of 1006 (1e-14) MYB88(MYB)/col-MYB88-DAP-Seq(GSE60143)/Homer\n",
      "\t\t124 of 1006 (1e-14) AT1G69570(C2C2dof)/col-AT1G69570-DAP-Seq(GSE60143)/Homer\n",
      "\t\t125 of 1006 (1e-13) Smad4(MAD)/ESC-SMAD4-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t126 of 1006 (1e-13) Brachyury(T-box)/Mesoendoderm-Brachyury-ChIP-exo(GSE54963)/Homer\n",
      "\t\t127 of 1006 (1e-13) Tcf4(HMG)/Hct116-Tcf4-ChIP-Seq(SRA012054)/Homer\n",
      "\t\t128 of 1006 (1e-13) CTCF(Zf)/CD4+-CTCF-ChIP-Seq(Barski_et_al.)/Homer\n",
      "\t\t129 of 1006 (1e-12) WRKY28(WRKY)/col-WRKY28-DAP-Seq(GSE60143)/Homer\n",
      "\t\t130 of 1006 (1e-12) MYB73(MYB)/col-MYB73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t131 of 1006 (1e-12) Tcf3(HMG)/mES-Tcf3-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t132 of 1006 (1e-12) MYB56(MYB)/colamp-MYB56-DAP-Seq(GSE60143)/Homer\n",
      "\t\t133 of 1006 (1e-12) VIP1(bZIP)/col-VIP1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t134 of 1006 (1e-11) ERF13(AP2EREBP)/colamp-ERF13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t135 of 1006 (1e-11) ANAC045(NAC)/col-ANAC045-DAP-Seq(GSE60143)/Homer\n",
      "\t\t136 of 1006 (1e-11) ZNF652/HepG2-ZNF652.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t137 of 1006 (1e-11) FoxD3(forkhead)/ZebrafishEmbryo-Foxd3.biotin-ChIP-seq(GSE106676)/Homer\n",
      "\t\t138 of 1006 (1e-11) NeuroD1(bHLH)/Islet-NeuroD1-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t139 of 1006 (1e-11) CArG(MADS)/PUER-Srf-ChIP-Seq(Sullivan_et_al.)/Homer\n",
      "\t\t140 of 1006 (1e-11) SND3(NAC)/col-SND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t141 of 1006 (1e-11) Rap210(AP2EREBP)/col-Rap210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t142 of 1006 (1e-10) RAP211(AP2EREBP)/colamp-RAP211-DAP-Seq(GSE60143)/Homer\n",
      "\t\t143 of 1006 (1e-10) VND2(NAC)/col-VND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t144 of 1006 (1e-10) MYB81(MYB)/col-MYB81-DAP-Seq(GSE60143)/Homer\n",
      "\t\t145 of 1006 (1e-10) ERF105(AP2EREBP)/colamp-ERF105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t146 of 1006 (1e-9) NeuroG2(bHLH)/Fibroblast-NeuroG2-ChIP-Seq(GSE75910)/Homer\n",
      "\t\t147 of 1006 (1e-9) GSC(Homeobox)/FrogEmbryos-GSC-ChIP-Seq(DRA000576)/Homer\n",
      "\t\t148 of 1006 (1e-9) CDF3(C2C2dof)/colamp-CDF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t149 of 1006 (1e-9) ANAC071(NAC)/col-ANAC071-DAP-Seq(GSE60143)/Homer\n",
      "\t\t150 of 1006 (1e-9) WRKY42(WRKY)/colamp-WRKY42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t151 of 1006 (1e-9) SMB(NAC)/colamp-SMB-DAP-Seq(GSE60143)/Homer\n",
      "\t\t152 of 1006 (1e-9) PBX1(Homeobox)/MCF7-PBX1-ChIP-Seq(GSE28007)/Homer\n",
      "\t\t153 of 1006 (1e-9) Tcf12(bHLH)/GM12878-Tcf12-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t154 of 1006 (1e-9) Bcl6(Zf)/Liver-Bcl6-ChIP-Seq(GSE31578)/Homer\n",
      "\t\t155 of 1006 (1e-9) ERF115(AP2EREBP)/colamp-ERF115-DAP-Seq(GSE60143)/Homer\n",
      "\t\t156 of 1006 (1e-9) WUS1(Homeobox)/colamp-WUS1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t157 of 1006 (1e-8) GATA3(Zf),DR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t158 of 1006 (1e-8) Knotted(Homeobox)/Corn-KN1-ChIP-Seq(GSE39161)/Homer\n",
      "\t\t159 of 1006 (1e-8) BORIS(Zf)/K562-CTCFL-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t160 of 1006 (1e-8) WRKY29(WRKY)/colamp-WRKY29-DAP-Seq(GSE60143)/Homer\n",
      "\t\t161 of 1006 (1e-8) Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer\n",
      "\t\t162 of 1006 (1e-8) AT2G33550(Trihelix)/colamp-AT2G33550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t163 of 1006 (1e-8) KAN2(G2like)/colamp-KAN2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t164 of 1006 (1e-8) BHLHA15(bHLH)/NIH3T3-BHLHB8.HA-ChIP-Seq(GSE119782)/Homer\n",
      "\t\t165 of 1006 (1e-8) Tlx?(NR)/NPC-H3K4me1-ChIP-Seq(GSE16256)/Homer\n",
      "\t\t166 of 1006 (1e-8) ERF104(AP2EREBP)/col-ERF104-DAP-Seq(GSE60143)/Homer\n",
      "\t\t167 of 1006 (1e-8) TR4(NR),DR1/Hela-TR4-ChIP-Seq(GSE24685)/Homer\n",
      "\t\t168 of 1006 (1e-8) TCF4(bHLH)/SHSY5Y-TCF4-ChIP-Seq(GSE96915)/Homer\n",
      "\t\t169 of 1006 (1e-8) NTM1(NAC)/col-NTM1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t170 of 1006 (1e-7) Tbox:Smad(T-box,MAD)/ESCd5-Smad2_3-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t171 of 1006 (1e-7) bZIP50(bZIP)/colamp-bZIP50-DAP-Seq(GSE60143)/Homer\n",
      "\t\t172 of 1006 (1e-7) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
      "\t\t173 of 1006 (1e-7) VND4(NAC)/colamp-VND4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t174 of 1006 (1e-7) WRKY27(WRKY)/colamp-WRKY27-DAP-Seq(GSE60143)/Homer\n",
      "\t\t175 of 1006 (1e-7) Fra1(bZIP)/BT549-Fra1-ChIP-Seq(GSE46166)/Homer\n",
      "\t\t176 of 1006 (1e-7) TCFL2(HMG)/K562-TCF7L2-ChIP-Seq(GSE29196)/Homer\n",
      "\t\t177 of 1006 (1e-7) VND3(NAC)/colamp-VND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t178 of 1006 (1e-7) bHLH122(bHLH)/col100-bHLH122-DAP-Seq(GSE60143)/Homer\n",
      "\t\t179 of 1006 (1e-7) ANAC083(NAC)/col-ANAC083-DAP-Seq(GSE60143)/Homer\n",
      "\t\t180 of 1006 (1e-7) NF1-halfsite(CTF)/LNCaP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t181 of 1006 (1e-7) PPARa(NR),DR1/Liver-Ppara-ChIP-Seq(GSE47954)/Homer\n",
      "\t\t182 of 1006 (1e-7) Fos(bZIP)/TSC-Fos-ChIP-Seq(GSE110950)/Homer\n",
      "\t\t183 of 1006 (1e-7) bHLH80(bHLH)/col-bHLH80-DAP-Seq(GSE60143)/Homer\n",
      "\t\t184 of 1006 (1e-7) Rfx2(HTH)/LoVo-RFX2-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t185 of 1006 (1e-7) Six1(Homeobox)/Myoblast-Six1-ChIP-Chip(GSE20150)/Homer\n",
      "\t\t186 of 1006 (1e-7) ANAC058(NAC)/col-ANAC058-DAP-Seq(GSE60143)/Homer\n",
      "\t\t187 of 1006 (1e-7) GATA(Zf),IR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t188 of 1006 (1e-6) TAGL1(MADS)/Tomato-TAGL1-ChIP-Seq(GSE116581)/Homer\n",
      "\t\t189 of 1006 (1e-6) AT3G25990(Trihelix)/colamp-AT3G25990-DAP-Seq(GSE60143)/Homer\n",
      "\t\t190 of 1006 (1e-6) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t191 of 1006 (1e-6) ANAC005(NAC)/col-ANAC005-DAP-Seq(GSE60143)/Homer\n",
      "\t\t192 of 1006 (1e-6) VND1(NAC)/col-VND1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t193 of 1006 (1e-6) RFX(HTH)/K562-RFX3-ChIP-Seq(SRA012198)/Homer\n",
      "\t\t194 of 1006 (1e-6) Zic(Zf)/Cerebellum-ZIC1.2-ChIP-Seq(GSE60731)/Homer\n",
      "\t\t195 of 1006 (1e-6) bZIP69(bZIP)/col-bZIP69-DAP-Seq(GSE60143)/Homer\n",
      "\t\t196 of 1006 (1e-6) CRF10(AP2EREBP)/col100-CRF10-DAP-Seq(GSE60143)/Homer\n",
      "\t\t197 of 1006 (1e-6) PHA-4(Forkhead)/cElegans-Embryos-PHA4-ChIP-Seq(modEncode)/Homer\n",
      "\t\t198 of 1006 (1e-6) Hoxc9(Homeobox)/Ainv15-Hoxc9-ChIP-Seq(GSE21812)/Homer\n",
      "\t\t199 of 1006 (1e-6) Twist2(bHLH)/Myoblast-Twist2.Ty1-ChIP-Seq(GSE127998)/Homer\n",
      "\t\t200 of 1006 (1e-6) STAT4(Stat)/CD4-Stat4-ChIP-Seq(GSE22104)/Homer\n",
      "\t\t201 of 1006 (1e-6) Tbet(T-box)/CD8-Tbet-ChIP-Seq(GSE33802)/Homer\n",
      "\t\t202 of 1006 (1e-6) NTM2(NAC)/col-NTM2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t203 of 1006 (1e-6) ANAC070(NAC)/colamp-ANAC070-DAP-Seq(GSE60143)/Homer\n",
      "\t\t204 of 1006 (1e-6) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t205 of 1006 (1e-6) ANAC004(NAC)/colamp-ANAC004-DAP-Seq(GSE60143)/Homer\n",
      "\t\t206 of 1006 (1e-5) Smad3(MAD)/NPC-Smad3-ChIP-Seq(GSE36673)/Homer\n",
      "\t\t207 of 1006 (1e-5) WRKY40(WRKY)/colamp-WRKY40-DAP-Seq(GSE60143)/Homer\n",
      "\t\t208 of 1006 (1e-5) RXR(NR),DR1/3T3L1-RXR-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t209 of 1006 (1e-5) Pitx1(Homeobox)/Chicken-Pitx1-ChIP-Seq(GSE38910)/Homer\n",
      "\t\t210 of 1006 (1e-5) WRKY65(WRKY)/colamp-WRKY65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t211 of 1006 (1e-5) ANAC096(NAC)/colamp-ANAC096-DAP-Seq(GSE60143)/Homer\n",
      "\t\t212 of 1006 (1e-5) Rbpj1(?)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t213 of 1006 (1e-5) Zic3(Zf)/mES-Zic3-ChIP-Seq(GSE37889)/Homer\n",
      "\t\t214 of 1006 (1e-5) WRKY15(WRKY)/col-WRKY15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t215 of 1006 (1e-5) ANAC057(NAC)/colamp-ANAC057-DAP-Seq(GSE60143)/Homer\n",
      "\t\t216 of 1006 (1e-5) WRKY33(WRKY)/col-WRKY33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t217 of 1006 (1e-5) NAC2(NAC)/colamp-NAC2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t218 of 1006 (1e-5) Fra2(bZIP)/Striatum-Fra2-ChIP-Seq(GSE43429)/Homer\n",
      "\t\t219 of 1006 (1e-5) HNF4a(NR),DR1/HepG2-HNF4a-ChIP-Seq(GSE25021)/Homer\n",
      "\t\t220 of 1006 (1e-5) SVP(MADS)/col-SVP-DAP-Seq(GSE60143)/Homer\n",
      "\t\t221 of 1006 (1e-5) MYB105(MYB)/colamp-MYB105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t222 of 1006 (1e-5) Tbx21(T-box)/GM12878-TBX21-ChIP-Seq(Encode)/Homer\n",
      "\t\t223 of 1006 (1e-5) ANAC028(NAC)/col-ANAC028-DAP-Seq(GSE60143)/Homer\n",
      "\t\t224 of 1006 (1e-5) WRKY14(WRKY)/colamp-WRKY14-DAP-Seq(GSE60143)/Homer\n",
      "\t\t225 of 1006 (1e-5) At3g60580(C2H2)/col-At3g60580-DAP-Seq(GSE60143)/Homer\n",
      "\t\t226 of 1006 (1e-5) PCF/Arabidopsis-Promoters/Homer\n",
      "\t\t227 of 1006 (1e-5) AT5G23930(mTERF)/col-AT5G23930-DAP-Seq(GSE60143)/Homer\n",
      "\t\t228 of 1006 (1e-5) EBF1(EBF)/Near-E2A-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t229 of 1006 (1e-5) JunB(bZIP)/DendriticCells-Junb-ChIP-Seq(GSE36099)/Homer\n",
      "\t\t230 of 1006 (1e-5) CBF1(AP2EREBP)/colamp-CBF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t231 of 1006 (1e-5) At1g19210(AP2EREBP)/colamp-At1g19210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t232 of 1006 (1e-5) WRKY6(WRKY)/colamp-WRKY6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t233 of 1006 (1e-5) ERF73(AP2EREBP)/col-ERF73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t234 of 1006 (1e-5) ERF15(AP2EREBP)/colamp-ERF15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t235 of 1006 (1e-5) Mef2b(MADS)/HEK293-Mef2b.V5-ChIP-Seq(GSE67450)/Homer\n",
      "\t\t236 of 1006 (1e-5) CBF3(AP2EREBP)/colamp-CBF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t237 of 1006 (1e-5) ANAC062(NAC)/colamp-ANAC062-DAP-Seq(GSE60143)/Homer\n",
      "\t\t238 of 1006 (1e-5) ERF11(AP2EREBP)/col-ERF11-DAP-Seq(GSE60143)/Homer\n",
      "\t\t239 of 1006 (1e-4) Unknown-ESC-element(?)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t240 of 1006 (1e-4) ZNF711(Zf)/SHSY5Y-ZNF711-ChIP-Seq(GSE20673)/Homer\n",
      "\t\t241 of 1006 (1e-4) SHN3(AP2EREBP)/col-SHN3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t242 of 1006 (1e-4) ERRg(NR)/Kidney-ESRRG-ChIP-Seq(GSE104905)/Homer\n",
      "\t\t243 of 1006 (1e-4) NST1(NAC)/colamp-NST1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t244 of 1006 (1e-4) TGA1(bZIP)/colamp-TGA1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t245 of 1006 (1e-4) Zic2(Zf)/ESC-Zic2-ChIP-Seq(SRP197560)/Homer\n",
      "\t\t246 of 1006 (1e-4) CBF2(AP2EREBP)/colamp-CBF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t247 of 1006 (1e-4) Rfx6(HTH)/Min6b1-Rfx6.HA-ChIP-Seq(GSE62844)/Homer\n",
      "\t\t248 of 1006 (1e-4) WRKY25(WRKY)/colamp-WRKY25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t249 of 1006 (1e-4) PUCHI(AP2EREBP)/colamp-PUCHI-DAP-Seq(GSE60143)/Homer\n",
      "\t\t250 of 1006 (1e-4) WRKY24(WRKY)/colamp-WRKY24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t251 of 1006 (1e-4) SOC1(MADS)/Seedling-SOC1-ChIP-Seq(GSE45846)/Homer\n",
      "\t\t252 of 1006 (1e-4) NF1(CTF)/LNCAP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t253 of 1006 (1e-4) ERF5(AP2EREBP)/colamp-ERF5-DAP-Seq(GSE60143)/Homer\n",
      "\t\t254 of 1006 (1e-4) HOXA9(Homeobox)/HSC-Hoxa9-ChIP-Seq(GSE33509)/Homer\n",
      "\t\t255 of 1006 (1e-4) STAT6(Stat)/Macrophage-Stat6-ChIP-Seq(GSE38377)/Homer\n",
      "\t\t256 of 1006 (1e-4) ANAC020(NAC)/col-ANAC020-DAP-Seq(GSE60143)/Homer\n",
      "\t\t257 of 1006 (1e-4) Six2(Homeobox)/NephronProgenitor-Six2-ChIP-Seq(GSE39837)/Homer\n",
      "\t\t258 of 1006 (1e-4) WRKY31(WRKY)/colamp-WRKY31-DAP-Seq(GSE60143)/Homer\n",
      "\t\t259 of 1006 (1e-4) Sox9(HMG)/Limb-SOX9-ChIP-Seq(GSE73225)/Homer\n",
      "\t\t260 of 1006 (1e-4) At4g16750(AP2EREBP)/col-At4g16750-DAP-Seq(GSE60143)/Homer\n",
      "\t\t261 of 1006 (1e-4) WRKY75(WRKY)/col-WRKY75-DAP-Seq(GSE60143)/Homer\n",
      "\t\t262 of 1006 (1e-4) AGL6(MADS)/col-AGL6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t263 of 1006 (1e-4) RAP26(AP2EREBP)/colamp-RAP26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t264 of 1006 (1e-4) Stat3+il21(Stat)/CD4-Stat3-ChIP-Seq(GSE19198)/Homer\n",
      "\t\t265 of 1006 (1e-4) ZNF7(Zf)/HepG2-ZNF7.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t266 of 1006 (1e-4) ERF2(AP2EREBP)/colamp-ERF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t267 of 1006 (1e-4) PRDM14(Zf)/H1-PRDM14-ChIP-Seq(GSE22767)/Homer\n",
      "\t\t268 of 1006 (1e-4) WRKY18(WRKY)/col-WRKY18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t269 of 1006 (1e-4) Atf3(bZIP)/GBM-ATF3-ChIP-Seq(GSE33912)/Homer\n",
      "\t\t270 of 1006 (1e-4) Prop1(Homeobox)/GHFT1-PROP1.biotin-ChIP-Seq(GSE77302)/Homer\n",
      "\t\t271 of 1006 (1e-4) WRKY3(WRKY)/col-WRKY3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t272 of 1006 (1e-4) STAT5(Stat)/mCD4+-Stat5-ChIP-Seq(GSE12346)/Homer\n",
      "\t\t273 of 1006 (1e-4) CUC1(NAC)/col-CUC1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t274 of 1006 (1e-4) CBF4(AP2EREBP)/colamp-CBF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t275 of 1006 (1e-4) bZIP42(bZIP)/colamp-bZIP42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t276 of 1006 (1e-4) NFAT(RHD)/Jurkat-NFATC1-ChIP-Seq(Jolma_et_al.)/Homer\n",
      "\t\t277 of 1006 (1e-3) Six4(Homeobox)/MCF7-SIX4-ChIP-Seq(Encode)/Homer\n",
      "\t\t278 of 1006 (1e-3) ERF8(AP2EREBP)/colamp-ERF8-DAP-Seq(GSE60143)/Homer\n",
      "\t\t279 of 1006 (1e-3) AT1G12630(AP2EREBP)/colamp-AT1G12630-DAP-Seq(GSE60143)/Homer\n",
      "\t\t280 of 1006 (1e-3) dof42(C2C2dof)/col-dof42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t281 of 1006 (1e-3) AGL16(MADS)/col-AGL16-DAP-Seq(GSE60143)/Homer\n",
      "\t\t282 of 1006 (1e-3) PPARE(NR),DR1/3T3L1-Pparg-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t283 of 1006 (1e-3) bZIP44(bZIP)/colamp-bZIP44-DAP-Seq(GSE60143)/Homer\n",
      "\t\t284 of 1006 (1e-3) SPL15(SBP)/colamp-SPL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t285 of 1006 (1e-3) PAX3:FKHR-fusion(Paired,Homeobox)/Rh4-PAX3:FKHR-ChIP-Seq(GSE19063)/Homer\n",
      "\t\t286 of 1006 (1e-3) Ascl1(bHLH)/NeuralTubes-Ascl1-ChIP-Seq(GSE55840)/Homer\n",
      "\t\t287 of 1006 (1e-3) ERF1(AP2EREBP)/colamp-ERF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t288 of 1006 (1e-3) CEJ1(AP2EREBP)/col-CEJ1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t289 of 1006 (1e-3) DDF2(AP2EREBP)/col-DDF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t290 of 1006 (1e-3) AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t291 of 1006 (1e-3) RAP212(AP2EREBP)/col-RAP212-DAP-Seq(GSE60143)/Homer\n",
      "\t\t292 of 1006 (1e-3) WRKY71(WRKY)/col-WRKY71-DAP-Seq(GSE60143)/Homer\n",
      "\t\t293 of 1006 (1e-3) WRKY26(WRKY)/colamp-WRKY26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t294 of 1006 (1e-3) Olig2(bHLH)/Neuron-Olig2-ChIP-Seq(GSE30882)/Homer\n",
      "\t\t295 of 1006 (1e-3) AT4G18450(AP2EREBP)/col-AT4G18450-DAP-Seq(GSE60143)/Homer\n",
      "\t\t296 of 1006 (1e-3) CRE(bZIP)/Promoter/Homer\n",
      "\t\t297 of 1006 (1e-3) AT1G76870(Trihelix)/col-AT1G76870-DAP-Seq(GSE60143)/Homer\n",
      "\t\t298 of 1006 (1e-3) Pitx1:Ebox(Homeobox,bHLH)/Hindlimb-Pitx1-ChIP-Seq(GSE41591)/Homer\n",
      "\t\t299 of 1006 (1e-3) REF6(Zf)/Arabidopsis-REF6-ChIP-Seq(GSE106942)/Homer\n",
      "\t\t300 of 1006 (1e-3) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
      "\t\t301 of 1006 (1e-3) EBF2(EBF)/BrownAdipose-EBF2-ChIP-Seq(GSE97114)/Homer\n",
      "\t\t302 of 1006 (1e-3) DEAR2(AP2EREBP)/colamp-DEAR2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t303 of 1006 (1e-3) WRKY22(WRKY)/colamp-WRKY22-DAP-Seq(GSE60143)/Homer\n",
      "\t\t304 of 1006 (1e-3) WRKY43(WRKY)/colamp-WRKY43-DAP-Seq(GSE60143)/Homer\n",
      "\t\t305 of 1006 (1e-3) AT5G47660(Trihelix)/colamp-AT5G47660-DAP-Seq(GSE60143)/Homer\n",
      "\t\t306 of 1006 (1e-3) Zac1(Zf)/Neuro2A-Plagl1-ChIP-Seq(GSE75942)/Homer\n",
      "\t\t307 of 1006 (1e-3) Initiator/Drosophila-Promoters/Homer\n",
      "\t\t308 of 1006 (1e-3) Sox10(HMG)/SciaticNerve-Sox3-ChIP-Seq(GSE35132)/Homer\n",
      "\t\t309 of 1006 (1e-3) WRKY20(WRKY)/col-WRKY20-DAP-Seq(GSE60143)/Homer\n",
      "\t\t310 of 1006 (1e-3) AGL15(MADS)/col-AGL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t311 of 1006 (1e-3) Stat3(Stat)/mES-Stat3-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t312 of 1006 (1e-3) MYB33(MYB)/col-MYB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t313 of 1006 (1e-3) TGA6(bZIP)/colamp-TGA6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t314 of 1006 (1e-3) ANAC053(NAC)/colamp-ANAC053-DAP-Seq(GSE60143)/Homer\n",
      "\t\t315 of 1006 (1e-3) ERF3(AP2EREBP)/colamp-ERF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t316 of 1006 (1e-3) TGA2(bZIP)/colamp-TGA2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t317 of 1006 (1e-3) ANAC050(NAC)/colamp-ANAC050-DAP-Seq(GSE60143)/Homer\n",
      "\t\t318 of 1006 (1e-3) ABR1(AP2EREBP)/colamp-ABR1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t319 of 1006 (1e-3) RKD2(RWPRK)/colamp-RKD2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t320 of 1006 (1e-3) AT1G77200(AP2EREBP)/colamp-AT1G77200-DAP-Seq(GSE60143)/Homer\n",
      "\t\t321 of 1006 (1e-3) RRTF1(AP2EREBP)/colamp-RRTF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t322 of 1006 (1e-3) ERF4(AP2EREBP)/colamp-ERF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t323 of 1006 (1e-3) ANAC092(NAC)/colamp-ANAC092-DAP-Seq(GSE60143)/Homer\n",
      "\t\t324 of 1006 (1e-2) MYB57(MYB)/col-MYB57-DAP-Seq(GSE60143)/Homer\n",
      "\t\t325 of 1006 (1e-2) ERF9(AP2EREBP)/colamp-ERF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t326 of 1006 (1e-2) At5g22890(C2H2)/col-At5g22890-DAP-Seq(GSE60143)/Homer\n",
      "\t\t327 of 1006 (1e-2) Esrrb(NR)/mES-Esrrb-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t328 of 1006 (1e-2) WRKY55(WRKY)/col-WRKY55-DAP-Seq(GSE60143)/Homer\n",
      "\t\t329 of 1006 (1e-2) Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer\n",
      "\t\t330 of 1006 (1e-2) E-box(bHLH)/Promoter/Homer\n",
      "\t\t331 of 1006 (1e-2) At1g36060(AP2EREBP)/colamp-At1g36060-DAP-Seq(GSE60143)/Homer\n",
      "\t\t332 of 1006 (1e-2) WT1(Zf)/Kidney-WT1-ChIP-Seq(GSE90016)/Homer\n",
      "\t\t333 of 1006 (1e-2) ANAC016(NAC)/col-ANAC016-DAP-Seq(GSE60143)/Homer\n",
      "\t\t334 of 1006 (1e-2) EBF(EBF)/proBcell-EBF-ChIP-Seq(GSE21978)/Homer\n",
      "\t\t335 of 1006 (1e-2) AGL13(MADS)/col-AGL13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t336 of 1006 (1e-2) SND2(NAC)/colamp-SND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t337 of 1006 (1e-2) Phox2a(Homeobox)/Neuron-Phox2a-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t338 of 1006 (1e-2) DDF1(AP2EREBP)/col-DDF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t339 of 1006 (1e-2) HAP3(CCAATHAP3)/col-HAP3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t340 of 1006 (1e-2) MYB116(MYB)/colamp-MYB116-DAP-Seq(GSE60143)/Homer\n",
      "\t\t341 of 1006 (1e-2) Bach1(bZIP)/K562-Bach1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t342 of 1006 (1e-2) Sox2(HMG)/mES-Sox2-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t343 of 1006 (1e-2) Sox3(HMG)/NPC-Sox3-ChIP-Seq(GSE33059)/Homer\n",
      "\t\t344 of 1006 (1e-2) RBPJ:Ebox(?,bHLH)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t345 of 1006 (1e-2) AT5G05550(Trihelix)/col-AT5G05550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t346 of 1006 (1e-2) EGL-5(Homeobox)/cElegans-L3-EGL5-ChIP-Seq(modEncode)/Homer\n",
      "\t\t347 of 1006 (1e-2) COUP-TFII(NR)/Artia-Nr2f2-ChIP-Seq(GSE46497)/Homer\n",
      "\t\t348 of 1006 (1e-2) OBP3(C2C2dof)/col-OBP3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t349 of 1006 (1e-2) At2g33710(AP2EREBP)/colamp-At2g33710-DAP-Seq(GSE60143)/Homer\n",
      "\t\t350 of 1006 (1e-2) Oct6(POU,Homeobox)/NPC-Pou3f1-ChIP-Seq(GSE35496)/Homer\n",
      "\t\t351 of 1006 (1e-2) RAR:RXR(NR),DR5/ES-RAR-ChIP-Seq(GSE56893)/Homer\n",
      "\t\t352 of 1006 (1e-2) STAT6(Stat)/CD4-Stat6-ChIP-Seq(GSE22104)/Homer\n",
      "\t\t353 of 1006 (1e-2) GEI-11(Myb?)/cElegans-L4-GEI11-ChIP-Seq(modEncode)/Homer\n",
      "\t\t354 of 1006 (1e-2) Pax7(Paired,Homeobox)/Myoblast-Pax7-ChIP-Seq(GSE25064)/Homer\n",
      "\t\t355 of 1006 (1e-2) LEP(AP2EREBP)/col-LEP-DAP-Seq(GSE60143)/Homer\n",
      "\t\t356 of 1006 (1e-2) REST-NRSF(Zf)/Jurkat-NRSF-ChIP-Seq/Homer\n",
      "\t\t357 of 1006 (1e-2) MYB3R4(MYB)/col-MYB3R4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t358 of 1006 (1e-2) Zfp281(Zf)/ES-Zfp281-ChIP-Seq(GSE81042)/Homer\n",
      "\t\t359 of 1006 (1e-2) WRKY46(WRKY)/colamp-WRKY46-DAP-Seq(GSE60143)/Homer\n",
      "\t\t360 of 1006 (1e-2) AP-2gamma(AP2)/MCF7-TFAP2C-ChIP-Seq(GSE21234)/Homer\n",
      "\t\t361 of 1006 (1e-2) Unknown2/Arabidopsis-Promoters/Homer\n",
      "\t\t362 of 1006 (1e-2) PRDM1(Zf)/Hela-PRDM1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t363 of 1006 (1e-2) MYB3R1(MYB)/col-MYB3R1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t364 of 1006 (1e-2) JunD(bZIP)/K562-JunD-ChIP-Seq/Homer\n",
      "\t\t365 of 1006 (1e-2) MYB93(MYB)/colamp-MYB93-DAP-Seq(GSE60143)/Homer\n",
      "\t\t366 of 1006 (1e-2) WRKY30(WRKY)/colamp-WRKY30-DAP-Seq(GSE60143)/Homer\n",
      "\t\t367 of 1006 (1e-2) USF1(bHLH)/GM12878-Usf1-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t368 of 1006 (1e-2) AT3G60490(AP2EREBP)/colamp-AT3G60490-DAP-Seq(GSE60143)/Homer\n",
      "\t\t369 of 1006 (1e-2) WRKY47(WRKY)/colamp-WRKY47-DAP-Seq(GSE60143)/Homer\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 32883\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 32883 of 34497 max\n",
      "\tTree  : 67200 of 172485 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 32883 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.87% skipped, 98.13% checked (32267 of 32883), of those checked:\n",
      "\t\t1.87% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 32883 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.87% skipped, 98.13% checked (32267 of 32883), of those checked:\n",
      "\t\t0.00% not in target, 48.27% increased p-value, 48.82% high p-value\n",
      "\n",
      "\tScreening oligos 32883 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t61.03% skipped, 38.97% checked (12816 of 32883), of those checked:\n",
      "\t\t0.00% not in target, 22.11% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAA... (-1267.546)\n",
      "\t\tRound 1: -2010.29 GCACTTAA T:3243.0(40.81%),B:4475.6(10.04%),P:1e-873\n",
      "\t\tRound 2: -2010.29 GCACTTAA T:3243.0(40.81%),B:4475.6(10.04%),P:1e-873\n",
      "\t\t=Final=: -2276.48 GCACTTAA T:2723.0(44.03%),B:4454.1(10.53%),P:1e-988\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: AGATAACG... (-895.513)\n",
      "\t\tRound 1: -2350.55 AGATAAGA T:3785.0(45.78%),B:4993.8(11.13%),P:1e-1020\n",
      "\t\tRound 2: -2350.55 AGATAAGA T:3785.0(45.78%),B:4993.8(11.13%),P:1e-1020\n",
      "\t\t=Final=: -2547.14 AGATAAGA T:2908.0(47.02%),B:4620.6(10.92%),P:1e-1106\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GCTAATGG... (-366.522)\n",
      "\t\tRound 1: -758.78 NSKAATGN T:3473.0(42.97%),B:9960.4(20.98%),P:1e-329\n",
      "\t\tRound 2: -845.52 AGGAATGY T:1169.0(17.23%),B:1637.0(3.80%),P:1e-367\n",
      "\t\tRound 3: -845.52 AGGAATGY T:1169.0(17.23%),B:1637.0(3.80%),P:1e-367\n",
      "\t\t=Final=: -852.64 AGGAATGY T:1067.0(17.25%),B:1598.0(3.78%),P:1e-370\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CTAATGGC... (-345.380)\n",
      "\t\tRound 1: -707.50 CTAATGGC T:1760.0(24.77%),B:3842.4(8.68%),P:1e-307\n",
      "\t\tRound 2: -707.50 CTAATGGC T:1760.0(24.77%),B:3842.4(8.68%),P:1e-307\n",
      "\t\t=Final=: -718.10 CTAATGGC T:1534.0(24.81%),B:3649.2(8.63%),P:1e-311\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GCTGTCAG... (-337.282)\n",
      "\t\tRound 1: -613.88 GCTGTCAG T:4927.0(54.92%),B:17098.7(33.25%),P:1e-266\n",
      "\t\tRound 2: -670.39 GYTGTCAG T:3069.0(39.12%),B:8950.4(19.07%),P:1e-291\n",
      "\t\tRound 3: -670.39 GYTGTCAG T:3069.0(39.12%),B:8950.4(19.07%),P:1e-291\n",
      "\t\t=Final=: -575.45 GYTGTCAG T:2179.0(35.24%),B:7315.8(17.30%),P:1e-249\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: CCCCCCCC... (-209.111)\n",
      "\t\tRound 1: -209.11 CCCCCCCC T:166.0(2.65%),B:134.0(0.31%),P:1e-90\n",
      "\t\tRound 2: -209.11 CCCCCCCC T:166.0(2.65%),B:134.0(0.31%),P:1e-90\n",
      "\t\t=Final=: -2.19 CCCCCCCC T:532.0(8.60%),B:3456.1(8.17%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: TGTTGACA... (-150.236)\n",
      "\t\tRound 1: -235.20 TGTTGACA T:1168.0(17.21%),B:3805.0(8.60%),P:1e-102\n",
      "\t\tRound 2: -235.20 TGTTGACA T:1168.0(17.21%),B:3805.0(8.60%),P:1e-102\n",
      "\t\t=Final=: -238.67 TGTTGACA T:1172.0(18.95%),B:4158.0(9.83%),P:1e-103\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: AATGGAAT... (-123.033)\n",
      "\t\tRound 1: -146.65 AATGGAAT T:264.0(4.18%),B:503.9(1.18%),P:1e-63\n",
      "\t\tRound 2: -146.65 AATGGAAT T:264.0(4.18%),B:503.9(1.18%),P:1e-63\n",
      "\t\t=Final=: -18.45 AATGGAAT T:111.0(1.79%),B:423.8(1.00%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: CTGAGTGC... (-120.790)\n",
      "\t\tRound 1: -146.62 BTGAGYGG T:1238.0(18.14%),B:4876.5(10.89%),P:1e-63\n",
      "\t\tRound 2: -249.01 BTRARHGS T:3512.0(43.33%),B:15075.1(29.98%),P:1e-108\n",
      "\t\tRound 3: -254.97 CTGARWGC T:3305.0(41.40%),B:13948.8(28.09%),P:1e-110\n",
      "\t\tRound 4: -254.97 CTGARWGC T:3305.0(41.40%),B:13948.8(28.09%),P:1e-110\n",
      "\t\t=Final=: -270.23 CTGARWGC T:2571.0(41.58%),B:11793.3(27.88%),P:1e-117\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: CGCTAATC... (-111.692)\n",
      "\t\tRound 1: -256.17 CGCTAATC T:1295.0(18.90%),B:4229.8(9.51%),P:1e-111\n",
      "\t\tRound 2: -260.30 CGCTAATC T:2240.0(30.39%),B:8624.8(18.45%),P:1e-113\n",
      "\t\tRound 3: -272.78 MGTTAATC T:2696.0(35.34%),B:10702.0(22.35%),P:1e-118\n",
      "\t\tRound 4: -273.19 HGCTAATC T:2723.0(35.62%),B:10824.9(22.58%),P:1e-118\n",
      "\t\tRound 5: -285.51 HGCTAATC T:2049.0(28.21%),B:7465.7(16.18%),P:1e-123\n",
      "\t\tRound 6: -285.51 HGCTAATC T:2049.0(28.21%),B:7465.7(16.18%),P:1e-123\n",
      "\t\t=Final=: -287.10 HGCTAATC T:1723.0(27.86%),B:6716.5(15.88%),P:1e-124\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: AAATTCCT... (-68.565)\n",
      "\t\tRound 1: -176.08 AAATTCCT T:619.0(9.53%),B:1762.0(4.08%),P:1e-76\n",
      "\t\tRound 2: -176.08 AAATTCCT T:619.0(9.53%),B:1762.0(4.08%),P:1e-76\n",
      "\t\t=Final=: -163.96 AAATTCCT T:561.0(9.07%),B:1660.4(3.93%),P:1e-71\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: AAAGCACT... (-59.381)\n",
      "\t\tRound 1: -127.55 AAAGCACT T:1394.0(20.18%),B:5885.4(12.99%),P:1e-55\n",
      "\t\tRound 2: -127.55 AAAGCACT T:1394.0(20.18%),B:5885.4(12.99%),P:1e-55\n",
      "\t\t=Final=: -124.13 AAAGCACT T:1274.0(20.60%),B:5679.5(13.43%),P:1e-53\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: AGGGCCTG... (-43.124)\n",
      "\t\tRound 1: -61.10 AGGGCCTG T:1817.0(25.46%),B:9343.4(19.82%),P:1e-26\n",
      "\t\tRound 2: -74.22 AGGGCCTG T:2418.0(32.36%),B:12495.5(25.58%),P:1e-32\n",
      "\t\tRound 3: -74.22 AGGGCCTG T:2418.0(32.36%),B:12495.5(25.58%),P:1e-32\n",
      "\t\t=Final=: -63.63 AGGGCCTG T:1728.0(27.94%),B:9298.0(21.98%),P:1e-27\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TGACAGCT... (-44.313)\n",
      "\t\tRound 1: -63.74 TGACAGCT T:689.0(10.54%),B:2962.1(6.76%),P:1e-27\n",
      "\t\tRound 2: -74.65 TGACAKCT T:1999.0(27.62%),B:10079.0(21.20%),P:1e-32\n",
      "\t\tRound 3: -79.61 TGACANYB T:1692.0(23.94%),B:8235.0(17.69%),P:1e-34\n",
      "\t\tRound 4: -79.61 TGACANYB T:1692.0(23.94%),B:8235.0(17.69%),P:1e-34\n",
      "\t\t=Final=: -70.69 TGACANYB T:1241.0(20.07%),B:6179.7(14.61%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: TGATAATG... (-42.701)\n",
      "\t\tRound 1: -54.04 TGATAATG T:2543.0(33.72%),B:13807.1(27.85%),P:1e-23\n",
      "\t\tRound 2: -73.33 AGATAAHG T:741.0(11.29%),B:3121.1(7.11%),P:1e-31\n",
      "\t\tRound 3: -99.14 WGATAABK T:1934.0(26.86%),B:9213.9(19.57%),P:1e-43\n",
      "\t\tRound 4: -99.14 WGATAABK T:1934.0(26.86%),B:9213.9(19.57%),P:1e-43\n",
      "\t\t=Final=: -122.08 WGATAABK T:1719.0(27.80%),B:8315.1(19.66%),P:1e-53\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: CAATTATC... (-61.593)\n",
      "\t\tRound 1: -18.10 CAATTATC T:182.0(2.90%),B:794.9(1.86%),P:1e-7\n",
      "\t\tRound 2: -70.01 YAMTTACC T:1948.0(27.02%),B:9893.3(20.86%),P:1e-30\n",
      "\t\tRound 3: -106.58 YACTTASC T:1635.0(23.23%),B:7451.1(16.15%),P:1e-46\n",
      "\t\tRound 4: -106.58 YACTTASC T:1635.0(23.23%),B:7451.1(16.15%),P:1e-46\n",
      "\t\t=Final=: -93.59 YACTTASC T:1373.0(22.20%),B:6632.1(15.68%),P:1e-40\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: GAGTGTTT... (-39.492)\n",
      "\t\tRound 1: -58.26 GAGTGCTT T:467.0(7.27%),B:1858.9(4.30%),P:1e-25\n",
      "\t\tRound 2: -62.88 GAGTGTTT T:270.0(4.27%),B:865.4(2.02%),P:1e-27\n",
      "\t\tRound 3: -62.88 GAGTGTTT T:270.0(4.27%),B:865.4(2.02%),P:1e-27\n",
      "\t\t=Final=: -61.94 GAGTGTTT T:263.0(4.25%),B:857.5(2.03%),P:1e-26\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: TGTTTACT... (-34.797)\n",
      "\t\tRound 1: -44.13 TGTTTACT T:209.0(3.32%),B:702.6(1.65%),P:1e-19\n",
      "\t\tRound 2: -44.13 TGTTTACT T:209.0(3.32%),B:702.6(1.65%),P:1e-19\n",
      "\t\t=Final=: -39.73 TGTTTACT T:199.0(3.22%),B:698.0(1.65%),P:1e-17\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: CGCCGCCG... (-31.605)\n",
      "\t\tRound 1: -64.64 CGCCGCCG T:213.0(3.39%),B:604.2(1.42%),P:1e-28\n",
      "\t\tRound 2: -64.64 CGCCGCCG T:213.0(3.39%),B:604.2(1.42%),P:1e-28\n",
      "\t\t=Final=: -20.59 CGCCGCCG T:124.0(2.01%),B:471.7(1.12%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: TGCTATCT... (-26.418)\n",
      "\t\tRound 1: -28.58 TGCTATCT T:1450.0(20.90%),B:8060.6(17.35%),P:1e-12\n",
      "\t\tRound 2: -38.60 TGCTVTYT T:1320.0(19.22%),B:6987.5(15.23%),P:1e-16\n",
      "\t\tRound 3: -38.60 TGCTVTYT T:1320.0(19.22%),B:6987.5(15.23%),P:1e-16\n",
      "\t\t=Final=: -42.77 TGCTVTYT T:1196.0(19.34%),B:6401.8(15.14%),P:1e-18\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: CACTTTGC... (-22.623)\n",
      "\t\tRound 1: -67.22 CACTTTMC T:853.0(12.89%),B:3791.6(8.57%),P:1e-29\n",
      "\t\tRound 2: -67.22 CACTTTMC T:853.0(12.89%),B:3791.6(8.57%),P:1e-29\n",
      "\t\t=Final=: -70.94 CACTTTMC T:801.0(12.95%),B:3609.5(8.53%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: AAGAATGC... (-22.527)\n",
      "\t\tRound 1: -54.79 AAGAATGC T:394.0(6.17%),B:1522.0(3.53%),P:1e-23\n",
      "\t\tRound 2: -61.62 RRGAATGY T:496.0(7.71%),B:1974.9(4.56%),P:1e-26\n",
      "\t\tRound 3: -63.84 RAGAATGT T:507.0(7.87%),B:2007.3(4.63%),P:1e-27\n",
      "\t\tRound 4: -63.84 RAGAATGT T:507.0(7.87%),B:2007.3(4.63%),P:1e-27\n",
      "\t\t=Final=: -59.49 RAGAATGT T:475.0(7.68%),B:1943.7(4.60%),P:1e-25\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: TGTTATCA... (-19.754)\n",
      "\t\tRound 1: -21.96 TGTTATCA T:64.0(1.03%),B:176.0(0.42%),P:1e-9\n",
      "\t\tRound 2: -32.33 YGTHRTCA T:96.0(1.54%),B:264.3(0.62%),P:1e-14\n",
      "\t\tRound 3: -32.33 YGTHRTCA T:96.0(1.54%),B:264.3(0.62%),P:1e-14\n",
      "\t\t=Final=: -33.70 YGTHRTCA T:96.0(1.55%),B:261.2(0.62%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: ATCGAATG... (-18.447)\n",
      "\t\tRound 1: -22.61 ATCRAABG T:83.0(1.33%),B:256.0(0.60%),P:1e-9\n",
      "\t\tRound 2: -22.61 ATCRAABG T:83.0(1.33%),B:256.0(0.60%),P:1e-9\n",
      "\t\t=Final=: -18.87 ATCRAABG T:77.0(1.25%),B:254.5(0.60%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: GCACTTCA... (-17.167)\n",
      "Optimal threshold is low - motif appears very degenerate!!!\n",
      "\t\tRound 1: -4.70 GCACTTCA T:39.0(0.63%),B:172.6(0.41%),P:1e-2\n",
      "\t\tRound 2: -19.31 GCACTTGA T:111.0(1.78%),B:414.2(0.97%),P:1e-8\n",
      "\t\tRound 3: -19.31 GCACTTGA T:111.0(1.78%),B:414.2(0.97%),P:1e-8\n",
      "\t\t=Final=: -20.07 GCACTTGA T:111.0(1.79%),B:410.0(0.97%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs8\n",
      "\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 480669\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 480669 of 551887 max\n",
      "\tTree  : 997284 of 2759435 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 480669 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.40% skipped, 67.60% checked (324956 of 480669), of those checked:\n",
      "\t\t32.40% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 480669 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.40% skipped, 67.60% checked (324956 of 480669), of those checked:\n",
      "\t\t0.00% not in target, 30.56% increased p-value, 32.08% high p-value\n",
      "\n",
      "\tScreening oligos 480669 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.02% skipped, 25.98% checked (124876 of 480669), of those checked:\n",
      "\t\t0.00% not in target, 7.23% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAACG... (-1131.863)\n",
      "\t\tRound 1: -2068.04 GCACTTAACG T:3031.0(38.75%),B:3857.1(8.72%),P:1e-898\n",
      "\t\tRound 2: -2147.76 GCACTTARCN T:3356.0(41.88%),B:4445.1(9.98%),P:1e-932\n",
      "\t\tRound 3: -2147.76 GCACTTARCN T:3356.0(41.88%),B:4445.1(9.98%),P:1e-932\n",
      "\t\t=Final=: -2468.27 GCACTTARCN T:3594.0(58.12%),B:7635.7(18.05%),P:1e-1071\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CTTATCTCCG... (-476.466)\n",
      "\t\tRound 1: -1662.91 CTTATCTCCG T:2143.0(29.29%),B:2562.9(5.88%),P:1e-722\n",
      "\t\tRound 2: -1954.73 CTTATCWBNN T:3359.0(41.91%),B:4887.0(10.91%),P:1e-848\n",
      "\t\tRound 3: -2004.50 STTATCTNDN T:3123.0(39.65%),B:4204.5(9.46%),P:1e-870\n",
      "\t\tRound 4: -2004.50 STTATCTNDN T:3123.0(39.65%),B:4204.5(9.46%),P:1e-870\n",
      "\t\t=Final=: -2143.53 STTATCTNDN T:2488.0(40.23%),B:3866.6(9.14%),P:1e-930\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GGGGGGGGGG... (-360.848)\n",
      "\t\tRound 1: -360.85 GGGGGGGGGG T:149.0(2.38%),B:35.4(0.08%),P:1e-156\n",
      "\t\tRound 2: -360.85 GGGGGGGGGG T:149.0(2.38%),B:35.4(0.08%),P:1e-156\n",
      "\t\t=Final=: -3.26 GGGGGGGGGG T:651.0(10.53%),B:4164.8(9.85%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CCGCTAATGG... (-293.869)\n",
      "\t\tRound 1: -576.12 CCGCTAATGG T:1032.0(15.37%),B:1845.7(4.27%),P:1e-250\n",
      "\t\tRound 2: -617.18 NCRCTAATGG T:1383.0(20.04%),B:2871.7(6.56%),P:1e-268\n",
      "\t\tRound 3: -666.11 NBGCTAATGG T:1424.0(20.57%),B:2850.0(6.52%),P:1e-289\n",
      "\t\tRound 4: -666.11 NBGCTAATGG T:1424.0(20.57%),B:2850.0(6.52%),P:1e-289\n",
      "\t\t=Final=: -672.98 NBGCTAATGG T:1270.0(20.54%),B:2728.8(6.45%),P:1e-292\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GTGTTGACAG... (-262.040)\n",
      "\t\tRound 1: -460.73 GTGTTGACAG T:1621.0(23.06%),B:4398.5(9.88%),P:1e-200\n",
      "\t\tRound 2: -509.63 GTGTTGACAG T:1767.0(24.86%),B:4725.3(10.57%),P:1e-221\n",
      "\t\tRound 3: -528.32 STGYTGACAG T:2125.0(29.08%),B:6043.3(13.31%),P:1e-229\n",
      "\t\tRound 4: -590.45 STGCTGACAG T:3074.0(39.17%),B:9518.3(20.15%),P:1e-256\n",
      "\t\tRound 5: -608.32 BTGCTGACAG T:3673.0(44.79%),B:11867.5(24.46%),P:1e-264\n",
      "\t\tRound 6: -652.19 BWGCTGACAG T:3252.0(40.90%),B:9807.7(20.69%),P:1e-283\n",
      "\t\tRound 7: -652.19 BWGCTGACAG T:3252.0(40.90%),B:9807.7(20.69%),P:1e-283\n",
      "\t\t=Final=: -543.81 BWGCTGACAG T:2247.0(36.34%),B:7852.3(18.56%),P:1e-236\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: CGAATGGAAT... (-236.476)\n",
      "\t\tRound 1: -318.90 SGAATGGAAT T:442.0(6.90%),B:660.7(1.55%),P:1e-138\n",
      "\t\tRound 2: -342.16 GGAATGBAAT T:520.0(8.07%),B:839.9(1.96%),P:1e-148\n",
      "\t\tRound 3: -493.58 GGAATGTAAT T:1384.0(20.05%),B:3335.9(7.58%),P:1e-214\n",
      "\t\tRound 4: -574.64 GGAATGTAAW T:1299.0(18.95%),B:2725.9(6.24%),P:1e-249\n",
      "\t\tRound 5: -694.15 GGAATGTRNT T:1313.0(19.13%),B:2409.8(5.54%),P:1e-301\n",
      "\t\tRound 6: -694.15 GGAATGTRNT T:1313.0(19.13%),B:2409.8(5.54%),P:1e-301\n",
      "\t\t=Final=: -632.49 GGAATGTRNT T:1122.0(18.14%),B:2281.8(5.39%),P:1e-274\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: CACTTATCTC... (-164.864)\n",
      "\t\tRound 1: -58.76 CACTTATCTC T:216.0(3.43%),B:647.2(1.52%),P:1e-25\n",
      "\t\tRound 2: -204.60 CHCTTATCWC T:2497.0(33.22%),B:10566.9(22.10%),P:1e-88\n",
      "\t\tRound 3: -286.34 BNYTTATCWC T:2791.0(36.32%),B:11005.2(22.91%),P:1e-124\n",
      "\t\tRound 4: -303.31 NNHTTATCWB T:3394.0(42.24%),B:13729.0(27.72%),P:1e-131\n",
      "\t\tRound 5: -335.58 BNHTTATCWC T:2070.0(28.45%),B:7129.0(15.51%),P:1e-145\n",
      "\t\tRound 6: -352.64 BNHTTATCWS T:2105.0(28.85%),B:7148.4(15.55%),P:1e-153\n",
      "\t\tRound 7: -393.49 BBCTTATCWC T:2328.0(31.37%),B:7819.5(16.88%),P:1e-170\n",
      "\t\tRound 8: -393.49 BBCTTATCWC T:2328.0(31.37%),B:7819.5(16.88%),P:1e-170\n",
      "\t\t=Final=: -428.79 BBCTTATCWC T:2023.0(32.71%),B:7354.2(17.39%),P:1e-186\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: AGATAAGATA... (-130.614)\n",
      "\t\tRound 1: -13.40 AGATAAGATA T:154.0(2.46%),B:702.0(1.64%),P:1e-5\n",
      "\t\tRound 2: -32.15 AGATTAGATW T:46.0(0.74%),B:76.3(0.18%),P:1e-13\n",
      "\t\tRound 3: -44.63 AGATTAGCTW T:101.0(1.62%),B:235.4(0.55%),P:1e-19\n",
      "\t\tRound 4: -44.63 AGATTAGCTW T:101.0(1.62%),B:235.4(0.55%),P:1e-19\n",
      "\t\t=Final=: -43.27 AGATTAGCTW T:99.0(1.60%),B:235.4(0.56%),P:1e-18\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: ATCGAATGGA... (-116.567)\n",
      "\t\tRound 1: -121.27 ATCGAATGGA T:33.0(0.53%),B:3.7(0.01%),P:1e-52\n",
      "\t\tRound 2: -121.27 ATCGAATGGA T:33.0(0.53%),B:3.7(0.01%),P:1e-52\n",
      "\t\t=Final=: -27.67 ATCGAATGGA T:10.0(0.16%),B:2.1(0.00%),P:1e-12\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: AGAGATAATG... (-106.261)\n",
      "\t\tRound 1: -17.13 AGAGATAATG T:312.0(4.92%),B:1538.8(3.57%),P:1e-7\n",
      "\t\tRound 2: -30.56 AGAGBTAATG T:294.0(4.64%),B:1251.5(2.91%),P:1e-13\n",
      "\t\tRound 3: -34.62 AGAGVTAATG T:240.0(3.81%),B:928.7(2.17%),P:1e-15\n",
      "\t\tRound 4: -43.12 AGAGNTAATG T:460.0(7.17%),B:1988.1(4.59%),P:1e-18\n",
      "\t\tRound 5: -45.68 RGAGNTAATG T:317.0(5.00%),B:1219.5(2.84%),P:1e-19\n",
      "\t\tRound 6: -45.68 RGAGNTAATG T:317.0(5.00%),B:1219.5(2.84%),P:1e-19\n",
      "\t\t=Final=: -45.70 RGAGNTAATG T:310.0(5.01%),B:1206.7(2.85%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: CACTTATCTT... (-90.948)\n",
      "\t\tRound 1: -18.51 CACTTATCTT T:141.0(2.25%),B:571.9(1.34%),P:1e-8\n",
      "\t\tRound 2: -49.48 CACTWATCTT T:75.0(1.21%),B:129.4(0.30%),P:1e-21\n",
      "\t\tRound 3: -50.70 SACTWAYCTT T:112.0(1.79%),B:256.6(0.60%),P:1e-22\n",
      "\t\tRound 4: -50.70 SACTWAYCTT T:112.0(1.79%),B:256.6(0.60%),P:1e-22\n",
      "\t\t=Final=: -48.78 SACTWAYCTT T:109.0(1.76%),B:254.3(0.60%),P:1e-21\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: CACATTCCTG... (-90.948)\n",
      "\t\tRound 1: -37.31 NACATTCCTN T:259.0(4.10%),B:999.0(2.33%),P:1e-16\n",
      "\t\tRound 2: -108.24 VACATTCCTB T:610.0(9.39%),B:2136.9(4.92%),P:1e-47\n",
      "\t\tRound 3: -150.34 VRMATTCCTG T:588.0(9.07%),B:1763.7(4.08%),P:1e-65\n",
      "\t\tRound 4: -150.34 VRMATTCCTG T:588.0(9.07%),B:1763.7(4.08%),P:1e-65\n",
      "\t\t=Final=: -153.00 VRMATTCCTG T:597.0(9.65%),B:1882.2(4.45%),P:1e-66\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: CACATTCCTC... (-80.960)\n",
      "\t\tRound 1: -13.63 CACATTCCTC T:112.0(1.79%),B:471.2(1.11%),P:1e-5\n",
      "\t\tRound 2: -32.74 CACATTCYTY T:75.0(1.21%),B:177.6(0.42%),P:1e-14\n",
      "\t\tRound 3: -39.97 CACATTCYTY T:91.0(1.46%),B:213.2(0.50%),P:1e-17\n",
      "\t\tRound 4: -62.07 VACATTCYTT T:191.0(3.04%),B:525.6(1.23%),P:1e-26\n",
      "\t\tRound 5: -72.22 VRCRTTCYTY T:176.0(2.81%),B:421.1(0.99%),P:1e-31\n",
      "\t\tRound 6: -73.54 AACATTCTTT T:160.0(2.55%),B:354.9(0.83%),P:1e-31\n",
      "\t\tRound 7: -73.54 AACATTCTTT T:160.0(2.55%),B:354.9(0.83%),P:1e-31\n",
      "\t\t=Final=: -70.15 AACATTCTTT T:154.0(2.49%),B:352.5(0.83%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TAAGTGCTTT... (-77.849)\n",
      "\t\tRound 1: -142.58 TAAGTGCTTT T:879.0(13.25%),B:3151.1(7.18%),P:1e-61\n",
      "\t\tRound 2: -183.96 WAAGTGCTTT T:668.0(10.24%),B:1927.1(4.45%),P:1e-79\n",
      "\t\tRound 3: -201.71 WRAGTGCTTT T:702.0(10.73%),B:1976.8(4.56%),P:1e-87\n",
      "\t\tRound 4: -204.10 WRAGTGCTTT T:1635.0(23.23%),B:6255.5(13.75%),P:1e-88\n",
      "\t\tRound 5: -219.93 WRAGWGCTTT T:2195.0(29.88%),B:8848.4(18.88%),P:1e-95\n",
      "\t\tRound 6: -222.73 TRAGWGCYTT T:2801.0(36.43%),B:11861.7(24.45%),P:1e-96\n",
      "\t\tRound 7: -222.73 TRAGWGCYTT T:2801.0(36.43%),B:11861.7(24.45%),P:1e-96\n",
      "\t\t=Final=: -203.56 TRAGWGCYTT T:2869.0(46.39%),B:14410.3(34.07%),P:1e-88\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: CAGATAAGAA... (-71.168)\n",
      "\t\tRound 1: -6.86 CGKMWMSGYG T:31.0(0.50%),B:110.4(0.26%),P:1e-2\n",
      "\t\tRound 2: -10.33 CGKCTMAGCG T:14.0(0.23%),B:23.4(0.05%),P:1e-4\n",
      "\t\tRound 3: -14.52 CGKCTCAGCG T:6.0(0.10%),B:1.7(0.00%),P:1e-6\n",
      "\t\tRound 4: -24.14 CGKCAMAGCG T:10.0(0.16%),B:3.9(0.01%),P:1e-10\n",
      "\t\tRound 5: -26.66 CGGCGCAGCG T:16.0(0.26%),B:9.3(0.02%),P:1e-11\n",
      "\t\tRound 6: -31.31 CGSCDSCGCG T:69.0(1.11%),B:159.3(0.38%),P:1e-13\n",
      "\t\tRound 7: -31.31 CGSCDSCGCG T:69.0(1.11%),B:159.3(0.38%),P:1e-13\n",
      "\t\t=Final=: -23.95 CGSCDSCGCG T:56.0(0.91%),B:138.2(0.33%),P:1e-10\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: TCACTTATCT... (-71.168)\n",
      "\t\tRound 1: -19.88 NCACTTATCN T:649.0(9.96%),B:3469.3(7.87%),P:1e-8\n",
      "\t\tRound 2: -52.65 TCACTTATCH T:1380.0(20.00%),B:7003.5(15.26%),P:1e-22\n",
      "\t\tRound 3: -52.73 TCACTTANCH T:1791.0(25.15%),B:9403.1(19.93%),P:1e-22\n",
      "\t\tRound 4: -79.11 KCACTTANCM T:1025.0(15.28%),B:4559.1(10.22%),P:1e-34\n",
      "\t\tRound 5: -104.55 BCAMTTANCM T:1220.0(17.91%),B:5263.2(11.70%),P:1e-45\n",
      "\t\tRound 6: -126.64 KTAATTANCM T:335.0(5.27%),B:830.8(1.94%),P:1e-54\n",
      "\t\tRound 7: -126.64 KTAATTANCM T:335.0(5.27%),B:830.8(1.94%),P:1e-54\n",
      "\t\t=Final=: -111.07 KTAATTANCM T:307.0(4.96%),B:806.0(1.91%),P:1e-48\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: GATAAGATAC... (-66.352)\n",
      "\t\tRound 1: -85.93 GATAAGATAC T:22.0(0.36%),B:2.5(0.00%),P:1e-37\n",
      "\t\tRound 2: -85.93 GATAAGATAC T:22.0(0.36%),B:2.5(0.00%),P:1e-37\n",
      "\t\t=Final=: -75.84 GATAAGATAC T:22.0(0.36%),B:2.5(0.01%),P:1e-32\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: AGCATTCCTT... (-61.593)\n",
      "\t\tRound 1: -7.75 AGCATTCCTT T:8.0(0.13%),B:10.4(0.02%),P:1e-3\n",
      "\t\tRound 2: -11.81 AGCATGCCTT T:11.0(0.18%),B:13.0(0.03%),P:1e-5\n",
      "\t\tRound 3: -14.52 AGCATGCGTT T:6.0(0.10%),B:1.5(0.00%),P:1e-6\n",
      "\t\tRound 4: -14.66 AGCATGCSHT T:12.0(0.19%),B:11.0(0.03%),P:1e-6\n",
      "\t\tRound 5: -17.63 AGCMTGCGAT T:31.0(0.50%),B:62.7(0.15%),P:1e-7\n",
      "\t\tRound 6: -18.37 AGCCTGCGWT T:94.0(1.51%),B:336.7(0.79%),P:1e-7\n",
      "\t\tRound 7: -18.37 AGCCTGCGWT T:94.0(1.51%),B:336.7(0.79%),P:1e-7\n",
      "\t\t=Final=: -18.78 AGCCTGCGWT T:93.0(1.50%),B:331.3(0.78%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: GCTTGTTTAC... (-52.689)\n",
      "\t\tRound 1: -56.10 GCTTGTTTAC T:23.0(0.37%),B:6.8(0.01%),P:1e-24\n",
      "\t\tRound 2: -56.10 GCTTGTTTAC T:23.0(0.37%),B:6.8(0.01%),P:1e-24\n",
      "\t\t=Final=: -55.50 GCTTGTTTAC T:23.0(0.37%),B:6.8(0.02%),P:1e-24\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: CTTATCTCTT... (-52.261)\n",
      "\t\tRound 1: -3.83 NTTATCTCTN T:4.0(0.06%),B:5.7(0.01%),P:1e-1\n",
      "\t\tRound 2: -23.61 WTTATCTCTN T:146.0(2.33%),B:551.3(1.29%),P:1e-10\n",
      "\t\tRound 3: -27.62 HTTATCTCTN T:132.0(2.11%),B:451.2(1.06%),P:1e-11\n",
      "\t\tRound 4: -31.95 TTTATCTCTN T:977.0(14.62%),B:5121.1(11.40%),P:1e-13\n",
      "\t\tRound 5: -44.65 YTTATCYCTN T:872.0(13.15%),B:4251.7(9.56%),P:1e-19\n",
      "\t\tRound 6: -66.91 BTTATCBYYN T:860.0(12.98%),B:3833.1(8.66%),P:1e-29\n",
      "\t\tRound 7: -76.54 YTAATCYCYH T:512.0(7.95%),B:1918.2(4.43%),P:1e-33\n",
      "\t\tRound 8: -77.27 BTAATCBCYN T:501.0(7.78%),B:1859.6(4.30%),P:1e-33\n",
      "\t\tRound 9: -78.57 TTAATCYCTH T:434.0(6.78%),B:1525.1(3.54%),P:1e-34\n",
      "\t\tRound 10: -78.57 TTAATCYCTH T:434.0(6.78%),B:1525.1(3.54%),P:1e-34\n",
      "\t\t=Final=: -70.24 TTAATCYCTH T:397.0(6.42%),B:1449.2(3.43%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: CTAAGAGCTG... (-47.696)\n",
      "\t\tRound 1: -14.12 CTAAGAGCTG T:241.0(3.82%),B:1181.7(2.75%),P:1e-6\n",
      "\t\tRound 2: -15.43 BWVAGAGCTN T:1019.0(15.19%),B:5873.5(12.96%),P:1e-6\n",
      "\t\tRound 3: -17.38 BWSAGAGCTN T:818.0(12.39%),B:4564.6(10.23%),P:1e-7\n",
      "\t\tRound 4: -17.38 BWSAGAGCTN T:818.0(12.39%),B:4564.6(10.23%),P:1e-7\n",
      "\t\t=Final=: -16.64 BWSAGAGCTN T:762.0(12.32%),B:4322.7(10.22%),P:1e-7\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: TCGAATGGAA... (-44.650)\n",
      "\t\tRound 1: -53.30 TCGAATGGAA T:104.0(1.67%),B:218.3(0.51%),P:1e-23\n",
      "\t\tRound 2: -53.30 TCGAATGGAA T:104.0(1.67%),B:218.3(0.51%),P:1e-23\n",
      "\t\t=Final=: -16.07 TCGAATGGAA T:63.0(1.02%),B:206.7(0.49%),P:1e-6\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: CTAAGAGCTC... (-43.206)\n",
      "\t\tRound 1: -7.67 CTAAGAGCTC T:4.0(0.06%),B:1.0(0.00%),P:1e-3\n",
      "\t\tRound 2: -10.44 CTAAKAGCTC T:15.0(0.24%),B:26.2(0.06%),P:1e-4\n",
      "\t\tRound 3: -12.30 CTAAKAGCTC T:59.0(0.95%),B:208.7(0.49%),P:1e-5\n",
      "\t\tRound 4: -18.24 CKAAKAGCWC T:7.0(0.11%),B:1.9(0.00%),P:1e-7\n",
      "\t\tRound 5: -18.24 CKAAKAGCWC T:7.0(0.11%),B:1.9(0.00%),P:1e-7\n",
      "\t\t=Final=: -22.12 CKAAKAGCWC T:7.0(0.11%),B:1.9(0.00%),P:1e-9\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: CTAGTGGTGT... (-43.206)\n",
      "\t\tRound 1: -46.63 CTAGTGGTGT T:16.0(0.26%),B:3.4(0.01%),P:1e-20\n",
      "\t\tRound 2: -46.63 CTAGTGGTGT T:16.0(0.26%),B:3.4(0.01%),P:1e-20\n",
      "\t\t=Final=: -44.29 CTAGTGGTGT T:16.0(0.26%),B:3.4(0.01%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: TCACTCAACA... (-38.815)\n",
      "\t\tRound 1: -63.90 TCACTCAACA T:228.0(3.62%),B:670.2(1.57%),P:1e-27\n",
      "\t\tRound 2: -75.51 TCACTCAACA T:268.0(4.24%),B:785.9(1.84%),P:1e-32\n",
      "\t\tRound 3: -78.70 TCACTCAACA T:359.0(5.64%),B:1169.0(2.73%),P:1e-34\n",
      "\t\tRound 4: -82.86 TCABTCAACA T:289.0(4.57%),B:837.6(1.96%),P:1e-35\n",
      "\t\tRound 5: -82.86 TCABTCAACA T:289.0(4.57%),B:837.6(1.96%),P:1e-35\n",
      "\t\t=Final=: -83.18 TCABTCAACA T:281.0(4.54%),B:822.9(1.95%),P:1e-36\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs10\n",
      "\n",
      "\n",
      "\t-blen automatically set to 2\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 3655721\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 3655721 of 8830118 max\n",
      "\tTree  : 8307096 of 44150590 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 3655721 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.11% skipped, 25.89% checked (946623 of 3655721), of those checked:\n",
      "\t\t74.11% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 3655721 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t74.11% skipped, 25.89% checked (946623 of 3655721), of those checked:\n",
      "\t\t0.00% not in target, 17.14% increased p-value, 4.14% high p-value\n",
      "\n",
      "\tScreening oligos 3655721 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t91.78% skipped, 8.22% checked (300367 of 3655721), of those checked:\n",
      "\t\t0.00% not in target, 2.58% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: AAAGCACTTAAC... (-705.985)\n",
      "\t\tRound 1: -1484.76 AAAGCACTTAAC T:3427.0(42.55%),B:6416.2(14.07%),P:1e-644\n",
      "\t\tRound 2: -1936.27 AAAGCACTTARC T:2850.0(36.93%),B:3697.4(8.37%),P:1e-840\n",
      "\t\tRound 3: -2138.81 AAAGCACTTARV T:3090.0(39.33%),B:3853.7(8.71%),P:1e-928\n",
      "\t\tRound 4: -2228.90 WAAGCACTTARS T:3026.0(38.70%),B:3543.6(8.04%),P:1e-967\n",
      "\t\tRound 5: -2228.90 WAAGCACTTARS T:3026.0(38.70%),B:3543.6(8.04%),P:1e-967\n",
      "\t\t=Final=: -2487.02 WAAGCACTTARS T:2516.0(40.69%),B:3383.4(8.00%),P:1e-1080\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: GGGGGGGGGGGG... (-521.603)\n",
      "\t\tRound 1: -521.60 GGGGGGGGGGGG T:139.0(2.22%),B:9.6(0.02%),P:1e-226\n",
      "\t\tRound 2: -521.60 GGGGGGGGGGGG T:139.0(2.22%),B:9.6(0.02%),P:1e-226\n",
      "\t\t=Final=: -5.26 GGGGGGGGGGGG T:144.0(2.33%),B:789.1(1.87%),P:1e-2\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: CCTTATCTTATC... (-348.717)\n",
      "\t\tRound 1: -681.39 CCTTATCTTATC T:1887.0(26.30%),B:4392.5(9.86%),P:1e-295\n",
      "\t\tRound 2: -876.40 CCTTATCTTATC T:1901.0(26.47%),B:3731.0(8.44%),P:1e-380\n",
      "\t\tRound 3: -1315.40 YCTTATCTTATC T:2204.0(29.98%),B:3420.7(7.77%),P:1e-571\n",
      "\t\tRound 4: -1813.52 YCTTATCTBHNN T:3000.0(38.44%),B:4330.0(9.73%),P:1e-787\n",
      "\t\tRound 5: -1814.69 YCTTATCTBHNN T:2572.0(34.03%),B:3284.4(7.47%),P:1e-788\n",
      "\t\tRound 6: -1952.52 YBTTATCTBHNN T:3506.0(43.28%),B:5263.1(11.70%),P:1e-847\n",
      "\t\tRound 7: -2097.03 NCTTATCTNHNN T:3797.0(45.88%),B:5619.0(12.44%),P:1e-910\n",
      "\t\tRound 8: -2097.03 NCTTATCTNHNN T:3797.0(45.88%),B:5619.0(12.44%),P:1e-910\n",
      "\t\t=Final=: -2266.69 NCTTATCTNHNN T:2900.0(46.90%),B:5142.9(12.16%),P:1e-984\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: ATCGAATGGAAT... (-278.897)\n",
      "\t\tRound 1: -303.45 ATSGAATGGAAT T:271.0(4.29%),B:253.0(0.59%),P:1e-131\n",
      "\t\tRound 2: -303.45 ATSGAATGGAAT T:271.0(4.29%),B:253.0(0.59%),P:1e-131\n",
      "\t\t=Final=: -33.73 ATSGAATGGAAT T:21.0(0.34%),B:14.0(0.03%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: CTGGAATGTGTC... (-187.066)\n",
      "\t\tRound 1: -320.93 CTGGAATGTGTC T:997.0(14.89%),B:2600.8(5.96%),P:1e-139\n",
      "\t\tRound 2: -320.93 CTGGAATGTGTC T:997.0(14.89%),B:2600.8(5.96%),P:1e-139\n",
      "\t\t=Final=: -302.36 CTGGAATGTGTC T:899.0(14.54%),B:2510.9(5.94%),P:1e-131\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: AGGTGTTGACAG... (-170.346)\n",
      "\t\tRound 1: -317.55 AGGTGTTGACAG T:1156.0(17.05%),B:3260.3(7.42%),P:1e-137\n",
      "\t\tRound 2: -346.48 AGGTGTTGACAG T:2331.0(31.41%),B:8225.4(17.67%),P:1e-150\n",
      "\t\tRound 3: -387.18 AGGTGTTGACAG T:987.0(14.75%),B:2288.1(5.27%),P:1e-168\n",
      "\t\tRound 4: -387.18 AGGTGTTGACAG T:987.0(14.75%),B:2288.1(5.27%),P:1e-168\n",
      "\t\t=Final=: -389.43 AGGTGTTGACAG T:913.0(14.76%),B:2223.8(5.26%),P:1e-169\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: AAGCCATTAGCG... (-131.472)\n",
      "\t\tRound 1: -296.47 NNGCCATTAGCG T:2053.0(28.25%),B:7385.5(16.02%),P:1e-128\n",
      "\t\tRound 2: -428.66 AWGCCATTAGCR T:717.0(10.95%),B:1241.1(2.89%),P:1e-186\n",
      "\t\tRound 3: -526.74 AWNCCATTAGSR T:660.0(10.12%),B:874.9(2.05%),P:1e-228\n",
      "\t\tRound 4: -575.23 NDBCCATTAGCN T:1022.0(15.23%),B:1818.4(4.21%),P:1e-249\n",
      "\t\tRound 5: -576.31 AWNCCATTAGBN T:1157.0(17.06%),B:2246.5(5.17%),P:1e-250\n",
      "\t\tRound 6: -576.31 AWNCCATTAGBN T:1157.0(17.06%),B:2246.5(5.17%),P:1e-250\n",
      "\t\t=Final=: -583.03 AWNCCATTAGBN T:1052.0(17.01%),B:2157.4(5.10%),P:1e-253\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: AGGAATGTTAAG... (-117.768)\n",
      "\t\tRound 1: -215.53 AGGAATGTTAAG T:174.0(2.77%),B:144.0(0.34%),P:1e-93\n",
      "\t\tRound 2: -280.75 AGGAATGTTAAG T:401.0(6.28%),B:619.4(1.45%),P:1e-121\n",
      "\t\tRound 3: -345.26 AGGAATGTDAAK T:579.0(8.94%),B:1012.3(2.36%),P:1e-149\n",
      "\t\tRound 4: -401.79 AGGAATGTWAWK T:1359.0(19.73%),B:3654.3(8.28%),P:1e-174\n",
      "\t\tRound 5: -461.94 AGGAATGYHNWN T:1097.0(16.26%),B:2412.7(5.54%),P:1e-200\n",
      "\t\tRound 6: -469.82 AGGAATGTWVWK T:1231.0(18.05%),B:2865.8(6.55%),P:1e-204\n",
      "\t\tRound 7: -469.82 AGGAATGTWVWK T:1231.0(18.05%),B:2865.8(6.55%),P:1e-204\n",
      "\t\t=Final=: -471.34 AGGAATGTWVWK T:1083.0(17.51%),B:2628.5(6.21%),P:1e-204\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: GACAGCTGTCAG... (-99.305)\n",
      "\t\tRound 1: -128.77 GACAGCTGTCAG T:554.0(8.57%),B:1739.7(4.03%),P:1e-55\n",
      "\t\tRound 2: -131.06 GACAGCTGTCAG T:827.0(12.52%),B:2997.4(6.84%),P:1e-56\n",
      "\t\tRound 3: -131.06 GACAGCTGTCAG T:827.0(12.52%),B:2997.4(6.84%),P:1e-56\n",
      "\t\t=Final=: -102.27 GACAGCTGTCAG T:408.0(6.60%),B:1293.2(3.06%),P:1e-44\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: TCGAATGGAATC... (-74.016)\n",
      "\t\tRound 1: -108.43 TCGAATGGAATC T:33.0(0.53%),B:4.6(0.01%),P:1e-47\n",
      "\t\tRound 2: -108.43 TCGAATGGAATC T:33.0(0.53%),B:4.6(0.01%),P:1e-47\n",
      "\t\t=Final=: -47.95 TCGAATGGAATC T:17.0(0.27%),B:4.0(0.01%),P:1e-20\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: AGCTTGTTTACT... (-71.168)\n",
      "\t\tRound 1: -84.62 AGCTTGTTTACT T:25.0(0.40%),B:3.8(0.01%),P:1e-36\n",
      "\t\tRound 2: -96.01 AGCTTGTTAACT T:24.0(0.39%),B:2.7(0.00%),P:1e-41\n",
      "\t\tRound 3: -96.01 AGCTTGTTAACT T:24.0(0.39%),B:2.7(0.00%),P:1e-41\n",
      "\t\t=Final=: -90.95 AGCTTGTTAACT T:22.0(0.36%),B:1.9(0.00%),P:1e-39\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: GGAGTCAGCTTG... (-71.168)\n",
      "\t\tRound 1: -85.93 GGAGTCAGCTTG T:22.0(0.36%),B:2.6(0.00%),P:1e-37\n",
      "\t\tRound 2: -85.93 GGAGTCAGCTTG T:22.0(0.36%),B:2.6(0.00%),P:1e-37\n",
      "\t\t=Final=: -76.04 GGAGTCAGCTTG T:19.0(0.31%),B:1.9(0.00%),P:1e-33\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: CAGTATCTTATC... (-66.352)\n",
      "\t\tRound 1: -85.93 CAGTATCTTATC T:22.0(0.36%),B:1.8(0.00%),P:1e-37\n",
      "\t\tRound 2: -85.93 CAGTATCTTATC T:22.0(0.36%),B:1.8(0.00%),P:1e-37\n",
      "\t\t=Final=: -90.95 CAGTATCTTATC T:22.0(0.36%),B:1.8(0.00%),P:1e-39\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: CTGGTTAATGTC... (-59.074)\n",
      "\t\tRound 1: -130.16 CTGGTTAATGTC T:515.0(7.99%),B:1562.1(3.63%),P:1e-56\n",
      "\t\tRound 2: -177.75 CTGKTTAATGTC T:587.0(9.06%),B:1620.8(3.76%),P:1e-77\n",
      "\t\tRound 3: -225.42 CTGBTTAATSTC T:543.0(8.41%),B:1246.0(2.90%),P:1e-97\n",
      "\t\tRound 4: -257.10 CTGBTTWHTSTC T:667.0(10.23%),B:1593.1(3.70%),P:1e-111\n",
      "\t\tRound 5: -290.94 CTGBTTWHTSTC T:749.0(11.41%),B:1771.5(4.10%),P:1e-126\n",
      "\t\tRound 6: -290.94 CTGBTTWHTSTC T:749.0(11.41%),B:1771.5(4.10%),P:1e-126\n",
      "\t\t=Final=: -297.02 CTGBTTWHTSTC T:709.0(11.47%),B:1728.9(4.09%),P:1e-128\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: GAAACACTCTTT... (-56.894)\n",
      "\t\tRound 1: -63.14 GAAACACTCTTT T:22.0(0.36%),B:4.7(0.01%),P:1e-27\n",
      "\t\tRound 2: -67.06 GAAACACTCTGT T:23.0(0.37%),B:4.7(0.01%),P:1e-29\n",
      "\t\tRound 3: -67.06 GAAACACTCTGT T:23.0(0.37%),B:4.7(0.01%),P:1e-29\n",
      "\t\t=Final=: -60.87 GAAACACTCTGT T:22.0(0.36%),B:4.7(0.01%),P:1e-26\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: GCTAAGTGAGTG... (-52.497)\n",
      "\t\tRound 1: -93.84 GCTAAGTGAGTG T:795.0(12.06%),B:3179.2(7.24%),P:1e-40\n",
      "\t\tRound 2: -101.79 GCTAAGTGAGTG T:640.0(9.83%),B:2335.6(5.37%),P:1e-44\n",
      "\t\tRound 3: -107.88 GCTAAGTGAGTG T:596.0(9.19%),B:2076.6(4.79%),P:1e-46\n",
      "\t\tRound 4: -107.88 GCTAAGTGAGTG T:596.0(9.19%),B:2076.6(4.79%),P:1e-46\n",
      "\t\t=Final=: -121.24 GCTAAGTGAGTG T:1178.0(19.05%),B:5166.4(12.21%),P:1e-52\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: CTAACCTTTGAT... (-51.621)\n",
      "\t\tRound 1: -88.58 CTAACCTTTGAT T:32.0(0.52%),B:6.5(0.01%),P:1e-38\n",
      "\t\tRound 2: -96.01 CTAACCTTTGAT T:24.0(0.39%),B:0.9(0.00%),P:1e-41\n",
      "\t\tRound 3: -96.01 CTAACCTTTGAT T:24.0(0.39%),B:0.9(0.00%),P:1e-41\n",
      "\t\t=Final=: -101.12 CTAACCTTTGAT T:24.0(0.39%),B:0.9(0.00%),P:1e-43\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: GGTAATTGACAT... (-50.205)\n",
      "\t\tRound 1: -99.06 GGTAATTGACAT T:45.0(0.73%),B:14.3(0.03%),P:1e-43\n",
      "\t\tRound 2: -99.06 GGTAATTGACAT T:45.0(0.73%),B:14.3(0.03%),P:1e-43\n",
      "\t\t=Final=: -99.04 GGTAATTGACAT T:45.0(0.73%),B:14.3(0.03%),P:1e-43\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: AAGCACTTAGCA... (-47.696)\n",
      "\t\tRound 1: -55.10 NNGCACTTAGBD T:812.0(12.31%),B:3745.1(8.47%),P:1e-23\n",
      "\t\tRound 2: -83.70 NDGCACTTAGND T:1008.0(15.04%),B:4408.0(9.90%),P:1e-36\n",
      "\t\tRound 3: -134.09 NDGCACTTAGNN T:2309.0(31.16%),B:10650.4(22.26%),P:1e-58\n",
      "\t\tRound 4: -134.09 NDGCACTTAGNN T:2309.0(31.16%),B:10650.4(22.26%),P:1e-58\n",
      "\t\t=Final=: -76.40 NDGCACTTAGNN T:2072.0(33.51%),B:11228.7(26.55%),P:1e-33\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: CTTGGATGTGCT... (-47.696)\n",
      "\t\tRound 1: -71.02 CTTGGATGTGCT T:24.0(0.39%),B:4.8(0.01%),P:1e-30\n",
      "\t\tRound 2: -71.17 CTTGGATGTGCT T:19.0(0.31%),B:1.5(0.00%),P:1e-30\n",
      "\t\tRound 3: -76.04 CTTGGATGTGCT T:20.0(0.32%),B:2.1(0.00%),P:1e-33\n",
      "\t\tRound 4: -76.04 CTTGGATGTGCT T:20.0(0.32%),B:2.1(0.00%),P:1e-33\n",
      "\t\t=Final=: -76.04 CTTGGATGTGCT T:19.0(0.31%),B:1.5(0.00%),P:1e-33\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: GCCTGTCAGCCG... (-45.934)\n",
      "\t\tRound 1: -70.30 GCCTGTCAGCCG T:1689.0(23.90%),B:8406.3(18.02%),P:1e-30\n",
      "\t\tRound 2: -89.63 GCCTGTCAGCMD T:1193.0(17.55%),B:5321.7(11.82%),P:1e-38\n",
      "\t\tRound 3: -107.43 GCCTGTCAGCHD T:1001.0(14.95%),B:4088.7(9.21%),P:1e-46\n",
      "\t\tRound 4: -110.03 GCCTGTCAGCHD T:1178.0(17.35%),B:4972.4(11.09%),P:1e-47\n",
      "\t\tRound 5: -110.94 GCCTGTCAGCMA T:498.0(7.74%),B:1599.5(3.71%),P:1e-48\n",
      "\t\tRound 6: -123.87 GCCTGTCAGSMA T:1567.0(22.39%),B:6839.1(14.93%),P:1e-53\n",
      "\t\tRound 7: -163.14 NCCTGWCARNNN T:987.0(14.75%),B:3500.2(7.94%),P:1e-70\n",
      "\t\tRound 8: -185.41 NSCTGWCARNNW T:1198.0(17.61%),B:4321.6(9.71%),P:1e-80\n",
      "\t\tRound 9: -185.41 NSCTGWCARNNW T:1198.0(17.61%),B:4321.6(9.71%),P:1e-80\n",
      "\t\t=Final=: -120.00 NSCTGWCARNNW T:681.0(11.01%),B:2503.3(5.92%),P:1e-52\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: GTTCTACGAATG... (-43.206)\n",
      "\t\tRound 1: -43.21 GTTCTACGAATG T:13.0(0.21%),B:1.8(0.00%),P:1e-18\n",
      "\t\tRound 2: -43.21 GTTCTACGAATG T:13.0(0.21%),B:1.8(0.00%),P:1e-18\n",
      "\t\t=Final=: -47.70 GTTCTACGAATG T:13.0(0.21%),B:1.8(0.00%),P:1e-20\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: TAAGGATAAGGA... (-43.206)\n",
      "\t\tRound 1: -47.70 TAAGGATAAGGA T:14.0(0.23%),B:0.8(0.00%),P:1e-20\n",
      "\t\tRound 2: -47.70 TAAGGATAAGGA T:14.0(0.23%),B:0.8(0.00%),P:1e-20\n",
      "\t\t=Final=: -7.67 TAAGGATAAGGA T:3.0(0.05%),B:0.8(0.00%),P:1e-3\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: CAGTATATAATA... (-43.206)\n",
      "\t\tRound 1: -47.70 CAGTATATAATA T:14.0(0.23%),B:2.8(0.00%),P:1e-20\n",
      "\t\tRound 2: -47.70 CAGTATATAATA T:14.0(0.23%),B:2.8(0.00%),P:1e-20\n",
      "\t\t=Final=: -8.33 CAGTATATAATA T:4.0(0.06%),B:2.8(0.01%),P:1e-3\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: CTCAGCACAGAC... (-43.206)\n",
      "\t\tRound 1: -47.70 CTCAGCACAGAC T:14.0(0.23%),B:2.8(0.00%),P:1e-20\n",
      "\t\tRound 2: -47.70 CTCAGCACAGAC T:14.0(0.23%),B:2.8(0.00%),P:1e-20\n",
      "\t\t=Final=: -42.69 CTCAGCACAGAC T:14.0(0.23%),B:2.8(0.01%),P:1e-18\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs12\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 75 reduced to 22 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 22 (1e-1166) similar to GATA5/MA0766.2/Jaspar(0.971)\n",
      "\t\t2 of 22 (1e-1099) similar to ISL2/MA0914.1/Jaspar(0.933)\n",
      "\t\t3 of 22 (1e-409) similar to TEAD3/MA0808.1/Jaspar(0.958)\n",
      "\t\t4 of 22 (1e-323) similar to Tgif1(Homeobox)/mES-Tgif1-ChIP-Seq(GSE55404)/Homer(0.908)\n",
      "\t\t5 of 22 (1e-311) similar to Isl1/MA1608.1/Jaspar(0.975)\n",
      "\t\t6 of 22 (1e-196) similar to Hand2(bHLH)/Mesoderm-Hand2-ChIP-Seq(GSE61475)/Homer(0.918)\n",
      "\t\t7 of 22 (1e-78) similar to brk/dmmpmm(Down)/fly(0.707)\n",
      "\t\t8 of 22 (1e-67) similar to MafA(bZIP)/Islet-MafA-ChIP-Seq(GSE30298)/Homer(0.760)\n",
      "\t\t9 of 22 (1e-54) similar to ARF34/MA1693.1/Jaspar(0.798)\n",
      "\t\t10 of 22 (1e-39) similar to PH0053.1_Hoxa6/Jaspar(0.734)\n",
      "\t\t11 of 22 (1e-33) similar to BAS1/BAS1_SM/2-BAS1(Harbison)/Yeast(0.711)\n",
      "\t\t12 of 22 (1e-25) similar to TCF7/MA0769.2/Jaspar(0.706)\n",
      "\t\t13 of 22 (1e-24) similar to PH0168.1_Hnf1b/Jaspar(0.801)\n",
      "\t\t14 of 22 (1e-22) similar to MOD(RRM)/Drosophila_melanogaster-RNCMPT00140-PBM/HughesRNA(0.712)\n",
      "\t\t15 of 22 (1e-22) similar to NCU08034(RRM)/Neurospora_crassa-RNCMPT00209-PBM/HughesRNA(0.653)\n",
      "\t\t16 of 22 (1e-15) similar to Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer(0.731)\n",
      "\t\t17 of 22 (1e-15) similar to PRDM4/MA1647.1/Jaspar(0.676)\n",
      "\t\t18 of 22 (1e-12) similar to Deaf1/dmmpmm(SeSiMCMC)/fly(0.698)\n",
      "\t\t19 of 22 (1e-11) similar to Os05g0497200/MA1034.1/Jaspar(0.938)\n",
      "\t\t20 of 22 (1e-11) similar to EIF-2ALPHA(S1)/Drosophila_melanogaster-RNCMPT00273-PBM/HughesRNA(0.709)\n",
      "\t\t21 of 22 (1e-3) similar to NHP6A/MA0345.1/Jaspar(0.738)\n",
      "\t\t22 of 22 (1e-1) similar to SeqBias: polyC-repeat(1.000)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# For the ISL1 TF dataset. Note: This command will around an hour to run.\n",
    "python -m timeit -r 1 -s 'import os' 'os.system(\"findMotifsGenome.pl analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed benchmark/hg19.fa homer-results/isl1 > /dev/null\")'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9694d4dc",
   "metadata": {},
   "source": [
    "#### Summary\n",
    "\n",
    "Runtime per loop of basic usage commands for `pyxis` vs. `HOMER` is as follows:\n",
    "\n",
    "| Tool | Dataset | Runtime Per Loop (1 loop, best of 5) |\n",
    "| :--- | :------ | :----------------------------------: |\n",
    "| Pyxis | Example Data | 1.81 sec |\n",
    "| Pyxis | ISL1 | 341 sec |\n",
    "| HOMER | Example Data | 186 sec |\n",
    "| HOMER | ISL1 | 3.12e+03 sec |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "6f1e1d5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Basic Usage Command:\n",
      "1 loop, best of 5: 1.81 sec per loop\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "printf 'Basic Usage Command:\\n'\n",
    "python -m timeit -s 'import os' 'os.system(\"pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms > /dev/null\")'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a911b2a",
   "metadata": {},
   "source": [
    "## II. Memory Consumption"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "9c00a752",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fatal: destination path 'memusg' already exists and is not an empty directory.\r\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/jhclark/memusg.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "41b463f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/esquan/final_project/pyxis'"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd # check that you are still in the pyxis directory"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97a4754f",
   "metadata": {},
   "source": [
    "IMPORTANT: Only run the below cell __once__ or you will encounter an error in the following cells."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "3417f26f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "mv: cannot stat 'memusg/': Not a directory\n",
      "mv: cannot stat 'memusg': No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Moving the memusg script to the pyxis directory for execution\n",
    "mv memusg/ memusage/\n",
    "cd memusage\n",
    "mv memusg ..\n",
    "cd ..\n",
    "rm -rf memusage\n",
    "export PATH=."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59316312",
   "metadata": {},
   "source": [
    "### Testing with Example Files Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "db0d33a8",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Background not specified. Generating background sequences from reference genome...\n",
      "Calculating thresholds for each PWM...\n",
      "[1/5] .... \n",
      "[2/5] .... \n",
      "[3/5] .... \n",
      "[4/5] .... \n",
      "[5/5] .... Done.\n",
      "\n",
      "Performing motif enrichment...\n",
      "[1/5] .... \n",
      "[2/5] .... \n",
      "[3/5] .... \n",
      "[4/5] .... \n",
      "[5/5] .... Done.\n",
      "\n",
      "Creating output file 'enrichment_results.tsv'...\n",
      "....\n",
      "....\n",
      "....\n",
      "....\n",
      ".... Done.\n",
      "\n",
      "\n",
      "Pyxis has run successfully, please check 'pyxis_enrichments.tsv' for motif enrichment info!\n",
      "If '-s' or '--seqlogo' was specified, the motif sequence logos should be in your directory.\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "memusg: vmpeak: 17959672 kb\n",
      "memusg: vmpeak: 17959676 kb\n",
      "memusg: vmpeak: 17959680 kb\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "# Testing pyxis memory usage\n",
    "# 1. Basic usage command\n",
    "./memusg pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms > /dev/null\n",
    "\n",
    "# 2. Including a background peaks file\n",
    "./memusg pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -b example-files/background.bed > /dev/null\n",
    "\n",
    "# 3. Including background + generation of sequence logos\n",
    "./memusg pyxis example-files/peaks.bed example-files/ref.fa example-files/test.pwms -b example-files/background.bed -s > /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "2f2be2ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = example-files/peaks.bed\n",
      "\tGenome = example-files/ref.fa\n",
      "\tOutput Directory = homer-results/example-data\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 10\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 10\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground files for 200 bp fragments found.\n",
      "!!!! Might have something wrong with preparsed files\n",
      "!!!! Rerun and add \"-preparse\" to the command line to force recreation of the files\n",
      "\tCustom genome sequence file: example-files/ref.fa\n",
      "\n",
      "\tExtracting sequences from file: example-files/ref.fa\n",
      "\tLooking for peak sequences in a single file (example-files/ref.fa)\n",
      "\tExtracting 2 sequences from chr1\n",
      "\tExtracting 1 sequences from chr2\n",
      "\tExtracting 2 sequences from chr3\n",
      "\tExtracting 1 sequences from chr4\n",
      "\tExtracting 3 sequences from chr5\n",
      "\tExtracting 1 sequences from chr6\n",
      "\n",
      "\n",
      "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "\tPotential Problem with Sequences (ignore if preparsing genome):\n",
      "\t\t9 of 10 removed because they had >70.00% Ns (i.e. masked repeats)\n",
      "\tDue to \thigh repeat content, consider using a non-repeat masked genome\n",
      "\n",
      "\n",
      "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 101 bp\n",
      "\t\t1 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.5\t6\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t6\t1\t62\t1.000\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t50.00%\t26.50%\t1.887\n",
      "\t\tC\t50.00%\t23.50%\t2.128\n",
      "\t\tG\t50.00%\t23.50%\t2.128\n",
      "\t\tT\t50.00%\t26.50%\t1.887\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t50.00%\t6.41%\t7.797\n",
      "\t\tCA\t50.00%\t6.08%\t8.226\n",
      "\t\tGA\t50.00%\t6.62%\t7.552\n",
      "\t\tTA\t50.00%\t7.30%\t6.845\n",
      "\t\tAC\t50.00%\t6.31%\t7.918\n",
      "\t\tCC\t50.00%\t5.33%\t9.384\n",
      "\t\tGC\t50.00%\t5.22%\t9.579\n",
      "\t\tTC\t50.00%\t6.62%\t7.552\n",
      "\t\tAG\t50.00%\t6.34%\t7.886\n",
      "\t\tCG\t50.00%\t5.72%\t8.738\n",
      "\t\tGG\t50.00%\t5.33%\t9.384\n",
      "\t\tTG\t50.00%\t6.08%\t8.226\n",
      "\t\tAT\t50.00%\t7.56%\t6.610\n",
      "\t\tCT\t50.00%\t6.34%\t7.886\n",
      "\t\tGT\t50.00%\t6.31%\t7.918\n",
      "\t\tTT\t50.00%\t6.41%\t7.797\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/example-data/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t50.00%\t26.50%\t1.887\n",
      "\t\tC\t50.00%\t23.50%\t2.128\n",
      "\t\tG\t50.00%\t23.50%\t2.128\n",
      "\t\tT\t50.00%\t26.50%\t1.887\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t50.00%\t6.41%\t7.799\n",
      "\t\tCA\t50.00%\t6.08%\t8.227\n",
      "\t\tGA\t50.00%\t6.62%\t7.549\n",
      "\t\tTA\t50.00%\t7.30%\t6.845\n",
      "\t\tAC\t50.00%\t6.31%\t7.919\n",
      "\t\tCC\t50.00%\t5.33%\t9.386\n",
      "\t\tGC\t50.00%\t5.22%\t9.580\n",
      "\t\tTC\t50.00%\t6.62%\t7.549\n",
      "\t\tAG\t50.00%\t6.34%\t7.885\n",
      "\t\tCG\t50.00%\t5.72%\t8.738\n",
      "\t\tGG\t50.00%\t5.33%\t9.386\n",
      "\t\tTG\t50.00%\t6.08%\t8.227\n",
      "\t\tAT\t50.00%\t7.57%\t6.609\n",
      "\t\tCT\t50.00%\t6.34%\t7.885\n",
      "\t\tGT\t50.00%\t6.31%\t7.919\n",
      "\t\tTT\t50.00%\t6.41%\t7.799\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===============================================================\n",
      "\tTotal number of Oligos: 533\n",
      "\tAutoadjustment for sequence coverage in background: 1.08x\n",
      "\n",
      "\tOligos: 533 of 6418 max\n",
      "\tTree  : 1452 of 32090 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 533 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t========================================================================================\n",
      "\t\t82.36% skipped, 17.64% checked (94 of 533), of those checked:\n",
      "\t\t82.36% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 533 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=============================================================================================\n",
      "\t\t82.36% skipped, 17.64% checked (94 of 533), of those checked:\n",
      "\t\t0.00% not in target, 17.64% increased p-value, 0.19% high p-value\n",
      "\n",
      "\tScreening oligos 533 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=============================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 533), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GGGCAACG... (-1.609)\n",
      "\t\tRound 1: -1.61 GGGCAACG T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\tRound 2: -1.61 GGGCAACG T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\t=Final=: -1.38 GGGCAACG T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CGTGTCAC... (-1.609)\n",
      "\t\tRound 1: -1.61 CGTGTCAC T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\tRound 2: -1.61 CGTGTCAC T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\t=Final=: -1.38 CGTGTCAC T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TTGATGCT... (-1.609)\n",
      "\t\tRound 1: -1.61 TTGATGCT T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\tRound 2: -1.61 TTGATGCT T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\t=Final=: -1.38 TTGATGCT T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: CCTAAAAA... (-1.609)\n",
      "\t\tRound 1: -1.61 CCTAAAAA T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\tRound 2: -1.61 CCTAAAAA T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\t=Final=: -1.38 CCTAAAAA T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: AAAACTCT... (-4.828)\n",
      "\t\tRound 1: -1.61 CAAACGAT T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\tRound 2: -1.61 CAAACGAT T:1.0(100.00%),B:29.7(20.76%),P:1e0\n",
      "\t\t=Final=: -1.38 CAAACGAT T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.609 > -6.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs8\n",
      "\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===============================================================\n",
      "\tTotal number of Oligos: 526\n",
      "\tAutoadjustment for sequence coverage in background: 1.09x\n",
      "\n",
      "\tOligos: 526 of 6292 max\n",
      "\tTree  : 1484 of 31460 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 526 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=======================================================================================\n",
      "\t\t82.51% skipped, 17.49% checked (92 of 526), of those checked:\n",
      "\t\t82.51% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 526 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t82.51% skipped, 17.49% checked (92 of 526), of those checked:\n",
      "\t\t0.00% not in target, 17.49% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 526 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===========================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 526), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GGGCAACGTT... (-1.609)\n",
      "\t\tRound 1: -1.61 GGGCAACGTT T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\tRound 2: -1.61 GGGCAACGTT T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\t=Final=: -1.38 GGGCAACGTT T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CCGTGTCACT... (-1.609)\n",
      "\t\tRound 1: -1.61 CCGTGTCACT T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\tRound 2: -1.61 CCGTGTCACT T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\t=Final=: -1.38 CCGTGTCACT T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GATGCTCCTA... (-1.609)\n",
      "\t\tRound 1: -1.61 GATGCTCCTA T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\tRound 2: -1.61 GATGCTCCTA T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\t=Final=: -1.38 GATGCTCCTA T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: AAAAACTCTG... (-1.609)\n",
      "\t\tRound 1: -1.61 AAAAACTCTG T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\tRound 2: -1.61 AAAAACTCTG T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\t=Final=: -1.38 AAAAACTCTG T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: AAAACTCTGG... (-4.828)\n",
      "\t\tRound 1: -1.61 GCAATTCAGG T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\tRound 2: -1.61 GCAATTCAGG T:1.0(100.00%),B:29.8(20.69%),P:1e0\n",
      "\t\t=Final=: -1.38 GCAATTCAGG T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.609 > -6.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs10\n",
      "\n",
      "\n",
      "\t-blen automatically set to 2\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t===============================================================\n",
      "\tTotal number of Oligos: 517\n",
      "\tAutoadjustment for sequence coverage in background: 1.09x\n",
      "\n",
      "\tOligos: 517 of 6166 max\n",
      "\tTree  : 1452 of 30830 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 517 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t======================================================================================\n",
      "\t\t82.59% skipped, 17.41% checked (90 of 517), of those checked:\n",
      "\t\t82.59% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 517 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================\n",
      "\t\t82.59% skipped, 17.41% checked (90 of 517), of those checked:\n",
      "\t\t0.00% not in target, 17.41% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 517 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t=========================================================================================\n",
      "\t\t100.00% skipped, 0.00% checked (0 of 517), of those checked:\n",
      "\t\t0.00% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GGGCAACGTTGA... (-1.609)\n",
      "\t\tRound 1: -1.61 GGGCAACGTTGA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\tRound 2: -1.61 GGGCAACGTTGA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\t=Final=: -1.38 GGGCAACGTTGA T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CCCGTGTCACTT... (-1.609)\n",
      "\t\tRound 1: -1.61 CCCGTGTCACTT T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\tRound 2: -1.61 CCCGTGTCACTT T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\t=Final=: -1.38 CCCGTGTCACTT T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TGCTCCTAAAAA... (-1.609)\n",
      "\t\tRound 1: -1.61 TGCTCCTAAAAA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\tRound 2: -1.61 TGCTCCTAAAAA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\t=Final=: -1.38 TGCTCCTAAAAA T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: AAAACTCTGGCT... (-4.828)\n",
      "\t\tRound 1: 0.00 AAAACTCTGGCT T:0.0(0.00%),B:0.0(0.00%),P:1e0\n",
      "\t\t=Final=: 0.00 AAAACTCTGGCT T:0.0(0.00%),B:33.6(29.22%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ACTCTGGCTGGA... (-1.609)\n",
      "\t\tRound 1: -1.61 ACTCTGGCTGGA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\tRound 2: -1.61 ACTCTGGCTGGA T:1.0(100.00%),B:29.8(20.65%),P:1e0\n",
      "\t\t=Final=: -1.38 ACTCTGGCTGGA T:1.0(100.00%),B:29.8(25.91%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\tRemaining seeds don't look promising (After initial 5 motifs, logp -1.609 > -6.000)\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t63 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 5 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/example-data/homerMotifs.motifs12\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 20 reduced to 13 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 13 (1e-1) similar to RBP1-LIKE(RRM)/Drosophila_melanogaster-RNCMPT00127-PBM/HughesRNA(0.738)\n",
      "\t\t2 of 13 (1e-1) similar to pho/dmmpmm(Bergman)/fly(0.739)\n",
      "\t\t3 of 13 (1e0) similar to TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer(0.757)\n",
      "\t\t4 of 13 (1e0) similar to HNF6(Homeobox)/Liver-Hnf6-ChIP-Seq(ERP000394)/Homer(0.775)\n",
      "\t\t5 of 13 (1e0) similar to nub/dmmpmm(Pollard)/fly(0.739)\n",
      "\t\t6 of 13 (1e0) similar to WRKY12/MA1075.1/Jaspar(0.710)\n",
      "\t\t7 of 13 (1e0) similar to Six4/MA0204.1/Jaspar(0.715)\n",
      "\t\t8 of 13 (1e0) similar to SRSF9(RRM)/Homo_sapiens-RNCMPT00074-PBM/HughesRNA(0.791)\n",
      "\t\t9 of 13 (1e0) similar to MET31/Literature(Harbison)/Yeast(0.703)\n",
      "\t\t10 of 13 (1e0) similar to Hmx1/MA0896.1/Jaspar(0.704)\n",
      "\t\t11 of 13 (1e0) similar to BAS1/MA0278.1/Jaspar(0.640)\n",
      "\t\t12 of 13 (1e0) similar to gt/dmmpmm(SeSiMCMC)/fly(0.845)\n",
      "\t\t13 of 13 (1e0) similar to PH0044.1_Homez/Jaspar(0.806)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "memusg: vmpeak: 167048 kb\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Testing findMotifsGenome.pl memory usage\n",
    "./memusg findMotifsGenome.pl example-files/peaks.bed example-files/ref.fa homer-results/example-data > /dev/null"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bba3f54",
   "metadata": {},
   "source": [
    "### Testing Memory Usage with ISL1 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "39cd3b09",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "memusg: vmpeak: 17966576 kb\n",
      "\n",
      "\tPosition file = /home/esquan/final_project/pyxis/analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed\n",
      "\tGenome = /home/esquan/final_project/pyxis/benchmark/hg19.fa\n",
      "\tOutput Directory = homer-results/isl1\n",
      "\tUsing Custom Genome\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 6184\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 6184\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground files for 200 bp fragments found.\n",
      "\tCustom genome sequence file: /home/esquan/final_project/pyxis/benchmark/hg19.fa\n",
      "\n",
      "\tExtracting sequences from file: /home/esquan/final_project/pyxis/benchmark/hg19.fa\n",
      "\tLooking for peak sequences in a single file (/home/esquan/final_project/pyxis/benchmark/hg19.fa)\n",
      "\tExtracting 566 sequences from chr1\n",
      "\tExtracting 546 sequences from chr2\n",
      "\tExtracting 422 sequences from chr3\n",
      "\tExtracting 424 sequences from chr4\n",
      "\tExtracting 423 sequences from chr5\n",
      "\tExtracting 377 sequences from chr6\n",
      "\tExtracting 335 sequences from chr7\n",
      "\tExtracting 95 sequences from chrX\n",
      "\tExtracting 318 sequences from chr8\n",
      "\tExtracting 267 sequences from chr9\n",
      "\tExtracting 330 sequences from chr10\n",
      "\tExtracting 273 sequences from chr11\n",
      "\tExtracting 328 sequences from chr12\n",
      "\tExtracting 220 sequences from chr13\n",
      "\tExtracting 193 sequences from chr14\n",
      "\tExtracting 201 sequences from chr15\n",
      "\tExtracting 167 sequences from chr16\n",
      "\tExtracting 146 sequences from chr17\n",
      "\tExtracting 203 sequences from chr18\n",
      "\tExtracting 136 sequences from chr20\n",
      "\tExtracting 12 sequences from chrY\n",
      "\tExtracting 63 sequences from chr19\n",
      "\tExtracting 60 sequences from chr22\n",
      "\tExtracting 54 sequences from chr21\n",
      "\tExtracting 1 sequences from chrUn_gl000212\n",
      "\tExtracting 2 sequences from chr7_gl000195_random\n",
      "\tExtracting 1 sequences from chrUn_gl000224\n",
      "\tExtracting 1 sequences from chr17_gl000205_random\n",
      "\tExtracting 2 sequences from chr9_gl000199_random\n",
      "\tExtracting 10 sequences from chrUn_gl000220\n",
      "\tExtracting 1 sequences from chr19_gl000208_random\n",
      "\tExtracting 6 sequences from chrM\n",
      "\tExtracting 1 sequences from chrUn_gl000226\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 201 bp\n",
      "\t\t6184 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.2\t0\t2\n",
      "\t0.25\t1\t5\n",
      "\t0.3\t2\t14\n",
      "\t0.35\t3\t203\n",
      "\t0.4\t4\t1108\n",
      "\t0.45\t5\t2102\n",
      "\t0.5\t6\t1713\n",
      "\t0.6\t7\t951\n",
      "\t0.7\t8\t75\n",
      "\t0.8\t9\t9\n",
      "\t10\t10\t2\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t0\t2\t14\t1.012\n",
      "\t1\t5\t35\t1.012\n",
      "\t2\t14\t99\t1.002\n",
      "\t3\t203\t1438\t1.000\n",
      "\t4\t1108\t7851\t1.000\n",
      "\t5\t2102\t14893\t1.000\n",
      "\t6\t1713\t12137\t1.000\n",
      "\t7\t951\t6738\t1.000\n",
      "\t8\t75\t531\t1.001\n",
      "\t9\t9\t64\t0.996\n",
      "\t10\t2\t14\t1.012\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.69%\t0.999\n",
      "\t\tC\t22.34%\t22.31%\t1.001\n",
      "\t\tG\t22.34%\t22.31%\t1.001\n",
      "\t\tT\t27.66%\t27.69%\t0.999\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.51%\t0.968\n",
      "\t\tCA\t7.50%\t7.64%\t0.982\n",
      "\t\tGA\t6.42%\t6.23%\t1.031\n",
      "\t\tTA\t5.50%\t5.30%\t1.038\n",
      "\t\tAC\t5.10%\t5.16%\t0.987\n",
      "\t\tCC\t5.59%\t6.05%\t0.923\n",
      "\t\tGC\t5.24%\t4.87%\t1.075\n",
      "\t\tTC\t6.42%\t6.23%\t1.031\n",
      "\t\tAG\t8.28%\t7.50%\t1.104\n",
      "\t\tCG\t0.97%\t1.12%\t0.870\n",
      "\t\tGG\t5.59%\t6.05%\t0.923\n",
      "\t\tTG\t7.50%\t7.64%\t0.982\n",
      "\t\tAT\t6.04%\t6.51%\t0.928\n",
      "\t\tCT\t8.28%\t7.50%\t1.104\n",
      "\t\tGT\t5.10%\t5.16%\t0.987\n",
      "\t\tTT\t8.24%\t8.51%\t0.968\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: homer-results/isl1/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t27.66%\t27.66%\t1.000\n",
      "\t\tC\t22.34%\t22.34%\t1.000\n",
      "\t\tG\t22.34%\t22.34%\t1.000\n",
      "\t\tT\t27.66%\t27.66%\t1.000\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t8.24%\t8.32%\t0.990\n",
      "\t\tCA\t7.50%\t7.57%\t0.991\n",
      "\t\tGA\t6.42%\t6.39%\t1.004\n",
      "\t\tTA\t5.50%\t5.38%\t1.022\n",
      "\t\tAC\t5.10%\t5.11%\t0.997\n",
      "\t\tCC\t5.59%\t5.77%\t0.968\n",
      "\t\tGC\t5.24%\t5.07%\t1.034\n",
      "\t\tTC\t6.42%\t6.39%\t1.004\n",
      "\t\tAG\t8.28%\t8.01%\t1.034\n",
      "\t\tCG\t0.97%\t0.99%\t0.982\n",
      "\t\tGG\t5.59%\t5.77%\t0.968\n",
      "\t\tTG\t7.50%\t7.57%\t0.991\n",
      "\t\tAT\t6.04%\t6.22%\t0.971\n",
      "\t\tCT\t8.28%\t8.01%\t1.034\n",
      "\t\tGT\t5.10%\t5.11%\t0.997\n",
      "\t\tTT\t8.24%\t8.32%\t0.990\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\t1006 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 1006 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t\t1 of 1006 (1e-1149) Gata6(Zf)/HUG1N-GATA6-ChIP-Seq(GSE51936)/Homer\n",
      "\t\t2 of 1006 (1e-1143) Gata1(Zf)/K562-GATA1-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t3 of 1006 (1e-1141) Gata2(Zf)/K562-GATA2-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t4 of 1006 (1e-1140) Gata4(Zf)/Heart-Gata4-ChIP-Seq(GSE35151)/Homer\n",
      "\t\t5 of 1006 (1e-1027) GATA3(Zf)/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t6 of 1006 (1e-908) TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer\n",
      "\t\t7 of 1006 (1e-864) Isl1(Homeobox)/Neuron-Isl1-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t8 of 1006 (1e-757) At5g05790(MYBrelated)/col-At5g05790-DAP-Seq(GSE60143)/Homer\n",
      "\t\t9 of 1006 (1e-734) At3g11280(MYBrelated)/col-At3g11280-DAP-Seq(GSE60143)/Homer\n",
      "\t\t10 of 1006 (1e-607) At1g49010(MYBrelated)/col-At1g49010-DAP-Seq(GSE60143)/Homer\n",
      "\t\t11 of 1006 (1e-542) At5g58900(MYBrelated)/colamp-At5g58900-DAP-Seq(GSE60143)/Homer\n",
      "\t\t12 of 1006 (1e-539) At5g08520(MYBrelated)/colamp-At5g08520-DAP-Seq(GSE60143)/Homer\n",
      "\t\t13 of 1006 (1e-477) ELT-3(Gata)/cElegans-L1-ELT3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t14 of 1006 (1e-474) PQM-1(?)/cElegans-L3-ChIP-Seq(modEncode)/Homer\n",
      "\t\t15 of 1006 (1e-415) Nkx2.5(Homeobox)/HL1-Nkx2.5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t16 of 1006 (1e-404) TEAD1(TEAD)/HepG2-TEAD1-ChIP-Seq(Encode)/Homer\n",
      "\t\t17 of 1006 (1e-402) TEAD3(TEA)/HepG2-TEAD3-ChIP-Seq(Encode)/Homer\n",
      "\t\t18 of 1006 (1e-391) TEAD4(TEA)/Tropoblast-Tead4-ChIP-Seq(GSE37350)/Homer\n",
      "\t\t19 of 1006 (1e-349) TEAD(TEA)/Fibroblast-PU.1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t20 of 1006 (1e-345) Nkx3.1(Homeobox)/LNCaP-Nkx3.1-ChIP-Seq(GSE28264)/Homer\n",
      "\t\t21 of 1006 (1e-338) Bapx1(Homeobox)/VertebralCol-Bapx1-ChIP-Seq(GSE36672)/Homer\n",
      "\t\t22 of 1006 (1e-302) Hand2(bHLH)/Mesoderm-Hand2-ChIP-Seq(GSE61475)/Homer\n",
      "\t\t23 of 1006 (1e-294) AT5G56840(MYBrelated)/colamp-AT5G56840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t24 of 1006 (1e-272) Nkx2.1(Homeobox)/LungAC-Nkx2.1-ChIP-Seq(GSE43252)/Homer\n",
      "\t\t25 of 1006 (1e-257) Unknown5/Drosophila-Promoters/Homer\n",
      "\t\t26 of 1006 (1e-246) At5g47390(MYBrelated)/col-At5g47390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t27 of 1006 (1e-238) TEAD2(TEA)/Py2T-Tead2-ChIP-Seq(GSE55709)/Homer\n",
      "\t\t28 of 1006 (1e-220) AT5G61620(MYBrelated)/colamp-AT5G61620-DAP-Seq(GSE60143)/Homer\n",
      "\t\t29 of 1006 (1e-219) Nkx6.1(Homeobox)/Islet-Nkx6.1-ChIP-Seq(GSE40975)/Homer\n",
      "\t\t30 of 1006 (1e-209) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
      "\t\t31 of 1006 (1e-192) Tbx20(T-box)/Heart-Tbx20-ChIP-Seq(GSE29636)/Homer\n",
      "\t\t32 of 1006 (1e-165) Tgif1(Homeobox)/mES-Tgif1-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t33 of 1006 (1e-163) En1(Homeobox)/SUM149-EN1-ChIP-Seq(GSE120957)/Homer\n",
      "\t\t34 of 1006 (1e-150) Lhx3(Homeobox)/Neuron-Lhx3-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t35 of 1006 (1e-145) Meis1(Homeobox)/MastCells-Meis1-ChIP-Seq(GSE48085)/Homer\n",
      "\t\t36 of 1006 (1e-136) Nanog(Homeobox)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t37 of 1006 (1e-130) Tgif2(Homeobox)/mES-Tgif2-ChIP-Seq(GSE55404)/Homer\n",
      "\t\t38 of 1006 (1e-115) DLX2(Homeobox)/BasalGanglia-Dlx2-ChIP-seq(GSE124936)/Homer\n",
      "\t\t39 of 1006 (1e-114) At1g19000(MYBrelated)/colamp-At1g19000-DAP-Seq(GSE60143)/Homer\n",
      "\t\t40 of 1006 (1e-111) Lhx2(Homeobox)/HFSC-Lhx2-ChIP-Seq(GSE48068)/Homer\n",
      "\t\t41 of 1006 (1e-108) ATHB33(ZFHD)/col-ATHB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t42 of 1006 (1e-106) LHX9(Homeobox)/Hct116-LHX9.V5-ChIP-Seq(GSE116822)/Homer\n",
      "\t\t43 of 1006 (1e-105) Eomes(T-box)/H9-Eomes-ChIP-Seq(GSE26097)/Homer\n",
      "\t\t44 of 1006 (1e-95) AT4G00250(GeBP)/col-AT4G00250-DAP-Seq(GSE60143)/Homer\n",
      "\t\t45 of 1006 (1e-90) DLX5(Homeobox)/BasalGanglia-Dlx5-ChIP-seq(GSE124936)/Homer\n",
      "\t\t46 of 1006 (1e-82) Lhx1(Homeobox)/EmbryoCarcinoma-Lhx1-ChIP-Seq(GSE70957)/Homer\n",
      "\t\t47 of 1006 (1e-77) Barx1(Homeobox)/Stomach-Barx1.3xFlag-ChIP-Seq(GSE69483)/Homer\n",
      "\t\t48 of 1006 (1e-69) DLX1(Homeobox)/BasalGanglia-Dlx1-ChIP-seq(GSE124936)/Homer\n",
      "\t\t49 of 1006 (1e-68) Otx2(Homeobox)/EpiLC-Otx2-ChIP-Seq(GSE56098)/Homer\n",
      "\t\t50 of 1006 (1e-65) At1g74840(MYBrelated)/col100-At1g74840-DAP-Seq(GSE60143)/Homer\n",
      "\t\t51 of 1006 (1e-64) AtGRF6(GRF)/col-AtGRF6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t52 of 1006 (1e-63) Dlx3(Homeobox)/Kerainocytes-Dlx3-ChIP-Seq(GSE89884)/Homer\n",
      "\t\t53 of 1006 (1e-62) bcd(Homeobox)/Embryo-Bcd-ChIP-Seq(GSE86966)/Homer\n",
      "\t\t54 of 1006 (1e-59) GRF9(GRF)/colamp-GRF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t55 of 1006 (1e-56) Foxo3(Forkhead)/U2OS-Foxo3-ChIP-Seq(E-MTAB-2701)/Homer\n",
      "\t\t56 of 1006 (1e-55) Foxa2(Forkhead)/Liver-Foxa2-ChIP-Seq(GSE25694)/Homer\n",
      "\t\t57 of 1006 (1e-53) ATHB23(ZFHD)/col-ATHB23-DAP-Seq(GSE60143)/Homer\n",
      "\t\t58 of 1006 (1e-53) Pdx1(Homeobox)/Islet-Pdx1-ChIP-Seq(SRA008281)/Homer\n",
      "\t\t59 of 1006 (1e-53) GATA:SCL(Zf,bHLH)/Ter119-SCL-ChIP-Seq(GSE18720)/Homer\n",
      "\t\t60 of 1006 (1e-52) ATHB34(ZFHD)/colamp-ATHB34-DAP-Seq(GSE60143)/Homer\n",
      "\t\t61 of 1006 (1e-52) Foxa3(Forkhead)/Liver-Foxa3-ChIP-Seq(GSE77670)/Homer\n",
      "\t\t62 of 1006 (1e-48) CRX(Homeobox)/Retina-Crx-ChIP-Seq(GSE20012)/Homer\n",
      "\t\t63 of 1006 (1e-47) Hoxb4(Homeobox)/ES-Hoxb4-ChIP-Seq(GSE34014)/Homer\n",
      "\t\t64 of 1006 (1e-45) ATHB25(ZFHD)/colamp-ATHB25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t65 of 1006 (1e-44) MYB101(MYB)/colamp-MYB101-DAP-Seq(GSE60143)/Homer\n",
      "\t\t66 of 1006 (1e-44) ATHB24(ZFHD)/colamp-ATHB24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t67 of 1006 (1e-42) MafA(bZIP)/Islet-MafA-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t68 of 1006 (1e-42) Foxo1(Forkhead)/RAW-Foxo1-ChIP-Seq(Fan_et_al.)/Homer\n",
      "\t\t69 of 1006 (1e-40) AT2G38300(G2like)/col-AT2G38300-DAP-Seq(GSE60143)/Homer\n",
      "\t\t70 of 1006 (1e-36) FOXA1(Forkhead)/LNCAP-FOXA1-ChIP-Seq(GSE27824)/Homer\n",
      "\t\t71 of 1006 (1e-35) FOXP1(Forkhead)/H9-FOXP1-ChIP-Seq(GSE31006)/Homer\n",
      "\t\t72 of 1006 (1e-35) AT3G10030(Trihelix)/colamp-AT3G10030-DAP-Seq(GSE60143)/Homer\n",
      "\t\t73 of 1006 (1e-35) FOXK2(Forkhead)/U2OS-FOXK2-ChIP-Seq(E-MTAB-2204)/Homer\n",
      "\t\t74 of 1006 (1e-34) FOXK1(Forkhead)/HEK293-FOXK1-ChIP-Seq(GSE51673)/Homer\n",
      "\t\t75 of 1006 (1e-33) bZIP52(bZIP)/colamp-bZIP52-DAP-Seq(GSE60143)/Homer\n",
      "\t\t76 of 1006 (1e-33) MYB65(MYB)/colamp-MYB65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t77 of 1006 (1e-33) AT2G40260(G2like)/colamp-AT2G40260-DAP-Seq(GSE60143)/Homer\n",
      "\t\t78 of 1006 (1e-32) FoxL2(Forkhead)/Ovary-FoxL2-ChIP-Seq(GSE60858)/Homer\n",
      "\t\t79 of 1006 (1e-32) FOXA1(Forkhead)/MCF7-FOXA1-ChIP-Seq(GSE26831)/Homer\n",
      "\t\t80 of 1006 (1e-32) Tbr1(T-box)/Cortex-Tbr1-ChIP-Seq(GSE71384)/Homer\n",
      "\t\t81 of 1006 (1e-31) At5g04390(C2H2)/col200-At5g04390-DAP-Seq(GSE60143)/Homer\n",
      "\t\t82 of 1006 (1e-30) Replumless(BLH)/Arabidopsis-RPL.GFP-ChIP-Seq(GSE78727)/Homer\n",
      "\t\t83 of 1006 (1e-30) Fox:Ebox(Forkhead,bHLH)/Panc1-Foxa2-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t84 of 1006 (1e-29) BMYB(HTH)/Hela-BMYB-ChIP-Seq(GSE27030)/Homer\n",
      "\t\t85 of 1006 (1e-28) HOXA2(Homeobox)/mES-Hoxa2-ChIP-Seq(Donaldson_et_al.)/Homer\n",
      "\t\t86 of 1006 (1e-27) HOXA1(Homeobox)/mES-Hoxa1-ChIP-Seq(SRP084292)/Homer\n",
      "\t\t87 of 1006 (1e-27) Mef2d(MADS)/Retina-Mef2d-ChIP-Seq(GSE61391)/Homer\n",
      "\t\t88 of 1006 (1e-26) FOXM1(Forkhead)/MCF7-FOXM1-ChIP-Seq(GSE72977)/Homer\n",
      "\t\t89 of 1006 (1e-25) PBX2(Homeobox)/K562-PBX2-ChIP-Seq(Encode)/Homer\n",
      "\t\t90 of 1006 (1e-24) Ap4(bHLH)/AML-Tfap4-ChIP-Seq(GSE45738)/Homer\n",
      "\t\t91 of 1006 (1e-24) Pbx3(Homeobox)/GM12878-PBX3-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t92 of 1006 (1e-24) STZ(C2H2)/colamp-STZ-DAP-Seq(GSE60143)/Homer\n",
      "\t\t93 of 1006 (1e-23) bZIP18(bZIP)/colamp-bZIP18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t94 of 1006 (1e-23) MYB70(MYB)/col-MYB70-DAP-Seq(GSE60143)/Homer\n",
      "\t\t95 of 1006 (1e-21) AMYB(HTH)/Testes-AMYB-ChIP-Seq(GSE44588)/Homer\n",
      "\t\t96 of 1006 (1e-20) Pknox1(Homeobox)/ES-Prep1-ChIP-Seq(GSE63282)/Homer\n",
      "\t\t97 of 1006 (1e-20) Tbx6(T-box)/ESC-Tbx6-ChIP-Seq(GSE93524)/Homer\n",
      "\t\t98 of 1006 (1e-19) Myf5(bHLH)/GM-Myf5-ChIP-Seq(GSE24852)/Homer\n",
      "\t\t99 of 1006 (1e-19) SCL(bHLH)/HPC7-Scl-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t100 of 1006 (1e-19) ZNF264(Zf)/HEK293-ZNF264.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t101 of 1006 (1e-19) NLP7(RWPRK)/col-NLP7-DAP-Seq(GSE60143)/Homer\n",
      "\t\t102 of 1006 (1e-18) Nur77(NR)/K562-NR4A1-ChIP-Seq(GSE31363)/Homer\n",
      "\t\t103 of 1006 (1e-18) MYB(HTH)/ERMYB-Myb-ChIPSeq(GSE22095)/Homer\n",
      "\t\t104 of 1006 (1e-18) TATA-Box(TBP)/Promoter/Homer\n",
      "\t\t105 of 1006 (1e-17) MyoG(bHLH)/C2C12-MyoG-ChIP-Seq(GSE36024)/Homer\n",
      "\t\t106 of 1006 (1e-17) AZF1(C2H2)/colamp-AZF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t107 of 1006 (1e-17) LIN-39(Homeobox)/cElegans.L3-LIN39-ChIP-Seq(modEncode)/Homer\n",
      "\t\t108 of 1006 (1e-17) Hoxa9(Homeobox)/ChickenMSG-Hoxa9.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t109 of 1006 (1e-17) HLH-1(bHLH)/cElegans-Embryo-HLH1-ChIP-Seq(modEncode)/Homer\n",
      "\t\t110 of 1006 (1e-16) E-box/Drosophila-Promoters/Homer\n",
      "\t\t111 of 1006 (1e-16) Tcf21(bHLH)/ArterySmoothMuscle-Tcf21-ChIP-Seq(GSE61369)/Homer\n",
      "\t\t112 of 1006 (1e-16) Foxh1(Forkhead)/hESC-FOXH1-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t113 of 1006 (1e-15) Foxf1(Forkhead)/Lung-Foxf1-ChIP-Seq(GSE77951)/Homer\n",
      "\t\t114 of 1006 (1e-15) CELF2(RRM)/JSL1-CELF2-CLIP-Seq(GSE71264)/Homer\n",
      "\t\t115 of 1006 (1e-15) LEF1(HMG)/H1-LEF1-ChIP-Seq(GSE64758)/Homer\n",
      "\t\t116 of 1006 (1e-15) VND6(NAC)/col-VND6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t117 of 1006 (1e-14) FoxD3(forkhead)/ZebrafishEmbryo-Foxd3.biotin-ChIP-seq(GSE106676)/Homer\n",
      "\t\t118 of 1006 (1e-14) MyoD(bHLH)/Myotube-MyoD-ChIP-Seq(GSE21614)/Homer\n",
      "\t\t119 of 1006 (1e-14) ZFX(Zf)/mES-Zfx-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t120 of 1006 (1e-14) Tcf4(HMG)/Hct116-Tcf4-ChIP-Seq(SRA012054)/Homer\n",
      "\t\t121 of 1006 (1e-14) CTCF(Zf)/CD4+-CTCF-ChIP-Seq(Barski_et_al.)/Homer\n",
      "\t\t122 of 1006 (1e-14) Tbx5(T-box)/HL1-Tbx5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t123 of 1006 (1e-14) MYB77(MYB)/col-MYB77-DAP-Seq(GSE60143)/Homer\n",
      "\t\t124 of 1006 (1e-14) Atoh1(bHLH)/Cerebellum-Atoh1-ChIP-Seq(GSE22111)/Homer\n",
      "\t\t125 of 1006 (1e-13) NTM1(NAC)/col-NTM1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t126 of 1006 (1e-12) Smad2(MAD)/ES-SMAD2-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t127 of 1006 (1e-12) Smad4(MAD)/ESC-SMAD4-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t128 of 1006 (1e-12) Brachyury(T-box)/Mesoendoderm-Brachyury-ChIP-exo(GSE54963)/Homer\n",
      "\t\t129 of 1006 (1e-12) AT1G69570(C2C2dof)/col-AT1G69570-DAP-Seq(GSE60143)/Homer\n",
      "\t\t130 of 1006 (1e-11) ZNF652/HepG2-ZNF652.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t131 of 1006 (1e-11) ANAC045(NAC)/col-ANAC045-DAP-Seq(GSE60143)/Homer\n",
      "\t\t132 of 1006 (1e-11) VIP1(bZIP)/col-VIP1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t133 of 1006 (1e-11) CDF3(C2C2dof)/colamp-CDF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t134 of 1006 (1e-11) RAP211(AP2EREBP)/colamp-RAP211-DAP-Seq(GSE60143)/Homer\n",
      "\t\t135 of 1006 (1e-11) CArG(MADS)/PUER-Srf-ChIP-Seq(Sullivan_et_al.)/Homer\n",
      "\t\t136 of 1006 (1e-11) MYB88(MYB)/col-MYB88-DAP-Seq(GSE60143)/Homer\n",
      "\t\t137 of 1006 (1e-10) WRKY28(WRKY)/col-WRKY28-DAP-Seq(GSE60143)/Homer\n",
      "\t\t138 of 1006 (1e-10) Rap210(AP2EREBP)/col-Rap210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t139 of 1006 (1e-10) GSC(Homeobox)/FrogEmbryos-GSC-ChIP-Seq(DRA000576)/Homer\n",
      "\t\t140 of 1006 (1e-10) ERF13(AP2EREBP)/colamp-ERF13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t141 of 1006 (1e-10) VND2(NAC)/col-VND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t142 of 1006 (1e-10) SND3(NAC)/col-SND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t143 of 1006 (1e-10) Bcl6(Zf)/Liver-Bcl6-ChIP-Seq(GSE31578)/Homer\n",
      "\t\t144 of 1006 (1e-10) Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer\n",
      "\t\t145 of 1006 (1e-10) MYB73(MYB)/col-MYB73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t146 of 1006 (1e-10) Tcf3(HMG)/mES-Tcf3-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t147 of 1006 (1e-9) Rfx2(HTH)/LoVo-RFX2-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t148 of 1006 (1e-9) MYB56(MYB)/colamp-MYB56-DAP-Seq(GSE60143)/Homer\n",
      "\t\t149 of 1006 (1e-9) WRKY42(WRKY)/colamp-WRKY42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t150 of 1006 (1e-9) NeuroD1(bHLH)/Islet-NeuroD1-ChIP-Seq(GSE30298)/Homer\n",
      "\t\t151 of 1006 (1e-9) GATA3(Zf),DR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t152 of 1006 (1e-9) Tcf12(bHLH)/GM12878-Tcf12-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t153 of 1006 (1e-9) KAN2(G2like)/colamp-KAN2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t154 of 1006 (1e-9) Pitx1(Homeobox)/Chicken-Pitx1-ChIP-Seq(GSE38910)/Homer\n",
      "\t\t155 of 1006 (1e-9) ERF115(AP2EREBP)/colamp-ERF115-DAP-Seq(GSE60143)/Homer\n",
      "\t\t156 of 1006 (1e-9) TR4(NR),DR1/Hela-TR4-ChIP-Seq(GSE24685)/Homer\n",
      "\t\t157 of 1006 (1e-9) NeuroG2(bHLH)/Fibroblast-NeuroG2-ChIP-Seq(GSE75910)/Homer\n",
      "\t\t158 of 1006 (1e-8) SMB(NAC)/colamp-SMB-DAP-Seq(GSE60143)/Homer\n",
      "\t\t159 of 1006 (1e-8) NF1-halfsite(CTF)/LNCaP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t160 of 1006 (1e-8) WUS1(Homeobox)/colamp-WUS1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t161 of 1006 (1e-8) bZIP50(bZIP)/colamp-bZIP50-DAP-Seq(GSE60143)/Homer\n",
      "\t\t162 of 1006 (1e-8) MYB81(MYB)/col-MYB81-DAP-Seq(GSE60143)/Homer\n",
      "\t\t163 of 1006 (1e-8) PBX1(Homeobox)/MCF7-PBX1-ChIP-Seq(GSE28007)/Homer\n",
      "\t\t164 of 1006 (1e-8) Tlx?(NR)/NPC-H3K4me1-ChIP-Seq(GSE16256)/Homer\n",
      "\t\t165 of 1006 (1e-8) bHLH80(bHLH)/col-bHLH80-DAP-Seq(GSE60143)/Homer\n",
      "\t\t166 of 1006 (1e-8) NTM2(NAC)/col-NTM2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t167 of 1006 (1e-8) ANAC071(NAC)/col-ANAC071-DAP-Seq(GSE60143)/Homer\n",
      "\t\t168 of 1006 (1e-8) bHLH122(bHLH)/col100-bHLH122-DAP-Seq(GSE60143)/Homer\n",
      "\t\t169 of 1006 (1e-8) ANAC005(NAC)/col-ANAC005-DAP-Seq(GSE60143)/Homer\n",
      "\t\t170 of 1006 (1e-7) RFX(HTH)/K562-RFX3-ChIP-Seq(SRA012198)/Homer\n",
      "\t\t171 of 1006 (1e-7) Zic(Zf)/Cerebellum-ZIC1.2-ChIP-Seq(GSE60731)/Homer\n",
      "\t\t172 of 1006 (1e-7) ERF105(AP2EREBP)/colamp-ERF105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t173 of 1006 (1e-7) VND4(NAC)/colamp-VND4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t174 of 1006 (1e-7) ANAC057(NAC)/colamp-ANAC057-DAP-Seq(GSE60143)/Homer\n",
      "\t\t175 of 1006 (1e-7) AT2G33550(Trihelix)/colamp-AT2G33550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t176 of 1006 (1e-7) PPARa(NR),DR1/Liver-Ppara-ChIP-Seq(GSE47954)/Homer\n",
      "\t\t177 of 1006 (1e-7) bZIP69(bZIP)/col-bZIP69-DAP-Seq(GSE60143)/Homer\n",
      "\t\t178 of 1006 (1e-7) WRKY29(WRKY)/colamp-WRKY29-DAP-Seq(GSE60143)/Homer\n",
      "\t\t179 of 1006 (1e-7) ERF104(AP2EREBP)/col-ERF104-DAP-Seq(GSE60143)/Homer\n",
      "\t\t180 of 1006 (1e-7) ANAC058(NAC)/col-ANAC058-DAP-Seq(GSE60143)/Homer\n",
      "\t\t181 of 1006 (1e-7) Knotted(Homeobox)/Corn-KN1-ChIP-Seq(GSE39161)/Homer\n",
      "\t\t182 of 1006 (1e-7) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
      "\t\t183 of 1006 (1e-7) ANAC004(NAC)/colamp-ANAC004-DAP-Seq(GSE60143)/Homer\n",
      "\t\t184 of 1006 (1e-7) RXR(NR),DR1/3T3L1-RXR-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t185 of 1006 (1e-7) Rbpj1(?)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t186 of 1006 (1e-7) NAC2(NAC)/colamp-NAC2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t187 of 1006 (1e-6) PHA-4(Forkhead)/cElegans-Embryos-PHA4-ChIP-Seq(modEncode)/Homer\n",
      "\t\t188 of 1006 (1e-6) AT3G25990(Trihelix)/colamp-AT3G25990-DAP-Seq(GSE60143)/Homer\n",
      "\t\t189 of 1006 (1e-6) TAGL1(MADS)/Tomato-TAGL1-ChIP-Seq(GSE116581)/Homer\n",
      "\t\t190 of 1006 (1e-6) Fos(bZIP)/TSC-Fos-ChIP-Seq(GSE110950)/Homer\n",
      "\t\t191 of 1006 (1e-6) Fra1(bZIP)/BT549-Fra1-ChIP-Seq(GSE46166)/Homer\n",
      "\t\t192 of 1006 (1e-6) Tbet(T-box)/CD8-Tbet-ChIP-Seq(GSE33802)/Homer\n",
      "\t\t193 of 1006 (1e-6) Smad3(MAD)/NPC-Smad3-ChIP-Seq(GSE36673)/Homer\n",
      "\t\t194 of 1006 (1e-6) BHLHA15(bHLH)/NIH3T3-BHLHB8.HA-ChIP-Seq(GSE119782)/Homer\n",
      "\t\t195 of 1006 (1e-6) Six1(Homeobox)/Myoblast-Six1-ChIP-Chip(GSE20150)/Homer\n",
      "\t\t196 of 1006 (1e-6) HNF4a(NR),DR1/HepG2-HNF4a-ChIP-Seq(GSE25021)/Homer\n",
      "\t\t197 of 1006 (1e-6) Tbx21(T-box)/GM12878-TBX21-ChIP-Seq(Encode)/Homer\n",
      "\t\t198 of 1006 (1e-6) VND1(NAC)/col-VND1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t199 of 1006 (1e-6) ANAC062(NAC)/colamp-ANAC062-DAP-Seq(GSE60143)/Homer\n",
      "\t\t200 of 1006 (1e-6) TCFL2(HMG)/K562-TCF7L2-ChIP-Seq(GSE29196)/Homer\n",
      "\t\t201 of 1006 (1e-6) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t202 of 1006 (1e-6) VND3(NAC)/colamp-VND3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t203 of 1006 (1e-6) CBF1(AP2EREBP)/colamp-CBF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t204 of 1006 (1e-6) SOC1(MADS)/Seedling-SOC1-ChIP-Seq(GSE45846)/Homer\n",
      "\t\t205 of 1006 (1e-6) BORIS(Zf)/K562-CTCFL-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t206 of 1006 (1e-6) Zic2(Zf)/ESC-Zic2-ChIP-Seq(SRP197560)/Homer\n",
      "\t\t207 of 1006 (1e-5) CRF10(AP2EREBP)/col100-CRF10-DAP-Seq(GSE60143)/Homer\n",
      "\t\t208 of 1006 (1e-5) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t209 of 1006 (1e-5) At1g19210(AP2EREBP)/colamp-At1g19210-DAP-Seq(GSE60143)/Homer\n",
      "\t\t210 of 1006 (1e-5) Rfx6(HTH)/Min6b1-Rfx6.HA-ChIP-Seq(GSE62844)/Homer\n",
      "\t\t211 of 1006 (1e-5) At3g60580(C2H2)/col-At3g60580-DAP-Seq(GSE60143)/Homer\n",
      "\t\t212 of 1006 (1e-5) ANAC028(NAC)/col-ANAC028-DAP-Seq(GSE60143)/Homer\n",
      "\t\t213 of 1006 (1e-5) ERF73(AP2EREBP)/col-ERF73-DAP-Seq(GSE60143)/Homer\n",
      "\t\t214 of 1006 (1e-5) ANAC096(NAC)/colamp-ANAC096-DAP-Seq(GSE60143)/Homer\n",
      "\t\t215 of 1006 (1e-5) Zic3(Zf)/mES-Zic3-ChIP-Seq(GSE37889)/Homer\n",
      "\t\t216 of 1006 (1e-5) WRKY27(WRKY)/colamp-WRKY27-DAP-Seq(GSE60143)/Homer\n",
      "\t\t217 of 1006 (1e-5) At4g16750(AP2EREBP)/col-At4g16750-DAP-Seq(GSE60143)/Homer\n",
      "\t\t218 of 1006 (1e-5) MYB105(MYB)/colamp-MYB105-DAP-Seq(GSE60143)/Homer\n",
      "\t\t219 of 1006 (1e-5) NFAT(RHD)/Jurkat-NFATC1-ChIP-Seq(Jolma_et_al.)/Homer\n",
      "\t\t220 of 1006 (1e-5) Hoxc9(Homeobox)/Ainv15-Hoxc9-ChIP-Seq(GSE21812)/Homer\n",
      "\t\t221 of 1006 (1e-5) CBF3(AP2EREBP)/colamp-CBF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t222 of 1006 (1e-5) ANAC083(NAC)/col-ANAC083-DAP-Seq(GSE60143)/Homer\n",
      "\t\t223 of 1006 (1e-5) ANAC070(NAC)/colamp-ANAC070-DAP-Seq(GSE60143)/Homer\n",
      "\t\t224 of 1006 (1e-5) CBF4(AP2EREBP)/colamp-CBF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t225 of 1006 (1e-5) ERF11(AP2EREBP)/col-ERF11-DAP-Seq(GSE60143)/Homer\n",
      "\t\t226 of 1006 (1e-5) TCF4(bHLH)/SHSY5Y-TCF4-ChIP-Seq(GSE96915)/Homer\n",
      "\t\t227 of 1006 (1e-5) SVP(MADS)/col-SVP-DAP-Seq(GSE60143)/Homer\n",
      "\t\t228 of 1006 (1e-5) Unknown-ESC-element(?)/mES-Nanog-ChIP-Seq(GSE11724)/Homer\n",
      "\t\t229 of 1006 (1e-5) CBF2(AP2EREBP)/colamp-CBF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t230 of 1006 (1e-5) Six2(Homeobox)/NephronProgenitor-Six2-ChIP-Seq(GSE39837)/Homer\n",
      "\t\t231 of 1006 (1e-5) ANAC020(NAC)/col-ANAC020-DAP-Seq(GSE60143)/Homer\n",
      "\t\t232 of 1006 (1e-4) ZNF711(Zf)/SHSY5Y-ZNF711-ChIP-Seq(GSE20673)/Homer\n",
      "\t\t233 of 1006 (1e-4) NST1(NAC)/colamp-NST1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t234 of 1006 (1e-4) ANAC016(NAC)/col-ANAC016-DAP-Seq(GSE60143)/Homer\n",
      "\t\t235 of 1006 (1e-4) ZNF7(Zf)/HepG2-ZNF7.Flag-ChIP-Seq(Encode)/Homer\n",
      "\t\t236 of 1006 (1e-4) CEJ1(AP2EREBP)/col-CEJ1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t237 of 1006 (1e-4) AT1G76870(Trihelix)/col-AT1G76870-DAP-Seq(GSE60143)/Homer\n",
      "\t\t238 of 1006 (1e-4) Twist2(bHLH)/Myoblast-Twist2.Ty1-ChIP-Seq(GSE127998)/Homer\n",
      "\t\t239 of 1006 (1e-4) ERF8(AP2EREBP)/colamp-ERF8-DAP-Seq(GSE60143)/Homer\n",
      "\t\t240 of 1006 (1e-4) GATA(Zf),IR4/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t241 of 1006 (1e-4) RAP26(AP2EREBP)/colamp-RAP26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t242 of 1006 (1e-4) CUC1(NAC)/col-CUC1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t243 of 1006 (1e-4) Mef2b(MADS)/HEK293-Mef2b.V5-ChIP-Seq(GSE67450)/Homer\n",
      "\t\t244 of 1006 (1e-4) Oct6(POU,Homeobox)/NPC-Pou3f1-ChIP-Seq(GSE35496)/Homer\n",
      "\t\t245 of 1006 (1e-4) STAT4(Stat)/CD4-Stat4-ChIP-Seq(GSE22104)/Homer\n",
      "\t\t246 of 1006 (1e-4) PCF/Arabidopsis-Promoters/Homer\n",
      "\t\t247 of 1006 (1e-4) AGL6(MADS)/col-AGL6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t248 of 1006 (1e-4) ANAC053(NAC)/colamp-ANAC053-DAP-Seq(GSE60143)/Homer\n",
      "\t\t249 of 1006 (1e-4) REF6(Zf)/Arabidopsis-REF6-ChIP-Seq(GSE106942)/Homer\n",
      "\t\t250 of 1006 (1e-4) Fra2(bZIP)/Striatum-Fra2-ChIP-Seq(GSE43429)/Homer\n",
      "\t\t251 of 1006 (1e-4) PPARE(NR),DR1/3T3L1-Pparg-ChIP-Seq(GSE13511)/Homer\n",
      "\t\t252 of 1006 (1e-4) At4g31060(AP2EREBP)/colamp-At4g31060-DAP-Seq(GSE60143)/Homer\n",
      "\t\t253 of 1006 (1e-4) ERF2(AP2EREBP)/colamp-ERF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t254 of 1006 (1e-4) Tbox:Smad(T-box,MAD)/ESCd5-Smad2_3-ChIP-Seq(GSE29422)/Homer\n",
      "\t\t255 of 1006 (1e-4) AT1G77200(AP2EREBP)/colamp-AT1G77200-DAP-Seq(GSE60143)/Homer\n",
      "\t\t256 of 1006 (1e-4) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
      "\t\t257 of 1006 (1e-4) SPL15(SBP)/colamp-SPL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t258 of 1006 (1e-4) MYB33(MYB)/col-MYB33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t259 of 1006 (1e-4) AT5G23930(mTERF)/col-AT5G23930-DAP-Seq(GSE60143)/Homer\n",
      "\t\t260 of 1006 (1e-4) TGA6(bZIP)/colamp-TGA6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t261 of 1006 (1e-4) AT5G47660(Trihelix)/colamp-AT5G47660-DAP-Seq(GSE60143)/Homer\n",
      "\t\t262 of 1006 (1e-4) ANAC050(NAC)/colamp-ANAC050-DAP-Seq(GSE60143)/Homer\n",
      "\t\t263 of 1006 (1e-4) bZIP42(bZIP)/colamp-bZIP42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t264 of 1006 (1e-4) Initiator/Drosophila-Promoters/Homer\n",
      "\t\t265 of 1006 (1e-3) AT1G12630(AP2EREBP)/colamp-AT1G12630-DAP-Seq(GSE60143)/Homer\n",
      "\t\t266 of 1006 (1e-3) Stat3+il21(Stat)/CD4-Stat3-ChIP-Seq(GSE19198)/Homer\n",
      "\t\t267 of 1006 (1e-3) HOXA9(Homeobox)/HSC-Hoxa9-ChIP-Seq(GSE33509)/Homer\n",
      "\t\t268 of 1006 (1e-3) ANAC092(NAC)/colamp-ANAC092-DAP-Seq(GSE60143)/Homer\n",
      "\t\t269 of 1006 (1e-3) Olig2(bHLH)/Neuron-Olig2-ChIP-Seq(GSE30882)/Homer\n",
      "\t\t270 of 1006 (1e-3) WRKY40(WRKY)/colamp-WRKY40-DAP-Seq(GSE60143)/Homer\n",
      "\t\t271 of 1006 (1e-3) EBF1(EBF)/Near-E2A-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t272 of 1006 (1e-3) JunB(bZIP)/DendriticCells-Junb-ChIP-Seq(GSE36099)/Homer\n",
      "\t\t273 of 1006 (1e-3) Bach1(bZIP)/K562-Bach1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t274 of 1006 (1e-3) EGL-5(Homeobox)/cElegans-L3-EGL5-ChIP-Seq(modEncode)/Homer\n",
      "\t\t275 of 1006 (1e-3) AGL15(MADS)/col-AGL15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t276 of 1006 (1e-3) WRKY65(WRKY)/colamp-WRKY65-DAP-Seq(GSE60143)/Homer\n",
      "\t\t277 of 1006 (1e-3) NF1(CTF)/LNCAP-NF1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t278 of 1006 (1e-3) TGA1(bZIP)/colamp-TGA1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t279 of 1006 (1e-3) WRKY18(WRKY)/col-WRKY18-DAP-Seq(GSE60143)/Homer\n",
      "\t\t280 of 1006 (1e-3) WRKY33(WRKY)/col-WRKY33-DAP-Seq(GSE60143)/Homer\n",
      "\t\t281 of 1006 (1e-3) Sox9(HMG)/Limb-SOX9-ChIP-Seq(GSE73225)/Homer\n",
      "\t\t282 of 1006 (1e-3) Atf3(bZIP)/GBM-ATF3-ChIP-Seq(GSE33912)/Homer\n",
      "\t\t283 of 1006 (1e-3) PAX3:FKHR-fusion(Paired,Homeobox)/Rh4-PAX3:FKHR-ChIP-Seq(GSE19063)/Homer\n",
      "\t\t284 of 1006 (1e-3) dof42(C2C2dof)/col-dof42-DAP-Seq(GSE60143)/Homer\n",
      "\t\t285 of 1006 (1e-3) DDF1(AP2EREBP)/col-DDF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t286 of 1006 (1e-3) SHN3(AP2EREBP)/col-SHN3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t287 of 1006 (1e-3) PUCHI(AP2EREBP)/colamp-PUCHI-DAP-Seq(GSE60143)/Homer\n",
      "\t\t288 of 1006 (1e-3) At1g22810(AP2EREBP)/colamp-At1g22810-DAP-Seq(GSE60143)/Homer\n",
      "\t\t289 of 1006 (1e-3) REST-NRSF(Zf)/Jurkat-NRSF-ChIP-Seq/Homer\n",
      "\t\t290 of 1006 (1e-3) Prop1(Homeobox)/GHFT1-PROP1.biotin-ChIP-Seq(GSE77302)/Homer\n",
      "\t\t291 of 1006 (1e-3) ERF1(AP2EREBP)/colamp-ERF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t292 of 1006 (1e-3) ERF5(AP2EREBP)/colamp-ERF5-DAP-Seq(GSE60143)/Homer\n",
      "\t\t293 of 1006 (1e-3) Stat3(Stat)/mES-Stat3-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t294 of 1006 (1e-3) ERRg(NR)/Kidney-ESRRG-ChIP-Seq(GSE104905)/Homer\n",
      "\t\t295 of 1006 (1e-3) AGL16(MADS)/col-AGL16-DAP-Seq(GSE60143)/Homer\n",
      "\t\t296 of 1006 (1e-3) WRKY15(WRKY)/col-WRKY15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t297 of 1006 (1e-3) ERF15(AP2EREBP)/colamp-ERF15-DAP-Seq(GSE60143)/Homer\n",
      "\t\t298 of 1006 (1e-3) AGL13(MADS)/col-AGL13-DAP-Seq(GSE60143)/Homer\n",
      "\t\t299 of 1006 (1e-3) ERF38(AP2EREBP)/col-ERF38-DAP-Seq(GSE60143)/Homer\n",
      "\t\t300 of 1006 (1e-3) Ronin(THAP)/ES-Thap11-ChIP-Seq(GSE51522)/Homer\n",
      "\t\t301 of 1006 (1e-3) DEAR2(AP2EREBP)/colamp-DEAR2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t302 of 1006 (1e-3) AT4G18450(AP2EREBP)/col-AT4G18450-DAP-Seq(GSE60143)/Homer\n",
      "\t\t303 of 1006 (1e-3) AT5G05550(Trihelix)/col-AT5G05550-DAP-Seq(GSE60143)/Homer\n",
      "\t\t304 of 1006 (1e-3) ERF3(AP2EREBP)/colamp-ERF3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t305 of 1006 (1e-3) Ascl1(bHLH)/NeuralTubes-Ascl1-ChIP-Seq(GSE55840)/Homer\n",
      "\t\t306 of 1006 (1e-3) AP-1(bZIP)/ThioMac-PU.1-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t307 of 1006 (1e-3) Sox10(HMG)/SciaticNerve-Sox3-ChIP-Seq(GSE35132)/Homer\n",
      "\t\t308 of 1006 (1e-3) WRKY24(WRKY)/colamp-WRKY24-DAP-Seq(GSE60143)/Homer\n",
      "\t\t309 of 1006 (1e-3) MYB57(MYB)/col-MYB57-DAP-Seq(GSE60143)/Homer\n",
      "\t\t310 of 1006 (1e-3) CRE(bZIP)/Promoter/Homer\n",
      "\t\t311 of 1006 (1e-3) STAT6(Stat)/Macrophage-Stat6-ChIP-Seq(GSE38377)/Homer\n",
      "\t\t312 of 1006 (1e-3) Phox2a(Homeobox)/Neuron-Phox2a-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t313 of 1006 (1e-3) SND2(NAC)/colamp-SND2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t314 of 1006 (1e-3) RKD2(RWPRK)/colamp-RKD2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t315 of 1006 (1e-3) RAP212(AP2EREBP)/col-RAP212-DAP-Seq(GSE60143)/Homer\n",
      "\t\t316 of 1006 (1e-3) WRKY25(WRKY)/colamp-WRKY25-DAP-Seq(GSE60143)/Homer\n",
      "\t\t317 of 1006 (1e-3) ABR1(AP2EREBP)/colamp-ABR1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t318 of 1006 (1e-3) PRDM14(Zf)/H1-PRDM14-ChIP-Seq(GSE22767)/Homer\n",
      "\t\t319 of 1006 (1e-3) ERF9(AP2EREBP)/colamp-ERF9-DAP-Seq(GSE60143)/Homer\n",
      "\t\t320 of 1006 (1e-2) WT1(Zf)/Kidney-WT1-ChIP-Seq(GSE90016)/Homer\n",
      "\t\t321 of 1006 (1e-2) AT3G60490(AP2EREBP)/colamp-AT3G60490-DAP-Seq(GSE60143)/Homer\n",
      "\t\t322 of 1006 (1e-2) Sox3(HMG)/NPC-Sox3-ChIP-Seq(GSE33059)/Homer\n",
      "\t\t323 of 1006 (1e-2) Sox2(HMG)/mES-Sox2-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t324 of 1006 (1e-2) WRKY14(WRKY)/colamp-WRKY14-DAP-Seq(GSE60143)/Homer\n",
      "\t\t325 of 1006 (1e-2) WRKY6(WRKY)/colamp-WRKY6-DAP-Seq(GSE60143)/Homer\n",
      "\t\t326 of 1006 (1e-2) MYB3R1(MYB)/col-MYB3R1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t327 of 1006 (1e-2) WRKY46(WRKY)/colamp-WRKY46-DAP-Seq(GSE60143)/Homer\n",
      "\t\t328 of 1006 (1e-2) TINY(AP2EREBP)/col-TINY-DAP-Seq(GSE60143)/Homer\n",
      "\t\t329 of 1006 (1e-2) At1g36060(AP2EREBP)/colamp-At1g36060-DAP-Seq(GSE60143)/Homer\n",
      "\t\t330 of 1006 (1e-2) ANAC013(NAC)/col-ANAC013-DAP-Seq(GSE60143)/Homer\n",
      "\t\t331 of 1006 (1e-2) Esrrb(NR)/mES-Esrrb-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t332 of 1006 (1e-2) WRKY26(WRKY)/colamp-WRKY26-DAP-Seq(GSE60143)/Homer\n",
      "\t\t333 of 1006 (1e-2) RRTF1(AP2EREBP)/colamp-RRTF1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t334 of 1006 (1e-2) WRKY75(WRKY)/col-WRKY75-DAP-Seq(GSE60143)/Homer\n",
      "\t\t335 of 1006 (1e-2) At5g18450(AP2EREBP)/col-At5g18450-DAP-Seq(GSE60143)/Homer\n",
      "\t\t336 of 1006 (1e-2) GATA(Zf),IR3/iTreg-Gata3-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t337 of 1006 (1e-2) Rfx5(HTH)/GM12878-Rfx5-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t338 of 1006 (1e-2) STAT5(Stat)/mCD4+-Stat5-ChIP-Seq(GSE12346)/Homer\n",
      "\t\t339 of 1006 (1e-2) CUC3(NAC)/col-CUC3-DAP-Seq(GSE60143)/Homer\n",
      "\t\t340 of 1006 (1e-2) ERF4(AP2EREBP)/colamp-ERF4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t341 of 1006 (1e-2) bZIP44(bZIP)/colamp-bZIP44-DAP-Seq(GSE60143)/Homer\n",
      "\t\t342 of 1006 (1e-2) DDF2(AP2EREBP)/col-DDF2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t343 of 1006 (1e-2) ZNF467(Zf)/HEK293-ZNF467.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t344 of 1006 (1e-2) MYB3R4(MYB)/col-MYB3R4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t345 of 1006 (1e-2) bHLH10(bHLH)/colamp-bHLH10-DAP-Seq(GSE60143)/Homer\n",
      "\t\t346 of 1006 (1e-2) TGA2(bZIP)/colamp-TGA2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t347 of 1006 (1e-2) At5g22890(C2H2)/col-At5g22890-DAP-Seq(GSE60143)/Homer\n",
      "\t\t348 of 1006 (1e-2) MYB3R5(MYB)/col-MYB3R5-DAP-Seq(GSE60143)/Homer\n",
      "\t\t349 of 1006 (1e-2) Rfx1(HTH)/NPC-H3K4me1-ChIP-Seq(GSE16256)/Homer\n",
      "\t\t350 of 1006 (1e-2) RBPJ:Ebox(?,bHLH)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t351 of 1006 (1e-2) Pitx1:Ebox(Homeobox,bHLH)/Hindlimb-Pitx1-ChIP-Seq(GSE41591)/Homer\n",
      "\t\t352 of 1006 (1e-2) GTL1(Trihelix)/colamp-GTL1-DAP-Seq(GSE60143)/Homer\n",
      "\t\t353 of 1006 (1e-2) At2g33710(AP2EREBP)/colamp-At2g33710-DAP-Seq(GSE60143)/Homer\n",
      "\t\t354 of 1006 (1e-2) ZNF317(Zf)/HEK293-ZNF317.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t355 of 1006 (1e-2) MafB(bZIP)/BMM-Mafb-ChIP-Seq(GSE75722)/Homer\n",
      "\t\t356 of 1006 (1e-2) AT1G44830(AP2EREBP)/col-AT1G44830-DAP-Seq(GSE60143)/Homer\n",
      "\t\t357 of 1006 (1e-2) TGA4(bZIP)/colamp-TGA4-DAP-Seq(GSE60143)/Homer\n",
      "\t\t358 of 1006 (1e-2) Nrf2(bZIP)/Lymphoblast-Nrf2-ChIP-Seq(GSE37589)/Homer\n",
      "\t\t359 of 1006 (1e-2) WRKY22(WRKY)/colamp-WRKY22-DAP-Seq(GSE60143)/Homer\n",
      "\t\t360 of 1006 (1e-2) ANAC103(NAC)/col-ANAC103-DAP-Seq(GSE60143)/Homer\n",
      "\t\t361 of 1006 (1e-2) MYB116(MYB)/colamp-MYB116-DAP-Seq(GSE60143)/Homer\n",
      "\t\t362 of 1006 (1e-2) Zfp281(Zf)/ES-Zfp281-ChIP-Seq(GSE81042)/Homer\n",
      "\t\t363 of 1006 (1e-2) WRKY71(WRKY)/col-WRKY71-DAP-Seq(GSE60143)/Homer\n",
      "\t\t364 of 1006 (1e-2) MYB93(MYB)/colamp-MYB93-DAP-Seq(GSE60143)/Homer\n",
      "\t\t365 of 1006 (1e-2) WRKY31(WRKY)/colamp-WRKY31-DAP-Seq(GSE60143)/Homer\n",
      "\t\t366 of 1006 (1e-2) Zac1(Zf)/Neuro2A-Plagl1-ChIP-Seq(GSE75942)/Homer\n",
      "\t\t367 of 1006 (1e-2) AP-2gamma(AP2)/MCF7-TFAP2C-ChIP-Seq(GSE21234)/Homer\n",
      "\t\t368 of 1006 (1e-2) AR-halfsite(NR)/LNCaP-AR-ChIP-Seq(GSE27824)/Homer\n",
      "\t\t369 of 1006 (1e-2) ANAC087(NAC)/col-ANAC087-DAP-Seq(GSE60143)/Homer\n",
      "\t\t370 of 1006 (1e-2) At2g44940(AP2EREBP)/colamp-At2g44940-DAP-Seq(GSE60143)/Homer\n",
      "\t\t371 of 1006 (1e-2) WRKY47(WRKY)/colamp-WRKY47-DAP-Seq(GSE60143)/Homer\n",
      "\t\t372 of 1006 (1e-2) CUC2(NAC)/colamp-CUC2-DAP-Seq(GSE60143)/Homer\n",
      "\t\t373 of 1006 (1e-2) EBNA1(EBV-virus)/Raji-EBNA1-ChIP-Seq(GSE30709)/Homer\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 32889\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 32889 of 34497 max\n",
      "\tTree  : 67212 of 172485 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 32889 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.89% skipped, 98.11% checked (32267 of 32889), of those checked:\n",
      "\t\t1.89% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 32889 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t1.89% skipped, 98.11% checked (32267 of 32889), of those checked:\n",
      "\t\t0.00% not in target, 48.21% increased p-value, 48.66% high p-value\n",
      "\n",
      "\tScreening oligos 32889 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t60.90% skipped, 39.10% checked (12859 of 32889), of those checked:\n",
      "\t\t0.00% not in target, 22.41% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAA... (-1320.415)\n",
      "\t\tRound 1: -2063.94 GCACTTAA T:3488.0(43.11%),B:4942.6(11.04%),P:1e-896\n",
      "\t\tRound 2: -2063.94 GCACTTAA T:3488.0(43.11%),B:4942.6(11.04%),P:1e-896\n",
      "\t\t=Final=: -2350.88 GCACTTAA T:2798.0(45.25%),B:4581.7(10.84%),P:1e-1020\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: AGATAACG... (-878.483)\n",
      "\t\tRound 1: -2366.34 AGATAAGA T:4091.0(48.40%),B:5649.4(12.52%),P:1e-1027\n",
      "\t\tRound 2: -2366.34 AGATAAGA T:4091.0(48.40%),B:5649.4(12.52%),P:1e-1027\n",
      "\t\t=Final=: -2527.11 AGATAAGA T:3060.0(49.48%),B:5224.5(12.37%),P:1e-1097\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: ACATTCCG... (-393.704)\n",
      "\t\tRound 1: -918.48 RCATTCCW T:1857.0(25.94%),B:3460.6(7.86%),P:1e-398\n",
      "\t\tRound 2: -922.32 ACATTCCW T:1961.0(27.18%),B:3773.1(8.54%),P:1e-400\n",
      "\t\tRound 3: -922.32 ACATTCCW T:1961.0(27.18%),B:3773.1(8.54%),P:1e-400\n",
      "\t\t=Final=: -914.49 ACATTCCW T:1666.0(26.94%),B:3576.5(8.47%),P:1e-397\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: GCTGTCAG... (-361.236)\n",
      "\t\tRound 1: -627.59 GCTGTCAG T:4543.0(52.03%),B:15321.8(30.42%),P:1e-272\n",
      "\t\tRound 2: -654.06 SCTGTCAG T:4672.0(53.03%),B:15612.0(30.89%),P:1e-284\n",
      "\t\tRound 3: -726.70 SCTGTCAR T:4840.0(54.28%),B:15618.6(30.90%),P:1e-315\n",
      "\t\tRound 4: -726.70 SCTGTCAR T:4840.0(54.28%),B:15618.6(30.90%),P:1e-315\n",
      "\t\t=Final=: -685.10 SCTGTCAR T:3133.0(50.66%),B:11975.6(28.34%),P:1e-297\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GCTAATGG... (-308.170)\n",
      "\t\tRound 1: -688.99 GCTAATGG T:1149.0(16.96%),B:1915.4(4.43%),P:1e-299\n",
      "\t\tRound 2: -688.99 GCTAATGG T:1149.0(16.96%),B:1915.4(4.43%),P:1e-299\n",
      "\t\t=Final=: -703.79 GCTAATGG T:1055.0(17.06%),B:1862.1(4.41%),P:1e-305\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: CCCCCCCC... (-186.032)\n",
      "\t\tRound 1: -186.03 CCCCCCCC T:166.0(2.65%),B:156.8(0.37%),P:1e-80\n",
      "\t\tRound 2: -186.03 CCCCCCCC T:166.0(2.65%),B:156.8(0.37%),P:1e-80\n",
      "\t\t=Final=: -1.10 CCCCCCCC T:527.0(8.52%),B:3533.6(8.36%),P:1e0\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: CTGAGTGC... (-144.410)\n",
      "\t\tRound 1: -195.34 CTRAKTRC T:533.0(8.26%),B:1321.7(3.08%),P:1e-84\n",
      "\t\tRound 2: -195.34 CTRAKTRC T:533.0(8.26%),B:1321.7(3.08%),P:1e-84\n",
      "\t\t=Final=: -185.16 CTRAKTRC T:497.0(8.04%),B:1287.9(3.05%),P:1e-80\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: TGTTGACA... (-128.486)\n",
      "\t\tRound 1: -235.18 TGTTGACA T:1447.0(20.86%),B:5077.7(11.32%),P:1e-102\n",
      "\t\tRound 2: -235.18 TGTTGACA T:1447.0(20.86%),B:5077.7(11.32%),P:1e-102\n",
      "\t\t=Final=: -224.39 TGTTGACA T:1270.0(20.54%),B:4757.2(11.26%),P:1e-97\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: AATGGAAT... (-123.815)\n",
      "\t\tRound 1: -169.40 AATSGAAT T:247.0(3.92%),B:395.1(0.93%),P:1e-73\n",
      "\t\tRound 2: -169.40 AATSGAAT T:247.0(3.92%),B:395.1(0.93%),P:1e-73\n",
      "\t\t=Final=: -17.74 AATSGAAT T:18.0(0.29%),B:23.3(0.06%),P:1e-7\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: CGCTAATC... (-116.162)\n",
      "\t\tRound 1: -317.96 CGCTAATC T:1557.0(22.26%),B:4962.0(11.08%),P:1e-138\n",
      "\t\tRound 2: -317.96 CGCTAATC T:1557.0(22.26%),B:4962.0(11.08%),P:1e-138\n",
      "\t\t=Final=: -327.26 CGCTAATC T:1368.0(22.12%),B:4585.2(10.85%),P:1e-142\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: TCCATTCG... (-69.681)\n",
      "\t\tRound 1: -91.89 TCCATTCG T:60.0(0.97%),B:37.6(0.09%),P:1e-39\n",
      "\t\tRound 2: -91.89 TCCATTCG T:60.0(0.97%),B:37.6(0.09%),P:1e-39\n",
      "\t\t=Final=: -36.16 TCCATTCG T:33.0(0.53%),B:35.6(0.08%),P:1e-15\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: AAAGCACT... (-58.580)\n",
      "\t\tRound 1: -117.14 AAAGCACT T:1312.0(19.12%),B:5580.7(12.37%),P:1e-50\n",
      "\t\tRound 2: -117.14 AAAGCACT T:1312.0(19.12%),B:5580.7(12.37%),P:1e-50\n",
      "\t\t=Final=: -114.61 AAAGCACT T:1176.0(19.02%),B:5221.7(12.36%),P:1e-49\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: CATTCCAT... (-54.928)\n",
      "\t\tRound 1: -66.77 CATTCCAT T:200.0(3.18%),B:537.4(1.26%),P:1e-28\n",
      "\t\tRound 2: -66.77 CATTCCAT T:200.0(3.18%),B:537.4(1.26%),P:1e-28\n",
      "\t\t=Final=: -12.66 CATTCCAT T:116.0(1.88%),B:505.2(1.20%),P:1e-5\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TGACAGCT... (-44.707)\n",
      "\t\tRound 1: -62.83 TGACAGCT T:753.0(11.47%),B:3312.5(7.54%),P:1e-27\n",
      "\t\tRound 2: -62.83 TGACAGCT T:753.0(11.47%),B:3312.5(7.54%),P:1e-27\n",
      "\t\t=Final=: -59.01 TGACAGCT T:695.0(11.24%),B:3153.6(7.46%),P:1e-25\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: TTAAGAGG... (-41.914)\n",
      "\t\tRound 1: -67.93 TTAAGAGG T:241.0(3.82%),B:706.2(1.66%),P:1e-29\n",
      "\t\tRound 2: -67.93 TTAAGAGG T:241.0(3.82%),B:706.2(1.66%),P:1e-29\n",
      "\t\t=Final=: -71.24 TTAAGAGG T:229.0(3.70%),B:653.1(1.55%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: CGCCGCCG... (-39.505)\n",
      "\t\tRound 1: -76.78 CGCCGCCG T:343.0(5.40%),B:1108.8(2.59%),P:1e-33\n",
      "\t\tRound 2: -76.92 SGCCGCCG T:505.0(7.84%),B:1876.6(4.34%),P:1e-33\n",
      "\t\tRound 3: -96.00 SGCCGCSG T:396.0(6.20%),B:1230.8(2.87%),P:1e-41\n",
      "\t\tRound 4: -96.86 GGCCGCSS T:646.0(9.92%),B:2400.6(5.52%),P:1e-42\n",
      "\t\tRound 5: -96.86 GGCCGCSS T:646.0(9.92%),B:2400.6(5.52%),P:1e-42\n",
      "\t\t=Final=: -30.18 GGCCGCSS T:292.0(4.72%),B:1262.2(2.99%),P:1e-13\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: TTATCACT... (-38.907)\n",
      "\t\tRound 1: -117.57 TTATCACT T:579.0(8.94%),B:1923.5(4.45%),P:1e-51\n",
      "\t\tRound 2: -168.51 TTATCACA T:1187.0(17.47%),B:4410.4(9.91%),P:1e-73\n",
      "\t\tRound 3: -168.51 TTATCACA T:1187.0(17.47%),B:4410.4(9.91%),P:1e-73\n",
      "\t\t=Final=: -171.17 TTATCACA T:1077.0(17.42%),B:4150.9(9.82%),P:1e-74\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: GAGTGTTT... (-37.694)\n",
      "\t\tRound 1: -89.87 GAGTGCTT T:1247.0(18.26%),B:5600.2(12.41%),P:1e-39\n",
      "\t\tRound 2: -89.87 GAGTGCTT T:1247.0(18.26%),B:5600.2(12.41%),P:1e-39\n",
      "\t\t=Final=: -104.20 GAGTGCTT T:1144.0(18.50%),B:5157.9(12.21%),P:1e-45\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: AGGGCCTG... (-30.326)\n",
      "\t\tRound 1: -48.62 AGGGCCTG T:822.0(12.45%),B:3894.3(8.80%),P:1e-21\n",
      "\t\tRound 2: -48.62 AGGGCCTG T:822.0(12.45%),B:3894.3(8.80%),P:1e-21\n",
      "\t\t=Final=: -40.47 AGGGCCTG T:841.0(13.60%),B:4277.3(10.12%),P:1e-17\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: GTTAATTA... (-27.250)\n",
      "\t\tRound 1: -100.85 GTTAATTA T:742.0(11.31%),B:2840.3(6.50%),P:1e-43\n",
      "\t\tRound 2: -100.85 GTTAATTA T:742.0(11.31%),B:2840.3(6.50%),P:1e-43\n",
      "\t\t=Final=: -70.82 GTTAATTA T:628.0(10.16%),B:2649.3(6.27%),P:1e-30\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: GTCAAGTG... (-26.233)\n",
      "\t\tRound 1: -41.70 GTCAAGTG T:274.0(4.33%),B:1037.7(2.42%),P:1e-18\n",
      "\t\tRound 2: -48.93 GTGGAGTG T:314.0(4.95%),B:1174.7(2.74%),P:1e-21\n",
      "\t\tRound 3: -48.93 GTGGAGTG T:314.0(4.95%),B:1174.7(2.74%),P:1e-21\n",
      "\t\t=Final=: -53.23 GTGGAGTG T:309.0(5.00%),B:1138.2(2.69%),P:1e-23\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: GCCGTTTG... (-25.607)\n",
      "\t\tRound 1: -65.60 GCCGTTTG T:839.0(12.69%),B:3735.9(8.46%),P:1e-28\n",
      "\t\tRound 2: -74.94 GCCRTTTG T:1226.0(17.99%),B:5714.3(12.65%),P:1e-32\n",
      "\t\tRound 3: -79.16 GCCRTTWG T:1093.0(16.20%),B:4918.6(10.99%),P:1e-34\n",
      "\t\tRound 4: -80.61 GCCRTTTR T:1715.0(24.22%),B:8332.8(17.90%),P:1e-35\n",
      "\t\tRound 5: -80.61 GCCRTTTR T:1715.0(24.22%),B:8332.8(17.90%),P:1e-35\n",
      "\t\t=Final=: -85.33 GCCRTTTR T:1336.0(21.60%),B:6521.9(15.44%),P:1e-37\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: CCCTATCT... (-25.263)\n",
      "\t\tRound 1: -37.82 CCYTATCT T:335.0(5.27%),B:1386.9(3.23%),P:1e-16\n",
      "\t\tRound 2: -41.77 CCYTATCY T:103.0(1.65%),B:253.8(0.60%),P:1e-18\n",
      "\t\tRound 3: -41.77 CCYTATCY T:103.0(1.65%),B:253.8(0.60%),P:1e-18\n",
      "\t\t=Final=: -30.97 CCYTATCY T:91.0(1.47%),B:252.4(0.60%),P:1e-13\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: TAAGTGGC... (-23.495)\n",
      "\t\tRound 1: -45.45 TAAGTGGC T:877.0(13.22%),B:4258.6(9.59%),P:1e-19\n",
      "\t\tRound 2: -59.36 TAAGTGGY T:632.0(9.72%),B:2704.8(6.20%),P:1e-25\n",
      "\t\tRound 3: -59.36 TAAGTGGY T:632.0(9.72%),B:2704.8(6.20%),P:1e-25\n",
      "\t\t=Final=: -60.26 TAAGTGGY T:600.0(9.70%),B:2609.6(6.18%),P:1e-26\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: ACATTCTT... (-21.699)\n",
      "\t\tRound 1: -57.83 ACATTCTT T:504.0(7.83%),B:2055.3(4.75%),P:1e-25\n",
      "\t\tRound 2: -57.83 ACATTCTT T:504.0(7.83%),B:2055.3(4.75%),P:1e-25\n",
      "\t\t=Final=: -56.29 ACATTCTT T:480.0(7.76%),B:1999.7(4.73%),P:1e-24\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: homer-results/isl1/homerMotifs.motifs8\n",
      "\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 480945\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 480945 of 551887 max\n",
      "\tTree  : 997508 of 2759435 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 480945 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.43% skipped, 67.57% checked (324956 of 480945), of those checked:\n",
      "\t\t32.43% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 480945 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t32.43% skipped, 67.57% checked (324956 of 480945), of those checked:\n",
      "\t\t0.00% not in target, 30.49% increased p-value, 31.90% high p-value\n",
      "\n",
      "\tScreening oligos 480945 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t73.94% skipped, 26.06% checked (125332 of 480945), of those checked:\n",
      "\t\t0.00% not in target, 7.34% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49998 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: GCACTTAACG... (-1158.245)\n",
      "\t\tRound 1: -2116.16 GCACTTAACG T:3085.0(39.28%),B:3881.6(8.78%),P:1e-919\n",
      "\t\tRound 2: -2275.70 GCACTTARCN T:3443.0(42.70%),B:4371.8(9.83%),P:1e-988\n",
      "\t\tRound 3: -2275.70 GCACTTARCN T:3443.0(42.70%),B:4371.8(9.83%),P:1e-988\n",
      "\t\t=Final=: -2568.91 GCACTTARCN T:2776.0(44.89%),B:4108.1(9.72%),P:1e-1115\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: AGATAAGGCG... (-496.229)\n",
      "\t\tRound 1: -1787.57 AGATAAGGCG T:2436.0(32.56%),B:3012.1(6.88%),P:1e-776\n",
      "\t\tRound 2: -2068.88 AGATAAGVSN T:3038.0(38.82%),B:3866.4(8.74%),P:1e-898\n",
      "\t\tRound 3: -2068.88 AGATAAGVSN T:3038.0(38.82%),B:3866.4(8.74%),P:1e-898\n",
      "\t\t=Final=: -2265.01 AGATAAGVSN T:2586.0(41.82%),B:3982.7(9.43%),P:1e-983\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: CGAATGGAAT... (-310.955)\n",
      "\t\tRound 1: -422.55 CGAATGGAAT T:1242.0(18.20%),B:3097.2(7.07%),P:1e-183\n",
      "\t\tRound 2: -565.82 SGAATGGAHT T:1624.0(23.10%),B:3914.2(8.85%),P:1e-245\n",
      "\t\tRound 3: -662.33 SKAATGGAAT T:2383.0(31.98%),B:6310.4(13.87%),P:1e-287\n",
      "\t\tRound 4: -764.38 SGAATGKHHT T:2192.0(29.85%),B:5125.4(11.42%),P:1e-331\n",
      "\t\tRound 5: -764.38 SGAATGKHHT T:2192.0(29.85%),B:5125.4(11.42%),P:1e-331\n",
      "\t\t=Final=: -616.66 SGAATGKHHT T:2168.0(35.06%),B:7032.7(16.65%),P:1e-267\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: GGGGGGGGGG... (-298.127)\n",
      "\t\tRound 1: -298.13 GGGGGGGGGG T:149.0(2.38%),B:54.2(0.13%),P:1e-129\n",
      "\t\tRound 2: -298.13 GGGGGGGGGG T:149.0(2.38%),B:54.2(0.13%),P:1e-129\n",
      "\t\t=Final=: -3.16 GGGGGGGGGG T:93.0(1.50%),B:527.1(1.25%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: GTGTTGACAG... (-264.313)\n",
      "\t\tRound 1: -453.95 GTGTTGACAG T:1316.0(19.17%),B:3242.0(7.39%),P:1e-197\n",
      "\t\tRound 2: -502.53 GTGYTGACAG T:1418.0(20.49%),B:3419.5(7.77%),P:1e-218\n",
      "\t\tRound 3: -516.54 GTGTTGACAG T:1419.0(20.51%),B:3366.5(7.66%),P:1e-224\n",
      "\t\tRound 4: -516.54 GTGTTGACAG T:1419.0(20.51%),B:3366.5(7.66%),P:1e-224\n",
      "\t\t=Final=: -541.68 GTGTTGACAG T:1284.0(20.76%),B:3210.9(7.60%),P:1e-235\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: TAAGTGCTTT... (-172.589)\n",
      "\t\tRound 1: -270.63 TAAGTGCTTT T:3894.0(46.73%),B:16640.5(32.56%),P:1e-117\n",
      "\t\tRound 2: -336.83 TAAGTGCTTT T:3164.0(40.05%),B:12193.8(25.07%),P:1e-146\n",
      "\t\tRound 3: -343.94 TAAGTGCTTT T:3246.0(40.84%),B:12501.6(25.61%),P:1e-149\n",
      "\t\tRound 4: -352.85 TAAGTGCYYT T:3106.0(39.49%),B:11740.0(24.26%),P:1e-153\n",
      "\t\tRound 5: -352.85 TAAGTGCYYT T:3106.0(39.49%),B:11740.0(24.26%),P:1e-153\n",
      "\t\t=Final=: -374.85 TAAGTGCYYT T:2528.0(40.88%),B:10582.1(25.05%),P:1e-162\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: AGATAAGATA... (-125.192)\n",
      "\t\tRound 1: -44.12 AGATAAGATA T:390.0(6.11%),B:1601.9(3.72%),P:1e-19\n",
      "\t\tRound 2: -88.11 AGATAAGATA T:160.0(2.55%),B:310.4(0.73%),P:1e-38\n",
      "\t\tRound 3: -188.66 AGATAAGATA T:1895.0(26.40%),B:7706.6(16.67%),P:1e-81\n",
      "\t\tRound 4: -284.55 WGATWAMHDH T:1681.0(23.80%),B:5756.5(12.74%),P:1e-123\n",
      "\t\tRound 5: -284.55 WGATWAMHDH T:1681.0(23.80%),B:5756.5(12.74%),P:1e-123\n",
      "\t\t=Final=: -335.77 WGATWAMHDH T:1537.0(24.85%),B:5387.5(12.75%),P:1e-145\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: GGCCTGTCAG... (-100.835)\n",
      "\t\tRound 1: -159.32 GGCCTGTCAG T:538.0(8.33%),B:1504.6(3.50%),P:1e-69\n",
      "\t\tRound 2: -206.78 GGCCTGTCAG T:1818.0(25.47%),B:7121.5(15.51%),P:1e-89\n",
      "\t\tRound 3: -269.43 RGCCTGTCAG T:1767.0(24.86%),B:6282.2(13.82%),P:1e-117\n",
      "\t\tRound 4: -298.53 NGSCTGTCAG T:2021.0(27.88%),B:7210.0(15.69%),P:1e-129\n",
      "\t\tRound 5: -317.48 NGSCTGTCAG T:2836.0(36.79%),B:10852.7(22.65%),P:1e-137\n",
      "\t\tRound 6: -317.48 NGSCTGTCAG T:2836.0(36.79%),B:10852.7(22.65%),P:1e-137\n",
      "\t\t=Final=: -254.31 NGSCTGTCAG T:1929.0(31.19%),B:8129.8(19.24%),P:1e-110\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: TTAAGAGCTG... (-80.938)\n",
      "\t\tRound 1: -21.07 TTAAGAGCTG T:442.0(6.90%),B:2213.6(5.10%),P:1e-9\n",
      "\t\tRound 2: -33.12 TWAAGAGCTG T:665.0(10.20%),B:3274.9(7.46%),P:1e-14\n",
      "\t\tRound 3: -41.40 YWAAGAGCTG T:352.0(5.53%),B:1437.9(3.34%),P:1e-17\n",
      "\t\tRound 4: -41.40 YWAAGAGCTG T:352.0(5.53%),B:1437.9(3.34%),P:1e-17\n",
      "\t\t=Final=: -40.77 YWAAGAGCTG T:338.0(5.47%),B:1397.8(3.31%),P:1e-17\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: CACATTCCTG... (-68.963)\n",
      "\t\tRound 1: -119.36 CACATTCCTG T:284.0(4.49%),B:659.7(1.55%),P:1e-51\n",
      "\t\tRound 2: -265.85 SACATTCCTG T:1008.0(15.04%),B:2919.8(6.68%),P:1e-115\n",
      "\t\tRound 3: -276.97 CACATTCCTG T:934.0(14.02%),B:2553.0(5.86%),P:1e-120\n",
      "\t\tRound 4: -276.97 CACATTCCTG T:934.0(14.02%),B:2553.0(5.86%),P:1e-120\n",
      "\t\t=Final=: -272.06 CACATTCCTG T:864.0(13.97%),B:2490.1(5.89%),P:1e-118\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: GATAAGATAC... (-66.332)\n",
      "\t\tRound 1: -71.15 GATAAGATAC T:19.0(0.31%),B:2.2(0.00%),P:1e-30\n",
      "\t\tRound 2: -71.49 GATAAGCTAC T:22.0(0.36%),B:3.6(0.01%),P:1e-31\n",
      "\t\tRound 3: -71.49 GATAAGCTAC T:22.0(0.36%),B:3.6(0.01%),P:1e-31\n",
      "\t\t=Final=: -67.03 GATAAGCTAC T:22.0(0.36%),B:3.6(0.01%),P:1e-29\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: CCATTAAGCG... (-63.615)\n",
      "\t\tRound 1: -85.18 CCATTAAGCG T:869.0(13.11%),B:3648.1(8.27%),P:1e-36\n",
      "\t\tRound 2: -111.37 CCATTAAGCG T:1257.0(18.40%),B:5363.6(11.92%),P:1e-48\n",
      "\t\tRound 3: -146.06 CCATTAAGHG T:1413.0(20.43%),B:5758.9(12.74%),P:1e-63\n",
      "\t\tRound 4: -196.99 CCATTAASHG T:1015.0(15.14%),B:3375.8(7.68%),P:1e-85\n",
      "\t\tRound 5: -229.53 CMATTARSHN T:1037.0(15.44%),B:3254.6(7.41%),P:1e-99\n",
      "\t\tRound 6: -229.53 CMATTARSHN T:1037.0(15.44%),B:3254.6(7.41%),P:1e-99\n",
      "\t\t=Final=: -239.25 CMATTARSHN T:964.0(15.59%),B:3123.3(7.39%),P:1e-103\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: TGTTTACTCA... (-61.688)\n",
      "\t\tRound 1: -103.32 TGTTTACTCA T:99.0(1.59%),B:106.0(0.25%),P:1e-44\n",
      "\t\tRound 2: -103.65 YGTTTACHCR T:105.0(1.68%),B:119.9(0.28%),P:1e-45\n",
      "\t\tRound 3: -104.81 TGTTKACHCR T:95.0(1.52%),B:94.1(0.22%),P:1e-45\n",
      "\t\tRound 4: -104.81 TGTTKACHCR T:95.0(1.52%),B:94.1(0.22%),P:1e-45\n",
      "\t\t=Final=: -92.68 TGTTKACHCR T:82.0(1.33%),B:80.6(0.19%),P:1e-40\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TGCATTCCTG... (-55.225)\n",
      "\t\tRound 1: -11.54 TGCATTCCTG T:27.0(0.44%),B:67.9(0.16%),P:1e-5\n",
      "\t\tRound 2: -13.92 TGCATTCCTG T:50.0(0.81%),B:155.8(0.37%),P:1e-6\n",
      "\t\tRound 3: -17.90 TGCATTCTTG T:136.0(2.18%),B:551.0(1.29%),P:1e-7\n",
      "\t\tRound 4: -19.23 TGCATTCTTG T:139.0(2.22%),B:553.6(1.30%),P:1e-8\n",
      "\t\tRound 5: -24.95 TRVATTCTYG T:108.0(1.73%),B:358.7(0.84%),P:1e-10\n",
      "\t\tRound 6: -27.06 TACATTCTTG T:92.0(1.48%),B:272.2(0.64%),P:1e-11\n",
      "\t\tRound 7: -27.06 TACATTCTTG T:92.0(1.48%),B:272.2(0.64%),P:1e-11\n",
      "\t\t=Final=: -27.92 TACATTCTTG T:92.0(1.49%),B:271.8(0.64%),P:1e-12\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: TTGGAAACGG... (-52.245)\n",
      "\t\tRound 1: -61.57 TTGGAAACGG T:17.0(0.27%),B:1.6(0.00%),P:1e-26\n",
      "\t\tRound 2: -71.15 TTGGAAACGG T:19.0(0.31%),B:2.5(0.00%),P:1e-30\n",
      "\t\tRound 3: -80.94 TCGTAAACGG T:21.0(0.34%),B:1.6(0.00%),P:1e-35\n",
      "\t\tRound 4: -80.94 TCGTAAACGG T:21.0(0.34%),B:1.6(0.00%),P:1e-35\n",
      "\t\t=Final=: -85.91 TCGTAAACGG T:21.0(0.34%),B:1.6(0.00%),P:1e-37\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: GAGTGATTAG... (-50.997)\n",
      "\t\tRound 1: -85.18 GAGTGATTAG T:633.0(9.73%),B:2438.2(5.61%),P:1e-36\n",
      "\t\tRound 2: -96.49 GAGTGATTAG T:654.0(10.04%),B:2442.2(5.62%),P:1e-41\n",
      "\t\tRound 3: -110.24 GAGTGATTAV T:1129.0(16.69%),B:4713.7(10.56%),P:1e-47\n",
      "\t\tRound 4: -116.06 GAGTGATTAR T:984.0(14.71%),B:3902.0(8.82%),P:1e-50\n",
      "\t\tRound 5: -121.39 RRGTGATTAV T:1823.0(25.53%),B:8224.4(17.69%),P:1e-52\n",
      "\t\tRound 6: -129.82 RRGTGATTAV T:1508.0(21.64%),B:6443.7(14.14%),P:1e-56\n",
      "\t\tRound 7: -129.82 RRGTGATTAV T:1508.0(21.64%),B:6443.7(14.14%),P:1e-56\n",
      "\t\t=Final=: -124.25 RRGTGATTAV T:1292.0(20.89%),B:5774.2(13.67%),P:1e-53\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: GCACAAGTTA... (-47.681)\n",
      "\t\tRound 1: -51.65 GCGCMAGTTA T:19.0(0.31%),B:4.9(0.01%),P:1e-22\n",
      "\t\tRound 2: -66.33 GCGCAAGTTA T:18.0(0.29%),B:2.4(0.00%),P:1e-28\n",
      "\t\tRound 3: -66.33 GCGCAAGTTA T:18.0(0.29%),B:2.4(0.00%),P:1e-28\n",
      "\t\t=Final=: -58.81 GCGCAAGTTA T:18.0(0.29%),B:2.4(0.01%),P:1e-25\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: CTAAGAGCTG... (-47.681)\n",
      "\t\tRound 1: -8.31 CTAAGAGCTG T:244.0(3.87%),B:1316.6(3.07%),P:1e-3\n",
      "\t\tRound 2: -15.04 CTVAGYGCTG T:56.0(0.90%),B:176.1(0.42%),P:1e-6\n",
      "\t\tRound 3: -22.41 CTVAGYGCTR T:124.0(1.99%),B:449.2(1.06%),P:1e-9\n",
      "\t\tRound 4: -29.83 CTRAGYGCTR T:91.0(1.46%),B:254.2(0.60%),P:1e-12\n",
      "\t\tRound 5: -43.80 CTRAGYGCTG T:169.0(2.70%),B:520.5(1.22%),P:1e-19\n",
      "\t\tRound 6: -43.80 CTRAGYGCTG T:169.0(2.70%),B:520.5(1.22%),P:1e-19\n",
      "\t\t=Final=: -40.93 CTRAGYGCTG T:124.0(2.01%),B:346.3(0.82%),P:1e-17\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: TATCCTTATC... (-47.681)\n",
      "\t\tRound 1: -61.57 TATCCTTATC T:17.0(0.27%),B:2.9(0.00%),P:1e-26\n",
      "\t\tRound 2: -61.57 TATCCTTATC T:17.0(0.27%),B:2.9(0.00%),P:1e-26\n",
      "\t\t=Final=: -14.20 TATCCTTATC T:6.0(0.10%),B:2.9(0.01%),P:1e-6\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: CACATTCCTC... (-46.151)\n",
      "\t\tRound 1: -4.48 CACATTCCTC T:63.0(1.01%),B:313.4(0.74%),P:1e-1\n",
      "\t\tRound 2: -11.09 CACATTCCYC T:19.0(0.31%),B:38.9(0.09%),P:1e-4\n",
      "\t\tRound 3: -11.09 CACATTCCYC T:19.0(0.31%),B:38.9(0.09%),P:1e-4\n",
      "\t\t=Final=: -11.99 CACATTCCYC T:19.0(0.31%),B:38.9(0.09%),P:1e-5\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: CTAAGAGCTC... (-43.192)\n",
      "\t\tRound 1: -12.55 SKAMGCGCGK T:41.0(0.66%),B:122.9(0.29%),P:1e-5\n",
      "\t\tRound 2: -23.45 GGAASYGCGG T:20.0(0.32%),B:19.4(0.04%),P:1e-10\n",
      "\t\tRound 3: -39.59 SGAACKGCGG T:19.0(0.31%),B:7.5(0.02%),P:1e-17\n",
      "\t\tRound 4: -40.87 CGAGCGGCGG T:36.0(0.58%),B:35.9(0.08%),P:1e-17\n",
      "\t\tRound 5: -42.99 CTAMSKGCGG T:86.0(1.38%),B:184.3(0.43%),P:1e-18\n",
      "\t\tRound 6: -42.99 CTAMSKGCGG T:86.0(1.38%),B:184.3(0.43%),P:1e-18\n",
      "\t\t=Final=: -44.17 CTAMSKGCGG T:86.0(1.39%),B:183.6(0.43%),P:1e-19\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: GGCGCCGCCG... (-39.998)\n",
      "\t\tRound 1: -74.45 GGCGCCGCCG T:386.0(6.05%),B:1325.9(3.09%),P:1e-32\n",
      "\t\tRound 2: -87.35 GGCGCCGCCG T:213.0(3.39%),B:507.7(1.19%),P:1e-37\n",
      "\t\tRound 3: -87.35 GGCGCCGCCG T:213.0(3.39%),B:507.7(1.19%),P:1e-37\n",
      "\t\t=Final=: -27.48 GGCGCCGCCG T:111.0(1.79%),B:360.0(0.85%),P:1e-11\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: TGATAAGCTC... (-38.783)\n",
      "\t\tRound 1: -15.84 TGATAAGCTC T:158.0(2.52%),B:692.2(1.62%),P:1e-6\n",
      "\t\tRound 2: -23.53 TGATAAGCTC T:144.0(2.30%),B:541.2(1.27%),P:1e-10\n",
      "\t\tRound 3: -27.92 KGATAADCTC T:162.0(2.59%),B:592.8(1.39%),P:1e-12\n",
      "\t\tRound 4: -27.92 KGATAADCTC T:162.0(2.59%),B:592.8(1.39%),P:1e-12\n",
      "\t\t=Final=: -26.36 KGATAADCTC T:169.0(2.73%),B:650.4(1.54%),P:1e-11\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: CTAAGAGCTA... (-34.461)\n",
      "\t\tRound 1: -14.59 CTAAGAGCTA T:219.0(3.48%),B:1047.1(2.45%),P:1e-6\n",
      "\t\tRound 2: -17.04 CTAAGAGCTA T:353.0(5.55%),B:1776.9(4.12%),P:1e-7\n",
      "\t\tRound 3: -17.23 CTAAGAGCTA T:330.0(5.20%),B:1638.3(3.80%),P:1e-7\n",
      "\t\tRound 4: -18.00 CTAAGAGCWA T:156.0(2.49%),B:657.1(1.54%),P:1e-7\n",
      "\t\tRound 5: -18.51 CTAAGAGCTA T:111.0(1.78%),B:420.7(0.99%),P:1e-8\n",
      "\t\tRound 6: -18.51 CTAAGAGCTA T:111.0(1.78%),B:420.7(0.99%),P:1e-8\n",
      "\t\t=Final=: -17.67 CTAAGAGCTA T:109.0(1.76%),B:419.8(0.99%),P:1e-7\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: TACCATTAGG... (-34.461)\n",
      "\t\tRound 1: -8.00 TACCATTAGG T:58.0(0.93%),B:239.3(0.56%),P:1e-3\n",
      "\t\tRound 2: -13.87 TACCATTAGG T:97.0(1.56%),B:388.6(0.91%),P:1e-6\n",
      "\t\tRound 3: -13.87 TACCATTAGG T:97.0(1.56%),B:388.6(0.91%),P:1e-6\n",
      "\t\t=Final=: -13.16 TACCATTAGG T:97.0(1.57%),B:398.1(0.94%),P:1e-5\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "!!! Could not open group file: homer-results/isl1/group.adj !!!\n",
      "\n",
      "\t-blen automatically set to 2\n",
      "\tScanning input files...\n",
      "!!! Could not open group file: homer-results/isl1/group.adj !!!\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 25 reduced to 10 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 10 (1e-1121) similar to GATA2/MA0036.3/Jaspar(0.992)\n",
      "\t\t2 of 10 (1e-1020) similar to ISL2/MA0914.1/Jaspar(0.963)\n",
      "\t\t3 of 10 (1e-437) similar to TEAD3/MA0808.1/Jaspar(0.978)\n",
      "\t\t4 of 10 (1e-383) similar to MEIS2/MA0774.1/Jaspar(0.867)\n",
      "\t\t5 of 10 (1e-355) similar to Isl1/MA1608.1/Jaspar(0.977)\n",
      "\t\t6 of 10 (1e-236) similar to Bapx1(Homeobox)/VertebralCol-Bapx1-ChIP-Seq(GSE36672)/Homer(0.851)\n",
      "\t\t7 of 10 (1e-108) similar to ATHB24(ZFHD)/colamp-ATHB24-DAP-Seq(GSE60143)/Homer(0.922)\n",
      "\t\t8 of 10 (1e-48) similar to ZFX(Zf)/mES-Zfx-ChIP-Seq(GSE11431)/Homer(0.693)\n",
      "\t\t9 of 10 (1e-16) similar to ERF094/MA1049.1/Jaspar(0.862)\n",
      "\t\t10 of 10 (1e0) similar to Maz(Zf)/HepG2-Maz-ChIP-Seq(GSE31477)/Homer(0.974)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n",
      "memusg: vmpeak: 1146152 kb\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "DIR=`cat scriptpath.txt`\n",
    "export PATH=$PATH:$DIR\n",
    "\n",
    "# pyxis\n",
    "./memusg pyxis analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed benchmark/hg19.fa benchmark/isl1.pwms > /dev/null\n",
    "# HOMER\n",
    "currdir=$(pwd)\n",
    "./memusg findMotifsGenome.pl $currdir'/analysis/GSM5838045_S1_ISL1_WT_d6CP_Rep1.bed' $currdir'/benchmark/hg19.fa' homer-results/isl1 > /dev/null"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0f180dae",
   "metadata": {},
   "source": [
    "#### Summary\n",
    "\n",
    "Memory Usage between `pyxis` vs. `HOMER` is as follows:\n",
    "\n",
    "| Tool | Dataset | Command | Vmpeak |\n",
    "| :--- | :------ | :------ | :----: |\n",
    "| Pyxis | Test Data | Basic Usage | 17959672 kb |\n",
    "| Pyxis | Test Data | Basic Usage + Background | 17959676 kb |\n",
    "| Pyxis | Test Data | Basic Usage + Background + Seqlogo | 17959680 kb |\n",
    "| Pyxis | ISL1 | Basic Usage | 17966576 kb |\n",
    "| HOMER | Test Data | Basic Usage | 167048 kb |\n",
    "| HOMER | ISL1 | Basic Usage | 1146152 kb |"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
